---
title: "Cyclotides Transcriptomics"
author: "Karin Steffen"
date: "29/06/2020"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Checking Macrogen work

Macrogen did RNA extraction, library preparation, sequencing and transcriptome assembly.

Sample number: 3
Species: ???
Library kit: TruSeq Stranded mRNA LT Sample Prep Kit
Instrument/Protocol: NovaSeq 6000
Qualtity encoding: phred33

## Assembly QC

 - TransRate
 - BUSCO

[Transrate](http://hibberdlab.com/transrate/getting_started.html) suggests trimmed and normalised reads, so we're going to do that.

Run trimmomatic, as I couldn't find trimmed reads
```{bash trimmomatic, eval=F}
#!/bin/bash
#SBATCH -A snic2017-7-396 
#SBATCH -p core
#SBATCH -n 4
#SBATCH -t 23:50:00
#SBATCH --mail-user karin.steffen@ilk.uu.se
#SBATCH --mail-type=ALL
#SBATCH -J transrate_cyclotides

module load bioinfo-tools trimmomatic/0.36 khmer/2.1.1

cp /proj/uppstore2017151/data/transcriptomes/cyclotides/VS-RL_A_1.fastq.gz $SNIC_TMP
cp /proj/uppstore2017151/data/transcriptomes/cyclotides/VS-RL_A_2.fastq.gz $SNIC_TMP

cd $SNIC_TMP

# trimmomatic:  trimmomatic/0.36
# Adapter seqs: /sw/bioinfo/trimmomatic/0.36/rackham/adapters/TruSeq2-PE.fa
#    			/sw/bioinfo/trimmomatic/0.36/rackham/adapters/TruSeq3-PE.fa
#    			/sw/bioinfo/trimmomatic/0.36/rackham/adapters/TruSeq3-PE-2.fa

java -jar $TRIMMOMATIC_HOME/trimmomatic.jar PE -threads 4 VS-RL_A_1.fastq.gz VS-RL_A_2.fastq.gz trimmomatic_paired.VS-RL_A_1.fq.gz trimmomatic_unpaired.VS-RL_A_1.fq.gz trimmomatic_paired.VS-RL_A_2.fq.gz trimmomatic_unpaired.VS-RL_A_2.fq.gz ILLUMINACLIP:/sw/bioinfo/trimmomatic/0.36/rackham/adapters/TruSeq3-PE-2.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36

cp trimmomatic_paired.VS-RL_A_1.fq.gz /proj/uppstore2017151/analyses/cyclotides
cp trimmomatic_paired.VS-RL_A_2.fq.gz /proj/uppstore2017151/analyses/cyclotides

```
This runs about 1 hour.

_In silico_ read normalisation
```{bash khmer, eval=F}
#!/bin/bash
#SBATCH -A snic2017-7-396 
#SBATCH -p core
#SBATCH -n 4
#SBATCH -t 23:50:00
#SBATCH --mail-user karin.steffen@ilk.uu.se
#SBATCH --mail-type=ALL
#SBATCH -J transrate_cyclotides

module load bioinfo-tools trimmomatic/0.36 khmer/2.1.1

cp /proj/uppstore2017151/analyses/cyclotides/trimmomatic_paired.VS-RL_A_1.fq.gz $SNIC_TMP
cp /proj/uppstore2017151/analyses/cyclotides/trimmomatic_paired.VS-RL_A_2.fq.gz $SNIC_TMP

cd $SNIC_TMP

# normalisation: khmer/2.1.1 
# -M 23e9 is 23 Gb
# Discard sequences based on whether or not their median k-mer abundance lies above a specified cutoff. 
# Kept sequences will be placed in <fileN>.keep.
# By default, paired end reads will be considered together; if either read should be kept, both will be kept. 

interleave-reads.py trimmomatic_paired.VS-RL_A_1.fq.gz trimmomatic_paired.VS-RL_A_2.fq.gz --gzip -o trimmomatic_interleaved.VS-RL_A.fq.gz
normalize-by-median.py -p -k 31 -C 20 -M 23e9 trimmomatic_interleaved.VS-RL_A.fq.gz

rm trimmomatic_paired.VS-RL_A_1.fq.gz
rm trimmomatic_paired.VS-RL_A_2.fq.gz
rm trimmomatic_interleaved.VS-RL_A.fq.gz

cp -r * /proj/uppstore2017151/analyses/cyclotides
```
Runs about 7 hours. Produces trimmomatic_interleaved.VS-RL_A.fq.gz.keep

To split them, use one liner below
```{bash, split-interleaved, eval=F}
paste - - - - - - - - < trimmomatic_interleaved.VS-RL_A.fq.gz.keep | tee >(cut -f 1-4 | tr "\t" "\n" > trimmomatic_khmer.VS-RL_A_1.fq) | cut -f 5-8 | tr "\t" "\n" > trimmomatic_khmer.VS-RL_A_2.fq
```

Run transrate
```{bash transrate, eval=F}
#!/bin/bash
#SBATCH -A snic2017-7-396 
#SBATCH -p core
#SBATCH -n 10
#SBATCH -t 23:50:00
#SBATCH --mail-user karin.steffen@ilk.uu.se
#SBATCH --mail-type=ALL
#SBATCH -J transrate_cyclotides

module load bioinfo-tools transrate/1.0.1

cp /proj/uppstore2017151/data/transcriptomes/cyclotides/VS-RL_A_Trinity.fasta $SNIC_TMP
cp /proj/uppstore2017151/analyses/cyclotides/trimmomatic_khmer.VS-RL_A_1.fq $SNIC_TMP
cp /proj/uppstore2017151/analyses/cyclotides/trimmomatic_khmer.VS-RL_A_2.fq $SNIC_TMP

cd $SNIC_TMP

transrate --assembly VS-RL_A_Trinity.fasta --left trimmomatic_khmer.VS-RL_A_1.fq --right trimmomatic_khmer.VS-RL_A_2.fq --threads 10

rm trimmomatic_khmer.VS-RL_A_1.fq
rm trimmomatic_khmer.VS-RL_A_2.fq
rm VS-RL_A_Trinity.fasta

cp -r * /proj/uppstore2017151/analyses/cyclotides
```
Runs very fast, 15 mins.

