<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Funding disclaimer | Oceanographic setting influences the prokaryote community and metabolome in deep-sea sponges</title>
  <meta name="description" content="5 Funding disclaimer | Oceanographic setting influences the prokaryote community and metabolome in deep-sea sponges" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Funding disclaimer | Oceanographic setting influences the prokaryote community and metabolome in deep-sea sponges" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Funding disclaimer | Oceanographic setting influences the prokaryote community and metabolome in deep-sea sponges" />
  
  
  

<meta name="author" content="Karin Steffen" />


<meta name="date" content="2020-06-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="inter-omics.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />










<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });

  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);

    div.before(buttonRow);

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text('Code');
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text('Hide');
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
.row { display: flex; }
.collapse { display: none; }
.in { display:block }
.pull-right > .dropdown-menu {
    right: 0;
    left: auto;
}
.open > .dropdown-menu {
    display: block;
}
.dropdown-menu {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    display: none;
    float: left;
    min-width: 160px;
    padding: 5px 0;
    margin: 2px 0 0;
    font-size: 14px;
    text-align: left;
    list-style: none;
    background-color: #fff;
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    border: 1px solid #ccc;
    border: 1px solid rgba(0,0,0,.15);
    border-radius: 4px;
    -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
    box-shadow: 0 6px 12px rgba(0,0,0,.175);
}
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>


<script>
document.write('<div class="btn-group pull-right" style="position: absolute; top: 20%; right: 2%; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>')
</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">SUPPLEMENTARY MATERIAL</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Content</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#experimental-setup"><i class="fa fa-check"></i><b>1.1</b> Experimental setup</a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#map"><i class="fa fa-check"></i><b>1.1.1</b> Map</a></li>
<li class="chapter" data-level="1.1.2" data-path="index.html"><a href="index.html#water-masses"><i class="fa fa-check"></i><b>1.1.2</b> Water masses</a></li>
<li class="chapter" data-level="1.1.3" data-path="index.html"><a href="index.html#sponges"><i class="fa fa-check"></i><b>1.1.3</b> Sponges</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#sample-metadata-analysis"><i class="fa fa-check"></i><b>1.2</b> Sample metadata analysis</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#index_cor_vif"><i class="fa fa-check"></i><b>1.2.1</b> Correlations, visual inspection and variance inflation factors (VIF) across all samples</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#correlations-visual-inspection-and-variance-inflation-factors-vif-across-g.-barretti-samples"><i class="fa fa-check"></i><b>1.2.2</b> Correlations, visual inspection and variance inflation factors (VIF) across <em>G. barretti</em> samples</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="metabolomics.html"><a href="metabolomics.html"><i class="fa fa-check"></i><b>2</b> Metabolomics</a><ul>
<li class="chapter" data-level="2.1" data-path="metabolomics.html"><a href="metabolomics.html#experimental-methods-and-detailed-description"><i class="fa fa-check"></i><b>2.1</b> Experimental methods and detailed description</a></li>
<li class="chapter" data-level="2.2" data-path="metabolomics.html"><a href="metabolomics.html#data-processing"><i class="fa fa-check"></i><b>2.2</b> Data processing</a><ul>
<li class="chapter" data-level="2.2.1" data-path="metabolomics.html"><a href="metabolomics.html#peak-picking-with-xcms-and-annotation-with-camera"><i class="fa fa-check"></i><b>2.2.1</b> Peak picking with XCMS and annotation with CAMERA</a></li>
<li class="chapter" data-level="2.2.2" data-path="metabolomics.html"><a href="metabolomics.html#hilic-hydrophilic-interaction-chromatography-column-with-positive-esi-electron-spray-ionisation"><i class="fa fa-check"></i><b>2.2.2</b> HILIC (Hydrophilic interaction chromatography) column with positive ESI (electron spray ionisation)</a></li>
<li class="chapter" data-level="2.2.3" data-path="metabolomics.html"><a href="metabolomics.html#hilic-chromatography-with-negative-esi"><i class="fa fa-check"></i><b>2.2.3</b> HILIC Chromatography with negative ESI</a></li>
<li class="chapter" data-level="2.2.4" data-path="metabolomics.html"><a href="metabolomics.html#rp-reversed-phase-chromatography-with-positive-esi"><i class="fa fa-check"></i><b>2.2.4</b> RP (Reversed phase) Chromatography with positive ESI</a></li>
<li class="chapter" data-level="2.2.5" data-path="metabolomics.html"><a href="metabolomics.html#rp-chromatography-with-negative-esi"><i class="fa fa-check"></i><b>2.2.5</b> RP Chromatography with negative ESI</a></li>
<li class="chapter" data-level="2.2.6" data-path="metabolomics.html"><a href="metabolomics.html#filtering-of-the-raw-data-set"><i class="fa fa-check"></i><b>2.2.6</b> Filtering of the raw data set</a></li>
<li class="chapter" data-level="2.2.7" data-path="metabolomics.html"><a href="metabolomics.html#data-quality-control"><i class="fa fa-check"></i><b>2.2.7</b> Data quality control</a></li>
<li class="chapter" data-level="2.2.8" data-path="metabolomics.html"><a href="metabolomics.html#summary-of-data-processing"><i class="fa fa-check"></i><b>2.2.8</b> Summary of data processing</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="metabolomics.html"><a href="metabolomics.html#multivariate-analyses"><i class="fa fa-check"></i><b>2.3</b> Multivariate analyses</a></li>
<li class="chapter" data-level="2.4" data-path="metabolomics.html"><a href="metabolomics.html#vips"><i class="fa fa-check"></i><b>2.4</b> VIPS</a></li>
<li class="chapter" data-level="2.5" data-path="metabolomics.html"><a href="metabolomics.html#manual-annotation-identification-and-extraction-of-metabolic-feature-signals"><i class="fa fa-check"></i><b>2.5</b> Manual annotation, identification and extraction of metabolic feature signals</a></li>
<li class="chapter" data-level="2.6" data-path="metabolomics.html"><a href="metabolomics.html#chromatorgrams"><i class="fa fa-check"></i><b>2.6</b> Chromatorgrams</a></li>
<li class="chapter" data-level="2.7" data-path="metabolomics.html"><a href="metabolomics.html#correlations"><i class="fa fa-check"></i><b>2.7</b> Correlations</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="microbiota.html"><a href="microbiota.html"><i class="fa fa-check"></i><b>3</b> Microbiota</a><ul>
<li class="chapter" data-level="3.1" data-path="microbiota.html"><a href="microbiota.html#otu-table-overview"><i class="fa fa-check"></i><b>3.1</b> OTU table overview</a></li>
<li class="chapter" data-level="3.2" data-path="microbiota.html"><a href="microbiota.html#alpha-diversity"><i class="fa fa-check"></i><b>3.2</b> Alpha diversity</a></li>
<li class="chapter" data-level="3.3" data-path="microbiota.html"><a href="microbiota.html#beta-diversity"><i class="fa fa-check"></i><b>3.3</b> Beta diversity</a></li>
<li class="chapter" data-level="3.4" data-path="microbiota.html"><a href="microbiota.html#EnvMod"><i class="fa fa-check"></i><b>3.4</b> Environmental modelling</a><ul>
<li class="chapter" data-level="3.4.1" data-path="microbiota.html"><a href="microbiota.html#constrained"><i class="fa fa-check"></i><b>3.4.1</b> Constrained modelling approach Automatic stepwise model building</a></li>
<li class="chapter" data-level="3.4.2" data-path="microbiota.html"><a href="microbiota.html#unconstrained"><i class="fa fa-check"></i><b>3.4.2</b> Unconstrained modelling approach</a></li>
<li class="chapter" data-level="3.4.3" data-path="microbiota.html"><a href="microbiota.html#Hclust"><i class="fa fa-check"></i><b>3.4.3</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="3.4.4" data-path="microbiota.html"><a href="microbiota.html#watermasses"><i class="fa fa-check"></i><b>3.4.4</b> Oceanographic setting: water masses</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="microbiota.html"><a href="microbiota.html#relative-abundances"><i class="fa fa-check"></i><b>3.5</b> Relative abundances</a><ul>
<li class="chapter" data-level="3.5.1" data-path="microbiota.html"><a href="microbiota.html#bar-plots"><i class="fa fa-check"></i><b>3.5.1</b> Bar plots</a></li>
<li class="chapter" data-level="3.5.2" data-path="microbiota.html"><a href="microbiota.html#tabular-overview"><i class="fa fa-check"></i><b>3.5.2</b> Tabular overview</a></li>
<li class="chapter" data-level="3.5.3" data-path="microbiota.html"><a href="microbiota.html#statistical-comparison-at-phylum-and-class-level"><i class="fa fa-check"></i><b>3.5.3</b> Statistical comparison at phylum and class level</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="microbiota.html"><a href="microbiota.html#depth-response-otu-perspective"><i class="fa fa-check"></i><b>3.6</b> Depth response: OTU perspective</a></li>
<li class="chapter" data-level="3.7" data-path="microbiota.html"><a href="microbiota.html#sequence-similarity"><i class="fa fa-check"></i><b>3.7</b> Sequence similarity</a></li>
<li class="chapter" data-level="3.8" data-path="microbiota.html"><a href="microbiota.html#phylogeny"><i class="fa fa-check"></i><b>3.8</b> Phylogeny</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="inter-omics.html"><a href="inter-omics.html"><i class="fa fa-check"></i><b>4</b> Inter-omics</a><ul>
<li class="chapter" data-level="4.1" data-path="inter-omics.html"><a href="inter-omics.html#mantel-test-and-procrustes-rotations"><i class="fa fa-check"></i><b>4.1</b> Mantel test and procrustes rotations</a><ul>
<li class="chapter" data-level="4.1.1" data-path="inter-omics.html"><a href="inter-omics.html#libraries-and-functions"><i class="fa fa-check"></i><b>4.1.1</b> Libraries and functions</a></li>
<li class="chapter" data-level="4.1.2" data-path="inter-omics.html"><a href="inter-omics.html#data-sets"><i class="fa fa-check"></i><b>4.1.2</b> Data sets</a></li>
<li class="chapter" data-level="4.1.3" data-path="inter-omics.html"><a href="inter-omics.html#mantel-test-code-and-tabular-output"><i class="fa fa-check"></i><b>4.1.3</b> Mantel test code and tabular output</a></li>
<li class="chapter" data-level="4.1.4" data-path="inter-omics.html"><a href="inter-omics.html#procrustes-rotation-and-protest-code-and-tabular-output"><i class="fa fa-check"></i><b>4.1.4</b> Procrustes rotation and protest code and tabular output</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="inter-omics.html"><a href="inter-omics.html#microbial-interaction-network"><i class="fa fa-check"></i><b>4.2</b> Microbial interaction network</a><ul>
<li class="chapter" data-level="4.2.1" data-path="inter-omics.html"><a href="inter-omics.html#generating-network-based-on-different-algorithms"><i class="fa fa-check"></i><b>4.2.1</b> Generating network based on different algorithms</a></li>
<li class="chapter" data-level="4.2.2" data-path="inter-omics.html"><a href="inter-omics.html#consolidation-of-the-different-networks"><i class="fa fa-check"></i><b>4.2.2</b> Consolidation of the different networks</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="inter-omics.html"><a href="inter-omics.html#shortlist"><i class="fa fa-check"></i><b>4.3</b> Shortlist</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="funding-disclaimer.html"><a href="funding-disclaimer.html"><i class="fa fa-check"></i><b>5</b> Funding disclaimer</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Oceanographic setting influences the prokaryote community and metabolome in deep-sea sponges</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="funding-disclaimer" class="section level1">
<h1><span class="header-section-number">5</span> Funding disclaimer</h1>
<p>This study was financially supported by the SponGES project from the European Union’s Horizon 2020 research and innovation programme under grant agreement No 679849. This document reflects only the authors’ view and the Executive Agency for Small and Medium-sized Enterprises (EASME) is not responsible for any use that may be made of the information it contains.<br />
For more information, please visit the <a href="www.deepseasponges.org">SponGES</a> website.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="funding-disclaimer.html#cb1-1"></a><span class="co"># automatically create a bib database for R packages</span></span>
<span id="cb1-2"><a href="funding-disclaimer.html#cb1-2"></a>knitr<span class="op">::</span><span class="kw">write_bib</span>(<span class="kw">c</span>(</span>
<span id="cb1-3"><a href="funding-disclaimer.html#cb1-3"></a>  <span class="kw">.packages</span>(), <span class="st">&#39;bookdown&#39;</span>, <span class="st">&#39;knitr&#39;</span>, <span class="st">&#39;rmarkdown&#39;</span>, <span class="st">&#39;kableExtra&#39;</span>, <span class="st">&#39;ggplot2&#39;</span>, <span class="st">&#39;ggmap&#39;</span>, <span class="st">&#39;maps&#39;</span>, <span class="st">&#39;mapdata&#39;</span>, <span class="st">&#39;marmap&#39;</span>, <span class="st">&#39;ggrepel&#39;</span>, <span class="st">&#39;lattice&#39;</span>, <span class="st">&#39;sf&#39;</span>, <span class="st">&#39;rnaturalearth&#39;</span>, <span class="st">&#39;rnaturalearthdata&#39;</span>, <span class="st">&#39;vegan&#39;</span>, <span class="st">&#39;xcms&#39;</span>,<span class="st">&#39;camera&#39;</span>, <span class="st">&#39;ropls&#39;</span>, <span class="st">&#39;tidyverse&#39;</span>, <span class="st">&#39;stringr&#39;</span>, <span class="st">&#39;usdm&#39;</span>, <span class="st">&#39;reshape2&#39;</span>, <span class="st">&#39;gridExtra&#39;</span>, <span class="st">&#39;dplyr&#39;</span>, <span class="st">&#39;plotly&#39;</span>, <span class="st">&#39;phyloseq&#39;</span>, <span class="st">&#39;seqinr&#39;</span>, <span class="st">&#39;igraph&#39;</span></span>
<span id="cb1-4"><a href="funding-disclaimer.html#cb1-4"></a>), <span class="st">&#39;packages.bib&#39;</span>)</span></code></pre></div>
<pre><code>## Warning in knitr::write_bib(c(.packages(), &quot;bookdown&quot;, &quot;knitr&quot;, &quot;rmarkdown&quot;, :
## package(s) camera not found</code></pre>
<pre><code>## Warning in citation(pkg, auto = if (pkg == &quot;base&quot;) NULL else TRUE): no date
## field in DESCRIPTION file of package &#39;ggmap&#39;</code></pre>
<pre><code>## Warning in citation(pkg, auto = if (pkg == &quot;base&quot;) NULL else TRUE): no date
## field in DESCRIPTION file of package &#39;kableExtra&#39;</code></pre>
<pre><code>## tweaking maps</code></pre>
<p>Maps were made with the pacakges marmap <span class="citation">(Pante and Simon-Bouhet <a href="#ref-marmap2013" role="doc-biblioref">2013</a>)</span>, ggmap <span class="citation">(Kahle and Wickham <a href="#ref-ggmap2013" role="doc-biblioref">2013</a>)</span>, maps <span class="citation">(Brownrigg <a href="#ref-R-maps" role="doc-biblioref">2018</a>)</span>, mapdata <span class="citation">(Richard A. Becker and Ray Brownrigg. <a href="#ref-R-mapdata" role="doc-biblioref">2018</a>)</span>, sf <span class="citation">(Pebesma <a href="#ref-sf2018" role="doc-biblioref">2018</a>)</span>, rnaturalearth <span class="citation">(South <a href="#ref-R-rnaturalearth" role="doc-biblioref">2017</a><a href="#ref-R-rnaturalearth" role="doc-biblioref">b</a>)</span>, rnaturalearthdata <span class="citation">(South <a href="#ref-R-rnaturalearthdata" role="doc-biblioref">2017</a><a href="#ref-R-rnaturalearthdata" role="doc-biblioref">a</a>)</span>, lattice <span class="citation">(Sarkar <a href="#ref-lattice2008" role="doc-biblioref">2008</a>)</span>.
Analyses used the functionalities from R <span class="citation">(R Core Team <a href="#ref-R-base" role="doc-biblioref">2018</a>)</span> as well as the packages usdm <span class="citation">(Naimi et al. <a href="#ref-usdm2014" role="doc-biblioref">2014</a>)</span>, tidyverse <span class="citation">(Wickham et al. <a href="#ref-tidyverse2019" role="doc-biblioref">2019</a>)</span>, plotly <span class="citation">(Sievert et al. <a href="#ref-R-plotly" role="doc-biblioref">2020</a>)</span>, dplyr <span class="citation">(Wickham et al. <a href="#ref-R-dplyr" role="doc-biblioref">2020</a>)</span>, stringr <span class="citation">(Wickham <a href="#ref-R-stringr" role="doc-biblioref">2019</a>)</span>, kableExtra <span class="citation">(Zhu <a href="#ref-R-kableExtra" role="doc-biblioref">2019</a>)</span>, GridExtra <span class="citation">(Auguie <a href="#ref-R-gridExtra" role="doc-biblioref">2017</a>)</span>, ggrepel <span class="citation">(Slowikowski <a href="#ref-R-ggrepel" role="doc-biblioref">2020</a>)</span>, reshape2 <span class="citation">(Wickham <a href="#ref-R-reshape2" role="doc-biblioref">2017</a>)</span>, knitr <span class="citation">(Xie <a href="#ref-knitr2015" role="doc-biblioref">2015</a>)</span>, Rmarkadown <span class="citation">(Xie, Allaire, and Grolemund <a href="#ref-rmarkdown2018" role="doc-biblioref">2018</a>)</span> and bookdown <span class="citation">(Xie <a href="#ref-R-bookdown" role="doc-biblioref">2020</a>)</span>. Specifically, metabolomics relied on XCMS <span class="citation">(Smith et al. <a href="#ref-xcms2006" role="doc-biblioref">2006</a>)</span>, <span class="citation">(Tautenhahn, Boettcher, and Neumann <a href="#ref-xcms2008" role="doc-biblioref">2008</a>)</span>, <span class="citation">(Benton, Want, and Ebbels <a href="#ref-xcms2010" role="doc-biblioref">2010</a>)</span> with CAMERA <span class="citation">(Kuhl et al. <a href="#ref-camera2012" role="doc-biblioref">2012</a>)</span> ROPLS <span class="citation">(Thevenot et al. <a href="#ref-ropls2015" role="doc-biblioref">2015</a>)</span>, and the microbiome analyses on vegan <span class="citation">(Oksanen et al. <a href="#ref-R-vegan" role="doc-biblioref">2019</a>)</span>, phyloseq <span class="citation">(McMurdie and Holmes <a href="#ref-phyloseq2013" role="doc-biblioref">2013</a>)</span>, seqinr <span class="citation">(Charif and Lobry <a href="#ref-R-seqinr" role="doc-biblioref">2019</a>)</span>, and igraph <span class="citation">(Csardi and Nepusz <a href="#ref-igraph2006" role="doc-biblioref">2006</a>)</span>.</p>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-R-gridExtra">
<p>Auguie, Baptiste. 2017. <em>GridExtra: Miscellaneous Functions for "Grid" Graphics</em>. <a href="https://CRAN.R-project.org/package=gridExtra">https://CRAN.R-project.org/package=gridExtra</a>.</p>
</div>
<div id="ref-xcms2010">
<p>Benton, H. Paul, Elizabeth J. Want, and Timothy M. D. Ebbels. 2010. “Correction of Mass Calibration Gaps in Liquid Chromatography-Mass Spectrometry Metabolomics Data.” <em>BIOINFORMATICS</em> 26: 2488.</p>
</div>
<div id="ref-R-maps">
<p>Brownrigg, Ray. 2018. <em>Maps: Draw Geographical Maps</em>. <a href="https://CRAN.R-project.org/package=maps">https://CRAN.R-project.org/package=maps</a>.</p>
</div>
<div id="ref-R-seqinr">
<p>Charif, Delphine, and Jean R. Lobry. 2019. <em>Seqinr: Biological Sequences Retrieval and Analysis</em>. <a href="https://CRAN.R-project.org/package=seqinr">https://CRAN.R-project.org/package=seqinr</a>.</p>
</div>
<div id="ref-igraph2006">
<p>Csardi, Gabor, and Tamas Nepusz. 2006. “The Igraph Software Package for Complex Network Research.” <em>InterJournal</em> Complex Systems: 1695. <a href="http://igraph.org">http://igraph.org</a>.</p>
</div>
<div id="ref-ggmap2013">
<p>Kahle, David, and Hadley Wickham. 2013. “Ggmap: Spatial Visualization with Ggplot2.” <em>The R Journal</em> 5 (1): 144–61. <a href="https://journal.r-project.org/archive/2013-1/kahle-wickham.pdf">https://journal.r-project.org/archive/2013-1/kahle-wickham.pdf</a>.</p>
</div>
<div id="ref-camera2012">
<p>Kuhl, C., R. Tautenhahn, C. Boettcher, T. R. Larson, and S. Neumann. 2012. “CAMERA: An Integrated Strategy for Compound Spectra Extraction and Annotation of Liquid Chromatography/Mass Spectrometry Data Sets.” <em>Analytical Chemistry</em> 84: 283–89. <a href="http://pubs.acs.org/doi/abs/10.1021/ac202450g">http://pubs.acs.org/doi/abs/10.1021/ac202450g</a>.</p>
</div>
<div id="ref-phyloseq2013">
<p>McMurdie, Paul J., and Susan Holmes. 2013. “Phyloseq: An R Package for Reproducible Interactive Analysis and Graphics of Microbiome Census Data.” <em>PLoS ONE</em> 8 (4): e61217. <a href="http://dx.plos.org/10.1371/journal.pone.0061217">http://dx.plos.org/10.1371/journal.pone.0061217</a>.</p>
</div>
<div id="ref-usdm2014">
<p>Naimi, Babak, Nicholas a.s. Hamm, Thomas A. Groen, Andrew K. Skidmore, and Albertus G. Toxopeus. 2014. “Where Is Positional Uncertainty a Problem for Species Distribution Modelling.” <em>Ecography</em> 37: 191–203. <a href="https://doi.org/10.1111/j.1600-0587.2013.00205.x">https://doi.org/10.1111/j.1600-0587.2013.00205.x</a>.</p>
</div>
<div id="ref-R-vegan">
<p>Oksanen, Jari, F. Guillaume Blanchet, Michael Friendly, Roeland Kindt, Pierre Legendre, Dan McGlinn, Peter R. Minchin, et al. 2019. <em>Vegan: Community Ecology Package</em>. <a href="https://CRAN.R-project.org/package=vegan">https://CRAN.R-project.org/package=vegan</a>.</p>
</div>
<div id="ref-marmap2013">
<p>Pante, Eric, and Benoit Simon-Bouhet. 2013. “Marmap: A Package for Importing, Plotting and Analyzing Bathymetric and Topographic Data in R.” <em>PLoS ONE</em> 8 (9): e73051.</p>
</div>
<div id="ref-sf2018">
<p>Pebesma, Edzer. 2018. “Simple Features for R: Standardized Support for Spatial Vector Data.” <em>The R Journal</em> 10 (1): 439–46. <a href="https://doi.org/10.32614/RJ-2018-009">https://doi.org/10.32614/RJ-2018-009</a>.</p>
</div>
<div id="ref-R-base">
<p>R Core Team. 2018. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.</p>
</div>
<div id="ref-R-mapdata">
<p>Richard A. Becker, Original S code by, and Allan R. Wilks. R version by Ray Brownrigg. 2018. <em>Mapdata: Extra Map Databases</em>. <a href="https://CRAN.R-project.org/package=mapdata">https://CRAN.R-project.org/package=mapdata</a>.</p>
</div>
<div id="ref-lattice2008">
<p>Sarkar, Deepayan. 2008. <em>Lattice: Multivariate Data Visualization with R</em>. New York: Springer. <a href="http://lmdvr.r-forge.r-project.org">http://lmdvr.r-forge.r-project.org</a>.</p>
</div>
<div id="ref-R-plotly">
<p>Sievert, Carson, Chris Parmer, Toby Hocking, Scott Chamberlain, Karthik Ram, Marianne Corvellec, and Pedro Despouy. 2020. <em>Plotly: Create Interactive Web Graphics via ’Plotly.js’</em>. <a href="https://CRAN.R-project.org/package=plotly">https://CRAN.R-project.org/package=plotly</a>.</p>
</div>
<div id="ref-R-ggrepel">
<p>Slowikowski, Kamil. 2020. <em>Ggrepel: Automatically Position Non-Overlapping Text Labels with ’Ggplot2’</em>. <a href="https://CRAN.R-project.org/package=ggrepel">https://CRAN.R-project.org/package=ggrepel</a>.</p>
</div>
<div id="ref-xcms2006">
<p>Smith, C.A., Want, E.J., O’Maille, G., Abagyan,R., Siuzdak, and G. 2006. “XCMS: Processing Mass Spectrometry Data for Metabolite Profiling Using Nonlinear Peak Alignment, Matching and Identification.” <em>Analytical Chemistry</em> 78: 779–87.</p>
</div>
<div id="ref-R-rnaturalearthdata">
<p>South, Andy. 2017a. <em>Rnaturalearthdata: World Vector Map Data from Natural Earth Used in ’Rnaturalearth’</em>. <a href="https://CRAN.R-project.org/package=rnaturalearthdata">https://CRAN.R-project.org/package=rnaturalearthdata</a>.</p>
</div>
<div id="ref-R-rnaturalearth">
<p>South, Andy. 2017b. <em>Rnaturalearth: World Map Data from Natural Earth</em>. <a href="https://CRAN.R-project.org/package=rnaturalearth">https://CRAN.R-project.org/package=rnaturalearth</a>.</p>
</div>
<div id="ref-xcms2008">
<p>Tautenhahn, Ralf, Christoph Boettcher, and Steffen Neumann. 2008. “Highly Sensitive Feature Detection for High Resolution Lc/Ms.” <em>BMC Bioinformatics</em> 9: 504.</p>
</div>
<div id="ref-ropls2015">
<p>Thevenot, E. A., A. Roux, Y. Xu, E. Ezan, and C. Junot. 2015. “Analysis of the Human Adult Urinary Metabolome Variations with Age, Body Mass Index and Gender by Implementing a Comprehensive Workflow for Univariate and Opls Statistical Analyses.” <em>Journal of Proteome Research</em> 14: 3322–35. <a href="http://pubs.acs.org/doi/full/10.1021/acs.jproteome.5b00354">http://pubs.acs.org/doi/full/10.1021/acs.jproteome.5b00354</a>.</p>
</div>
<div id="ref-R-reshape2">
<p>Wickham, Hadley. 2017. <em>Reshape2: Flexibly Reshape Data: A Reboot of the Reshape Package</em>. <a href="https://CRAN.R-project.org/package=reshape2">https://CRAN.R-project.org/package=reshape2</a>.</p>
</div>
<div id="ref-R-stringr">
<p>Wickham, Hadley. 2019. <em>Stringr: Simple, Consistent Wrappers for Common String Operations</em>. <a href="https://CRAN.R-project.org/package=stringr">https://CRAN.R-project.org/package=stringr</a>.</p>
</div>
<div id="ref-tidyverse2019">
<p>Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. “Welcome to the tidyverse.” <em>Journal of Open Source Software</em> 4 (43): 1686. <a href="https://doi.org/10.21105/joss.01686">https://doi.org/10.21105/joss.01686</a>.</p>
</div>
<div id="ref-R-dplyr">
<p>Wickham, Hadley, Romain François, Lionel Henry, and Kirill Müller. 2020. <em>Dplyr: A Grammar of Data Manipulation</em>. <a href="https://CRAN.R-project.org/package=dplyr">https://CRAN.R-project.org/package=dplyr</a>.</p>
</div>
<div id="ref-knitr2015">
<p>Xie, Yihui. 2015. <em>Dynamic Documents with R and Knitr</em>. 2nd ed. Boca Raton, Florida: Chapman; Hall/CRC. <a href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>.</p>
</div>
<div id="ref-R-bookdown">
<p>Xie, Yihui. 2020. <em>Bookdown: Authoring Books and Technical Documents with R Markdown</em>. <a href="https://CRAN.R-project.org/package=bookdown">https://CRAN.R-project.org/package=bookdown</a>.</p>
</div>
<div id="ref-rmarkdown2018">
<p>Xie, Yihui, J. J. Allaire, and Garrett Grolemund. 2018. <em>R Markdown: The Definitive Guide</em>. Boca Raton, Florida: Chapman; Hall/CRC. <a href="https://bookdown.org/yihui/rmarkdown">https://bookdown.org/yihui/rmarkdown</a>.</p>
</div>
<div id="ref-R-kableExtra">
<p>Zhu, Hao. 2019. <em>KableExtra: Construct Complex Table with ’Kable’ and Pipe Syntax</em>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="inter-omics.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Oceanographic-setting-influences-the-prokaryote-community-and-metabolome-in-deep-sea-sponges.pdf", "Oceanographic-setting-influences-the-prokaryote-community-and-metabolome-in-deep-sea-sponges.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
