<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Inter-omics | Oceanographic setting influences the prokaryote community and metabolome in deep-sea sponges</title>
  <meta name="description" content="4 Inter-omics | Oceanographic setting influences the prokaryote community and metabolome in deep-sea sponges" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Inter-omics | Oceanographic setting influences the prokaryote community and metabolome in deep-sea sponges" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Inter-omics | Oceanographic setting influences the prokaryote community and metabolome in deep-sea sponges" />
  
  
  

<meta name="author" content="Karin Steffen" />


<meta name="date" content="2020-06-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="microbiota.html"/>
<link rel="next" href="funding-disclaimer.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />

<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });

  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);

    div.before(buttonRow);

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text('Code');
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text('Hide');
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
.row { display: flex; }
.collapse { display: none; }
.in { display:block }
.pull-right > .dropdown-menu {
    right: 0;
    left: auto;
}
.open > .dropdown-menu {
    display: block;
}
.dropdown-menu {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    display: none;
    float: left;
    min-width: 160px;
    padding: 5px 0;
    margin: 2px 0 0;
    font-size: 14px;
    text-align: left;
    list-style: none;
    background-color: #fff;
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    border: 1px solid #ccc;
    border: 1px solid rgba(0,0,0,.15);
    border-radius: 4px;
    -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
    box-shadow: 0 6px 12px rgba(0,0,0,.175);
}
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>


<script>
document.write('<div class="btn-group pull-right" style="position: absolute; top: 20%; right: 2%; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>')
</script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">SUPPLEMENTARY MATERIAL</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Content</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#experimental-setup"><i class="fa fa-check"></i><b>1.1</b> Experimental setup</a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#map"><i class="fa fa-check"></i><b>1.1.1</b> Map</a></li>
<li class="chapter" data-level="1.1.2" data-path="index.html"><a href="index.html#water-masses"><i class="fa fa-check"></i><b>1.1.2</b> Water masses</a></li>
<li class="chapter" data-level="1.1.3" data-path="index.html"><a href="index.html#sponges"><i class="fa fa-check"></i><b>1.1.3</b> Sponges</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#sample-metadata-analysis"><i class="fa fa-check"></i><b>1.2</b> Sample metadata analysis</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#index_cor_vif"><i class="fa fa-check"></i><b>1.2.1</b> Correlations, visual inspection and variance inflation factors (VIF) across all samples</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#correlations-visual-inspection-and-variance-inflation-factors-vif-across-g.-barretti-samples"><i class="fa fa-check"></i><b>1.2.2</b> Correlations, visual inspection and variance inflation factors (VIF) across <em>G. barretti</em> samples</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="metabolomics.html"><a href="metabolomics.html"><i class="fa fa-check"></i><b>2</b> Metabolomics</a><ul>
<li class="chapter" data-level="2.1" data-path="metabolomics.html"><a href="metabolomics.html#experimental-methods-and-detailed-description"><i class="fa fa-check"></i><b>2.1</b> Experimental methods and detailed description</a></li>
<li class="chapter" data-level="2.2" data-path="metabolomics.html"><a href="metabolomics.html#data-processing"><i class="fa fa-check"></i><b>2.2</b> Data processing</a><ul>
<li class="chapter" data-level="2.2.1" data-path="metabolomics.html"><a href="metabolomics.html#peak-picking-with-xcms-and-annotation-with-camera"><i class="fa fa-check"></i><b>2.2.1</b> Peak picking with XCMS and annotation with CAMERA</a></li>
<li class="chapter" data-level="2.2.2" data-path="metabolomics.html"><a href="metabolomics.html#hilic-hydrophilic-interaction-chromatography-column-with-positive-esi-electron-spray-ionisation"><i class="fa fa-check"></i><b>2.2.2</b> HILIC (Hydrophilic interaction chromatography) column with positive ESI (electron spray ionisation)</a></li>
<li class="chapter" data-level="2.2.3" data-path="metabolomics.html"><a href="metabolomics.html#hilic-chromatography-with-negative-esi"><i class="fa fa-check"></i><b>2.2.3</b> HILIC Chromatography with negative ESI</a></li>
<li class="chapter" data-level="2.2.4" data-path="metabolomics.html"><a href="metabolomics.html#rp-reversed-phase-chromatography-with-positive-esi"><i class="fa fa-check"></i><b>2.2.4</b> RP (Reversed phase) Chromatography with positive ESI</a></li>
<li class="chapter" data-level="2.2.5" data-path="metabolomics.html"><a href="metabolomics.html#rp-chromatography-with-negative-esi"><i class="fa fa-check"></i><b>2.2.5</b> RP Chromatography with negative ESI</a></li>
<li class="chapter" data-level="2.2.6" data-path="metabolomics.html"><a href="metabolomics.html#filtering-of-the-raw-data-set"><i class="fa fa-check"></i><b>2.2.6</b> Filtering of the raw data set</a></li>
<li class="chapter" data-level="2.2.7" data-path="metabolomics.html"><a href="metabolomics.html#data-quality-control"><i class="fa fa-check"></i><b>2.2.7</b> Data quality control</a></li>
<li class="chapter" data-level="2.2.8" data-path="metabolomics.html"><a href="metabolomics.html#summary-of-data-processing"><i class="fa fa-check"></i><b>2.2.8</b> Summary of data processing</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="metabolomics.html"><a href="metabolomics.html#multivariate-analyses"><i class="fa fa-check"></i><b>2.3</b> Multivariate analyses</a></li>
<li class="chapter" data-level="2.4" data-path="metabolomics.html"><a href="metabolomics.html#vips"><i class="fa fa-check"></i><b>2.4</b> VIPS</a></li>
<li class="chapter" data-level="2.5" data-path="metabolomics.html"><a href="metabolomics.html#manual-annotation-identification-and-extraction-of-metabolic-feature-signals"><i class="fa fa-check"></i><b>2.5</b> Manual annotation, identification and extraction of metabolic feature signals</a></li>
<li class="chapter" data-level="2.6" data-path="metabolomics.html"><a href="metabolomics.html#chromatorgrams"><i class="fa fa-check"></i><b>2.6</b> Chromatorgrams</a></li>
<li class="chapter" data-level="2.7" data-path="metabolomics.html"><a href="metabolomics.html#correlations"><i class="fa fa-check"></i><b>2.7</b> Correlations</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="microbiota.html"><a href="microbiota.html"><i class="fa fa-check"></i><b>3</b> Microbiota</a><ul>
<li class="chapter" data-level="3.1" data-path="microbiota.html"><a href="microbiota.html#otu-table-overview"><i class="fa fa-check"></i><b>3.1</b> OTU table overview</a></li>
<li class="chapter" data-level="3.2" data-path="microbiota.html"><a href="microbiota.html#alpha-diversity"><i class="fa fa-check"></i><b>3.2</b> Alpha diversity</a></li>
<li class="chapter" data-level="3.3" data-path="microbiota.html"><a href="microbiota.html#beta-diversity"><i class="fa fa-check"></i><b>3.3</b> Beta diversity</a></li>
<li class="chapter" data-level="3.4" data-path="microbiota.html"><a href="microbiota.html#EnvMod"><i class="fa fa-check"></i><b>3.4</b> Environmental modelling</a><ul>
<li class="chapter" data-level="3.4.1" data-path="microbiota.html"><a href="microbiota.html#constrained"><i class="fa fa-check"></i><b>3.4.1</b> Constrained modelling approach Automatic stepwise model building</a></li>
<li class="chapter" data-level="3.4.2" data-path="microbiota.html"><a href="microbiota.html#unconstrained"><i class="fa fa-check"></i><b>3.4.2</b> Unconstrained modelling approach</a></li>
<li class="chapter" data-level="3.4.3" data-path="microbiota.html"><a href="microbiota.html#Hclust"><i class="fa fa-check"></i><b>3.4.3</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="3.4.4" data-path="microbiota.html"><a href="microbiota.html#watermasses"><i class="fa fa-check"></i><b>3.4.4</b> Oceanographic setting: water masses</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="microbiota.html"><a href="microbiota.html#relative-abundances"><i class="fa fa-check"></i><b>3.5</b> Relative abundances</a><ul>
<li class="chapter" data-level="3.5.1" data-path="microbiota.html"><a href="microbiota.html#bar-plots"><i class="fa fa-check"></i><b>3.5.1</b> Bar plots</a></li>
<li class="chapter" data-level="3.5.2" data-path="microbiota.html"><a href="microbiota.html#tabular-overview"><i class="fa fa-check"></i><b>3.5.2</b> Tabular overview</a></li>
<li class="chapter" data-level="3.5.3" data-path="microbiota.html"><a href="microbiota.html#statistical-comparison-at-phylum-and-class-level"><i class="fa fa-check"></i><b>3.5.3</b> Statistical comparison at phylum and class level</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="microbiota.html"><a href="microbiota.html#depth-response-otu-perspective"><i class="fa fa-check"></i><b>3.6</b> Depth response: OTU perspective</a></li>
<li class="chapter" data-level="3.7" data-path="microbiota.html"><a href="microbiota.html#sequence-similarity"><i class="fa fa-check"></i><b>3.7</b> Sequence similarity</a></li>
<li class="chapter" data-level="3.8" data-path="microbiota.html"><a href="microbiota.html#annotated-phylogeny"><i class="fa fa-check"></i><b>3.8</b> Annotated phylogeny</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="inter-omics.html"><a href="inter-omics.html"><i class="fa fa-check"></i><b>4</b> Inter-omics</a><ul>
<li class="chapter" data-level="4.1" data-path="inter-omics.html"><a href="inter-omics.html#mantel-test-and-procrustes-rotations"><i class="fa fa-check"></i><b>4.1</b> Mantel test and procrustes rotations</a><ul>
<li class="chapter" data-level="4.1.1" data-path="inter-omics.html"><a href="inter-omics.html#libraries-and-functions"><i class="fa fa-check"></i><b>4.1.1</b> Libraries and functions</a></li>
<li class="chapter" data-level="4.1.2" data-path="inter-omics.html"><a href="inter-omics.html#data-sets"><i class="fa fa-check"></i><b>4.1.2</b> Data sets</a></li>
<li class="chapter" data-level="4.1.3" data-path="inter-omics.html"><a href="inter-omics.html#mantel-test-code-and-tabular-output"><i class="fa fa-check"></i><b>4.1.3</b> Mantel test code and tabular output</a></li>
<li class="chapter" data-level="4.1.4" data-path="inter-omics.html"><a href="inter-omics.html#procrustes-rotation-and-protest-code-and-tabular-output"><i class="fa fa-check"></i><b>4.1.4</b> Procrustes rotation and protest code and tabular output</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="inter-omics.html"><a href="inter-omics.html#correlating-barettin-and-prokaryotic-relative-abundance"><i class="fa fa-check"></i><b>4.2</b> Correlating barettin and prokaryotic relative abundance</a></li>
<li class="chapter" data-level="4.3" data-path="inter-omics.html"><a href="inter-omics.html#microbial-interaction-network"><i class="fa fa-check"></i><b>4.3</b> Microbial interaction network</a><ul>
<li class="chapter" data-level="4.3.1" data-path="inter-omics.html"><a href="inter-omics.html#generating-network-based-on-different-algorithms"><i class="fa fa-check"></i><b>4.3.1</b> Generating network based on different algorithms</a></li>
<li class="chapter" data-level="4.3.2" data-path="inter-omics.html"><a href="inter-omics.html#consolidation-of-the-different-networks"><i class="fa fa-check"></i><b>4.3.2</b> Consolidation of the different networks</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="inter-omics.html"><a href="inter-omics.html#shortlist"><i class="fa fa-check"></i><b>4.4</b> Shortlist</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="funding-disclaimer.html"><a href="funding-disclaimer.html"><i class="fa fa-check"></i><b>5</b> Funding disclaimer</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Oceanographic setting influences the prokaryote community and metabolome in deep-sea sponges</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="inter-omics" class="section level1">
<h1><span class="header-section-number">4</span> Inter-omics</h1>
<p>The inter-omics analyses are comprised of three parts. In the first part, we evaluate congruency of the prokaryotic and metabolomic data set as a whole using numerical methods (Mantel test) and ordination (Procrustes rotation and Protest). The second part consists of generating a microbial interaction network and annotating it with depth response of the OTUs and correlation with the barettin signal. In the third part, we rank OTUs based on properties hypothesised to be true forthe producer of barettin.</p>
<div id="mantel-test-and-procrustes-rotations" class="section level2">
<h2><span class="header-section-number">4.1</span> Mantel test and procrustes rotations</h2>
<div id="libraries-and-functions" class="section level3">
<h3><span class="header-section-number">4.1.1</span> Libraries and functions</h3>
</div>
<div id="data-sets" class="section level3">
<h3><span class="header-section-number">4.1.2</span> Data sets</h3>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="inter-omics.html#cb2-1"></a><span class="co"># LOAD DATA</span></span>
<span id="cb2-2"><a href="inter-omics.html#cb2-2"></a>micro &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/OTU_all_R.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb2-3"><a href="inter-omics.html#cb2-3"></a><span class="kw">colnames</span>(micro)[<span class="kw">colnames</span>(micro) <span class="op">==</span><span class="st"> &quot;Sample_ID&quot;</span>] &lt;-<span class="st"> &quot;unified_ID&quot;</span></span>
<span id="cb2-4"><a href="inter-omics.html#cb2-4"></a>micro &lt;-<span class="st"> </span>micro[<span class="kw">order</span>(micro<span class="op">$</span>unified_ID), ]</span>
<span id="cb2-5"><a href="inter-omics.html#cb2-5"></a><span class="kw">rownames</span>(micro) &lt;-<span class="st"> </span>micro<span class="op">$</span>unified_ID</span>
<span id="cb2-6"><a href="inter-omics.html#cb2-6"></a>micro[<span class="st">&quot;spec&quot;</span>] &lt;-<span class="st"> </span><span class="kw">str_sub</span>(micro<span class="op">$</span>unified_ID, <span class="dv">1</span>, <span class="dv">2</span>)</span>
<span id="cb2-7"><a href="inter-omics.html#cb2-7"></a></span>
<span id="cb2-8"><a href="inter-omics.html#cb2-8"></a>meta_data &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/PANGAEA_Final.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb2-9"><a href="inter-omics.html#cb2-9"></a>masslynx &lt;-<span class="st"> </span>meta_data</span>
<span id="cb2-10"><a href="inter-omics.html#cb2-10"></a>masslynx &lt;-<span class="st"> </span>masslynx[<span class="kw">c</span>(<span class="st">&quot;unified_ID&quot;</span>, <span class="st">&quot;LC.MS.HILIC.positive&quot;</span>, <span class="st">&quot;LC.MS.HILIC.negative&quot;</span>, </span>
<span id="cb2-11"><a href="inter-omics.html#cb2-11"></a>    <span class="st">&quot;LC.MS.RP.positive&quot;</span>, <span class="st">&quot;LC.MS.RP.negative&quot;</span>)]</span>
<span id="cb2-12"><a href="inter-omics.html#cb2-12"></a><span class="kw">colnames</span>(masslynx) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;unified_ID&quot;</span>, <span class="st">&quot;H_p&quot;</span>, <span class="st">&quot;H_n&quot;</span>, <span class="st">&quot;R_p&quot;</span>, <span class="st">&quot;R_n&quot;</span>)</span>
<span id="cb2-13"><a href="inter-omics.html#cb2-13"></a>masslynx &lt;-<span class="st"> </span><span class="kw">na.omit</span>(masslynx)</span>
<span id="cb2-14"><a href="inter-omics.html#cb2-14"></a>masslynx[<span class="st">&quot;HILIC_pos&quot;</span>] &lt;-<span class="st"> </span><span class="kw">str_sub</span>(masslynx<span class="op">$</span>H_p, <span class="dv">1</span>, <span class="dv">-3</span>)</span>
<span id="cb2-15"><a href="inter-omics.html#cb2-15"></a>masslynx[<span class="st">&quot;HILIC_neg&quot;</span>] &lt;-<span class="st"> </span><span class="kw">str_sub</span>(masslynx<span class="op">$</span>H_n, <span class="dv">1</span>, <span class="dv">-3</span>)</span>
<span id="cb2-16"><a href="inter-omics.html#cb2-16"></a>masslynx[<span class="st">&quot;RP_pos&quot;</span>] &lt;-<span class="st"> </span><span class="kw">str_sub</span>(masslynx<span class="op">$</span>R_p, <span class="dv">1</span>, <span class="dv">-3</span>)</span>
<span id="cb2-17"><a href="inter-omics.html#cb2-17"></a>masslynx[<span class="st">&quot;RP_neg&quot;</span>] &lt;-<span class="st"> </span><span class="kw">str_sub</span>(masslynx<span class="op">$</span>R_n, <span class="dv">1</span>, <span class="dv">-3</span>)</span>
<span id="cb2-18"><a href="inter-omics.html#cb2-18"></a></span>
<span id="cb2-19"><a href="inter-omics.html#cb2-19"></a><span class="co"># load one set of experiments at a time, i.e. CLEANED, ION or PC_GROUPS</span></span>
<span id="cb2-20"><a href="inter-omics.html#cb2-20"></a></span>
<span id="cb2-21"><a href="inter-omics.html#cb2-21"></a><span class="co">### CLEANED</span></span>
<span id="cb2-22"><a href="inter-omics.html#cb2-22"></a>hilic_pos &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/HILIC_pos_20190417_cleaned.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb2-23"><a href="inter-omics.html#cb2-23"></a>hilic_neg &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/HILIC_neg_20190421_cleaned.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb2-24"><a href="inter-omics.html#cb2-24"></a>rp_pos &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/RP_pos_20190421_cleaned.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb2-25"><a href="inter-omics.html#cb2-25"></a>rp_neg &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/RP_neg_20190422_cleaned.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb2-26"><a href="inter-omics.html#cb2-26"></a></span>
<span id="cb2-27"><a href="inter-omics.html#cb2-27"></a><span class="co">### ION</span></span>
<span id="cb2-28"><a href="inter-omics.html#cb2-28"></a>hilic_pos &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/HILIC_pos_20190417_cleaned_MH.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb2-29"><a href="inter-omics.html#cb2-29"></a>hilic_neg &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/HILIC_neg_20190421_cleaned_MH.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb2-30"><a href="inter-omics.html#cb2-30"></a>rp_pos &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/RP_pos_20190421_cleaned_MH.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb2-31"><a href="inter-omics.html#cb2-31"></a>rp_neg &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/RP_neg_20190422_cleaned_MH.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb2-32"><a href="inter-omics.html#cb2-32"></a></span>
<span id="cb2-33"><a href="inter-omics.html#cb2-33"></a><span class="co">### PC_GROUPS</span></span>
<span id="cb2-34"><a href="inter-omics.html#cb2-34"></a>hilic_pos &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/HILIC_pos_20190417_cleaned_pcgroup.csv&quot;</span>, <span class="dt">header =</span> T, </span>
<span id="cb2-35"><a href="inter-omics.html#cb2-35"></a>    <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb2-36"><a href="inter-omics.html#cb2-36"></a>hilic_neg &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/HILIC_neg_20190421_cleaned_pcgroup.csv&quot;</span>, <span class="dt">header =</span> T, </span>
<span id="cb2-37"><a href="inter-omics.html#cb2-37"></a>    <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb2-38"><a href="inter-omics.html#cb2-38"></a>rp_pos &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/RP_pos_20190421_cleaned_pcgroup.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb2-39"><a href="inter-omics.html#cb2-39"></a>rp_neg &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/RP_neg_20190422_cleaned_pcgroup.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span></code></pre></div>
</div>
<div id="mantel-test-code-and-tabular-output" class="section level3">
<h3><span class="header-section-number">4.1.3</span> Mantel test code and tabular output</h3>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="inter-omics.html#cb3-1"></a><span class="co">### =============================== MANTEL </span><span class="al">TEST</span><span class="co">=================================</span></span>
<span id="cb3-2"><a href="inter-omics.html#cb3-2"></a></span>
<span id="cb3-3"><a href="inter-omics.html#cb3-3"></a><span class="co"># run one of this at a time</span></span>
<span id="cb3-4"><a href="inter-omics.html#cb3-4"></a>metabolomes &lt;-<span class="st"> </span><span class="kw">gymnastics</span>(hilic_pos, <span class="st">&quot;H_p&quot;</span>)</span>
<span id="cb3-5"><a href="inter-omics.html#cb3-5"></a>metabolomes &lt;-<span class="st"> </span><span class="kw">gymnastics</span>(hilic_neg, <span class="st">&quot;H_n&quot;</span>)</span>
<span id="cb3-6"><a href="inter-omics.html#cb3-6"></a>metabolomes &lt;-<span class="st"> </span><span class="kw">gymnastics</span>(rp_pos, <span class="st">&quot;R_p&quot;</span>)</span>
<span id="cb3-7"><a href="inter-omics.html#cb3-7"></a>metabolomes &lt;-<span class="st"> </span><span class="kw">gymnastics</span>(rp_neg, <span class="st">&quot;R_n&quot;</span>)</span>
<span id="cb3-8"><a href="inter-omics.html#cb3-8"></a></span>
<span id="cb3-9"><a href="inter-omics.html#cb3-9"></a><span class="co"># run this</span></span>
<span id="cb3-10"><a href="inter-omics.html#cb3-10"></a>congruent_dfs &lt;-<span class="st"> </span><span class="kw">congruency</span>(metabolomes)</span>
<span id="cb3-11"><a href="inter-omics.html#cb3-11"></a></span>
<span id="cb3-12"><a href="inter-omics.html#cb3-12"></a><span class="co"># run one of these: MANTEL </span><span class="al">TEST</span></span>
<span id="cb3-13"><a href="inter-omics.html#cb3-13"></a>diagnostics_hp_cleaned &lt;-<span class="st"> </span><span class="kw">cloak</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;HILIC pos&quot;</span>, </span>
<span id="cb3-14"><a href="inter-omics.html#cb3-14"></a>    <span class="dt">filtering =</span> <span class="st">&quot;cleaned&quot;</span>)</span>
<span id="cb3-15"><a href="inter-omics.html#cb3-15"></a>diagnostics_hn_cleaned &lt;-<span class="st"> </span><span class="kw">cloak</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;HILIC neg&quot;</span>, </span>
<span id="cb3-16"><a href="inter-omics.html#cb3-16"></a>    <span class="dt">filtering =</span> <span class="st">&quot;cleaned&quot;</span>)</span>
<span id="cb3-17"><a href="inter-omics.html#cb3-17"></a>diagnostics_rp_cleaned &lt;-<span class="st"> </span><span class="kw">cloak</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;RP pos&quot;</span>, </span>
<span id="cb3-18"><a href="inter-omics.html#cb3-18"></a>    <span class="dt">filtering =</span> <span class="st">&quot;cleaned&quot;</span>)</span>
<span id="cb3-19"><a href="inter-omics.html#cb3-19"></a>diagnostics_rn_cleaned &lt;-<span class="st"> </span><span class="kw">cloak</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;RP neg&quot;</span>, </span>
<span id="cb3-20"><a href="inter-omics.html#cb3-20"></a>    <span class="dt">filtering =</span> <span class="st">&quot;cleaned&quot;</span>)</span>
<span id="cb3-21"><a href="inter-omics.html#cb3-21"></a></span>
<span id="cb3-22"><a href="inter-omics.html#cb3-22"></a>diagnostics_hp_ion &lt;-<span class="st"> </span><span class="kw">cloak</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;HILIC pos&quot;</span>, </span>
<span id="cb3-23"><a href="inter-omics.html#cb3-23"></a>    <span class="dt">filtering =</span> <span class="st">&quot;ion&quot;</span>)</span>
<span id="cb3-24"><a href="inter-omics.html#cb3-24"></a>diagnostics_hn_ion &lt;-<span class="st"> </span><span class="kw">cloak</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;HILIC neg&quot;</span>, </span>
<span id="cb3-25"><a href="inter-omics.html#cb3-25"></a>    <span class="dt">filtering =</span> <span class="st">&quot;ion&quot;</span>)</span>
<span id="cb3-26"><a href="inter-omics.html#cb3-26"></a>diagnostics_rp_ion &lt;-<span class="st"> </span><span class="kw">cloak</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;RP pos&quot;</span>, </span>
<span id="cb3-27"><a href="inter-omics.html#cb3-27"></a>    <span class="dt">filtering =</span> <span class="st">&quot;ion&quot;</span>)</span>
<span id="cb3-28"><a href="inter-omics.html#cb3-28"></a>diagnostics_rn_ion &lt;-<span class="st"> </span><span class="kw">cloak</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;RP neg&quot;</span>, </span>
<span id="cb3-29"><a href="inter-omics.html#cb3-29"></a>    <span class="dt">filtering =</span> <span class="st">&quot;ion&quot;</span>)</span>
<span id="cb3-30"><a href="inter-omics.html#cb3-30"></a></span>
<span id="cb3-31"><a href="inter-omics.html#cb3-31"></a>diagnostics_hp_pc_group &lt;-<span class="st"> </span><span class="kw">cloak</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;HILIC pos&quot;</span>, </span>
<span id="cb3-32"><a href="inter-omics.html#cb3-32"></a>    <span class="dt">filtering =</span> <span class="st">&quot;pc_group&quot;</span>)</span>
<span id="cb3-33"><a href="inter-omics.html#cb3-33"></a>diagnostics_hn_pc_group &lt;-<span class="st"> </span><span class="kw">cloak</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;HILIC neg&quot;</span>, </span>
<span id="cb3-34"><a href="inter-omics.html#cb3-34"></a>    <span class="dt">filtering =</span> <span class="st">&quot;pc_group&quot;</span>)</span>
<span id="cb3-35"><a href="inter-omics.html#cb3-35"></a>diagnostics_rp_pc_group &lt;-<span class="st"> </span><span class="kw">cloak</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;RP pos&quot;</span>, </span>
<span id="cb3-36"><a href="inter-omics.html#cb3-36"></a>    <span class="dt">filtering =</span> <span class="st">&quot;pc_group&quot;</span>)</span>
<span id="cb3-37"><a href="inter-omics.html#cb3-37"></a>diagnostics_rn_pc_group &lt;-<span class="st"> </span><span class="kw">cloak</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;RP neg&quot;</span>, </span>
<span id="cb3-38"><a href="inter-omics.html#cb3-38"></a>    <span class="dt">filtering =</span> <span class="st">&quot;pc_group&quot;</span>)</span>
<span id="cb3-39"><a href="inter-omics.html#cb3-39"></a></span>
<span id="cb3-40"><a href="inter-omics.html#cb3-40"></a></span>
<span id="cb3-41"><a href="inter-omics.html#cb3-41"></a><span class="co">### Combine all test results into one file</span></span>
<span id="cb3-42"><a href="inter-omics.html#cb3-42"></a></span>
<span id="cb3-43"><a href="inter-omics.html#cb3-43"></a>diagnostics &lt;-<span class="st"> </span>diagnostics_hp_cleaned</span>
<span id="cb3-44"><a href="inter-omics.html#cb3-44"></a>diagnostics &lt;-<span class="st"> </span><span class="kw">rbind</span>(diagnostics, diagnostics_hn_cleaned, diagnostics_rp_cleaned, </span>
<span id="cb3-45"><a href="inter-omics.html#cb3-45"></a>    diagnostics_rn_cleaned, diagnostics_hp_pc_group, diagnostics_hn_pc_group, diagnostics_rp_pc_group, </span>
<span id="cb3-46"><a href="inter-omics.html#cb3-46"></a>    diagnostics_rn_pc_group, diagnostics_hp_ion, diagnostics_hn_ion, diagnostics_rp_ion, </span>
<span id="cb3-47"><a href="inter-omics.html#cb3-47"></a>    diagnostics_rn_ion)</span>
<span id="cb3-48"><a href="inter-omics.html#cb3-48"></a>diagnostics</span>
<span id="cb3-49"><a href="inter-omics.html#cb3-49"></a><span class="kw">write.csv</span>(diagnostics, <span class="st">&quot;mantel_stats_FUN.csv&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="inter-omics.html#cb4-1"></a>diagnostics &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/mantel_stats_FUN.csv&quot;</span>)</span>
<span id="cb4-2"><a href="inter-omics.html#cb4-2"></a>diagnostics<span class="op">$</span>X &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb4-3"><a href="inter-omics.html#cb4-3"></a>diagnostics &lt;-<span class="st"> </span>diagnostics[, <span class="kw">c</span>(<span class="st">&quot;statistic&quot;</span>, <span class="st">&quot;signif&quot;</span>, <span class="st">&quot;micro_samples&quot;</span>, <span class="st">&quot;micro_OTUs&quot;</span>, </span>
<span id="cb4-4"><a href="inter-omics.html#cb4-4"></a>    <span class="st">&quot;meta_samples&quot;</span>, <span class="st">&quot;meta_features&quot;</span>, <span class="st">&quot;Sponge.species&quot;</span>, <span class="st">&quot;Experiment&quot;</span>, <span class="st">&quot;data.set&quot;</span>)]</span>
<span id="cb4-5"><a href="inter-omics.html#cb4-5"></a></span>
<span id="cb4-6"><a href="inter-omics.html#cb4-6"></a><span class="kw">options</span>(<span class="dt">kableExtra.html.bsTable =</span> T)</span>
<span id="cb4-7"><a href="inter-omics.html#cb4-7"></a><span class="kw">kable</span>(diagnostics, <span class="dt">col.names =</span> <span class="kw">c</span>(<span class="st">&quot;Mantel statistic r&quot;</span>, <span class="st">&quot;significance&quot;</span>, <span class="st">&quot;N microbiome samples&quot;</span>, </span>
<span id="cb4-8"><a href="inter-omics.html#cb4-8"></a>    <span class="st">&quot;N OTUs&quot;</span>, <span class="st">&quot;N metabolome samples&quot;</span>, <span class="st">&quot;N features&quot;</span>, <span class="st">&quot;Sponge species&quot;</span>, <span class="st">&quot;Experiment&quot;</span>, </span>
<span id="cb4-9"><a href="inter-omics.html#cb4-9"></a>    <span class="st">&quot;data set&quot;</span>), <span class="dt">longtable =</span> T, <span class="dt">booktabs =</span> T, <span class="dt">caption =</span> <span class="st">&quot;Mantel test diagnostics diagnositcs comparing the microbiome and metabolome of the same sponge specimens&quot;</span>, </span>
<span id="cb4-10"><a href="inter-omics.html#cb4-10"></a>    <span class="dt">row.names =</span> <span class="ot">FALSE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">add_header_above</span>(<span class="kw">c</span>(<span class="dt">Diagnostics =</span> <span class="dv">6</span>, <span class="st">`</span><span class="dt">Data set attribution</span><span class="st">`</span> =<span class="st"> </span><span class="dv">3</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb4-11"><a href="inter-omics.html#cb4-11"></a><span class="st">    </span><span class="kw">kable_styling</span>(<span class="dt">bootstrap_options =</span> <span class="kw">c</span>(<span class="st">&quot;striped&quot;</span>, <span class="st">&quot;hover&quot;</span>, <span class="st">&quot;bordered&quot;</span>, <span class="st">&quot;condensed&quot;</span>, </span>
<span id="cb4-12"><a href="inter-omics.html#cb4-12"></a>        <span class="st">&quot;responsive&quot;</span>), <span class="dt">font_size =</span> <span class="dv">12</span>, <span class="dt">full_width =</span> F, <span class="dt">latex_options =</span> <span class="kw">c</span>(<span class="st">&quot;striped&quot;</span>, </span>
<span id="cb4-13"><a href="inter-omics.html#cb4-13"></a>        <span class="st">&quot;scale_down&quot;</span>))</span></code></pre></div>
<table class="table table-striped table-hover table-bordered table-condensed table-responsive" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
(#tab:io4_mantel_results)Mantel test diagnostics diagnositcs comparing the microbiome and metabolome of the same sponge specimens
</caption>
<thead>
<tr>
<th style="border-bottom:hidden; padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="6">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Diagnostics
</div>
</th>
<th style="border-bottom:hidden; padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="3">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Data set attribution
</div>
</th>
</tr>
<tr>
<th style="text-align:right;">
Mantel statistic r
</th>
<th style="text-align:right;">
significance
</th>
<th style="text-align:right;">
N microbiome samples
</th>
<th style="text-align:right;">
N OTUs
</th>
<th style="text-align:right;">
N metabolome samples
</th>
<th style="text-align:right;">
N features
</th>
<th style="text-align:left;">
Sponge species
</th>
<th style="text-align:left;">
Experiment
</th>
<th style="text-align:left;">
data set
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
0.6076416
</td>
<td style="text-align:right;">
0.004
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
3507
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
HILIC pos
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.3857788
</td>
<td style="text-align:right;">
0.011
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
3507
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
HILIC pos
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.4576405
</td>
<td style="text-align:right;">
0.008
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
3507
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
HILIC pos
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.4516469
</td>
<td style="text-align:right;">
0.006
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
2808
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
HILIC neg
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.1513297
</td>
<td style="text-align:right;">
0.186
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
2808
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
HILIC neg
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.1854966
</td>
<td style="text-align:right;">
0.168
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
2808
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
HILIC neg
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
-0.2014493
</td>
<td style="text-align:right;">
0.833
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
4673
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
RP pos
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.3127632
</td>
<td style="text-align:right;">
0.064
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
4673
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
RP pos
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.3323657
</td>
<td style="text-align:right;">
0.035
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
4673
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
RP pos
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
-0.2465894
</td>
<td style="text-align:right;">
0.845
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
3166
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
RP neg
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.1960950
</td>
<td style="text-align:right;">
0.173
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
3166
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
RP neg
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.1721750
</td>
<td style="text-align:right;">
0.174
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
3166
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
RP neg
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.5836627
</td>
<td style="text-align:right;">
0.009
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
2212
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
HILIC pos
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.4082626
</td>
<td style="text-align:right;">
0.016
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
2212
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
HILIC pos
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.4483931
</td>
<td style="text-align:right;">
0.012
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
2212
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
HILIC pos
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.3951252
</td>
<td style="text-align:right;">
0.018
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
1351
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
HILIC neg
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.3211093
</td>
<td style="text-align:right;">
0.040
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
1351
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
HILIC neg
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.0315675
</td>
<td style="text-align:right;">
0.414
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
1351
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
HILIC neg
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
-0.1600791
</td>
<td style="text-align:right;">
0.774
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
2736
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
RP pos
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.3181249
</td>
<td style="text-align:right;">
0.064
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
2736
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
RP pos
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.3008501
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
2736
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
RP pos
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
-0.1879022
</td>
<td style="text-align:right;">
0.790
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
1678
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
RP neg
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.2034548
</td>
<td style="text-align:right;">
0.156
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
1678
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
RP neg
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.1507257
</td>
<td style="text-align:right;">
0.198
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
1678
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
RP neg
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.3351779
</td>
<td style="text-align:right;">
0.050
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
105
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
HILIC pos
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
0.2908357
</td>
<td style="text-align:right;">
0.086
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
105
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
HILIC pos
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
0.4683599
</td>
<td style="text-align:right;">
0.010
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
105
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
HILIC pos
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
0.4429513
</td>
<td style="text-align:right;">
0.005
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
123
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
HILIC neg
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
0.0727247
</td>
<td style="text-align:right;">
0.320
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
123
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
HILIC neg
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
0.2320236
</td>
<td style="text-align:right;">
0.115
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
123
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
HILIC neg
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
-0.2569170
</td>
<td style="text-align:right;">
0.908
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
171
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
RP pos
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
0.3070980
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
171
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
RP pos
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
0.2656853
</td>
<td style="text-align:right;">
0.072
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
171
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
RP pos
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
-0.1220077
</td>
<td style="text-align:right;">
0.703
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
105
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
RP neg
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
-0.0505950
</td>
<td style="text-align:right;">
0.597
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
105
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
RP neg
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
0.1218225
</td>
<td style="text-align:right;">
0.217
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
105
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
RP neg
</td>
<td style="text-align:left;">
ion
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="inter-omics.html#cb5-1"></a>dig &lt;-<span class="st"> </span>diagnostics[diagnostics<span class="op">$</span>signif <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.05</span>, ]</span>
<span id="cb5-2"><a href="inter-omics.html#cb5-2"></a>a &lt;-<span class="st"> </span><span class="kw">aggregate</span>(dig, <span class="dt">by =</span> <span class="kw">list</span>(dig<span class="op">$</span>Sponge.species, dig<span class="op">$</span>Experiment, dig<span class="op">$</span>data.set), </span>
<span id="cb5-3"><a href="inter-omics.html#cb5-3"></a>    <span class="dt">FUN =</span> <span class="st">&quot;length&quot;</span>)</span>
<span id="cb5-4"><a href="inter-omics.html#cb5-4"></a><span class="kw">summary</span>(a<span class="op">$</span>Group<span class="fl">.1</span>)</span></code></pre></div>
<pre><code>##  Geodia barretti Stryphnus fortis  Weberella bursa 
##                6                3                4</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="inter-omics.html#cb7-1"></a><span class="kw">summary</span>(a<span class="op">$</span>Group<span class="fl">.2</span>)</span></code></pre></div>
<pre><code>## HILIC neg HILIC pos    RP neg    RP pos 
##         4         8         0         1</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="inter-omics.html#cb9-1"></a><span class="kw">summary</span>(a<span class="op">$</span>Group<span class="fl">.3</span>)</span></code></pre></div>
<pre><code>##  cleaned      ion pc_group 
##        5        3        5</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="inter-omics.html#cb11-1"></a><span class="kw">rm</span>(diagnostics_hp_cleaned, diagnostics_hn_cleaned, diagnostics_rp_cleaned, diagnostics_rn_cleaned, </span>
<span id="cb11-2"><a href="inter-omics.html#cb11-2"></a>    diagnostics_hp_ion, diagnostics_hn_ion, diagnostics_rp_ion, diagnostics_rn_ion, </span>
<span id="cb11-3"><a href="inter-omics.html#cb11-3"></a>    diagnostics_hp_pc_group, diagnostics_hn_pc_group, diagnostics_rp_pc_group, diagnostics_rn_pc_group)</span>
<span id="cb11-4"><a href="inter-omics.html#cb11-4"></a><span class="kw">rm</span>(hilic_pos, hilic_neg, rp_pos, rp_neg)</span></code></pre></div>
<p>As we can see from the table, in 13 cases, the Mantel test returns a significant correlation between the two matrices. Above you can see the the significant tests broken down by sponge species, HPCL-experiment and filtering approach.</p>
</div>
<div id="procrustes-rotation-and-protest-code-and-tabular-output" class="section level3">
<h3><span class="header-section-number">4.1.4</span> Procrustes rotation and protest code and tabular output</h3>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="inter-omics.html#cb12-1"></a><span class="co">### =============================== PROTEST ====================================</span></span>
<span id="cb12-2"><a href="inter-omics.html#cb12-2"></a></span>
<span id="cb12-3"><a href="inter-omics.html#cb12-3"></a><span class="co"># run one of this at a time</span></span>
<span id="cb12-4"><a href="inter-omics.html#cb12-4"></a>metabolomes &lt;-<span class="st"> </span><span class="kw">gymnastics</span>(hilic_pos, <span class="st">&quot;H_p&quot;</span>)</span>
<span id="cb12-5"><a href="inter-omics.html#cb12-5"></a>metabolomes &lt;-<span class="st"> </span><span class="kw">gymnastics</span>(hilic_neg, <span class="st">&quot;H_n&quot;</span>)</span>
<span id="cb12-6"><a href="inter-omics.html#cb12-6"></a>metabolomes &lt;-<span class="st"> </span><span class="kw">gymnastics</span>(rp_pos, <span class="st">&quot;R_p&quot;</span>)</span>
<span id="cb12-7"><a href="inter-omics.html#cb12-7"></a>metabolomes &lt;-<span class="st"> </span><span class="kw">gymnastics</span>(rp_neg, <span class="st">&quot;R_n&quot;</span>)</span>
<span id="cb12-8"><a href="inter-omics.html#cb12-8"></a></span>
<span id="cb12-9"><a href="inter-omics.html#cb12-9"></a><span class="co"># run this</span></span>
<span id="cb12-10"><a href="inter-omics.html#cb12-10"></a>congruent_dfs &lt;-<span class="st"> </span><span class="kw">congruency</span>(metabolomes)</span>
<span id="cb12-11"><a href="inter-omics.html#cb12-11"></a></span>
<span id="cb12-12"><a href="inter-omics.html#cb12-12"></a><span class="co"># run one of these: PROTEST </span><span class="al">TEST</span></span>
<span id="cb12-13"><a href="inter-omics.html#cb12-13"></a>diagnostics_hp_cleaned &lt;-<span class="st"> </span><span class="kw">ordination</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;HILIC pos&quot;</span>, </span>
<span id="cb12-14"><a href="inter-omics.html#cb12-14"></a>    <span class="dt">filtering =</span> <span class="st">&quot;cleaned&quot;</span>)</span>
<span id="cb12-15"><a href="inter-omics.html#cb12-15"></a>diagnostics_hn_cleaned &lt;-<span class="st"> </span><span class="kw">ordination</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;HILIC neg&quot;</span>, </span>
<span id="cb12-16"><a href="inter-omics.html#cb12-16"></a>    <span class="dt">filtering =</span> <span class="st">&quot;cleaned&quot;</span>)</span>
<span id="cb12-17"><a href="inter-omics.html#cb12-17"></a>diagnostics_rp_cleaned &lt;-<span class="st"> </span><span class="kw">ordination</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;RP pos&quot;</span>, </span>
<span id="cb12-18"><a href="inter-omics.html#cb12-18"></a>    <span class="dt">filtering =</span> <span class="st">&quot;cleaned&quot;</span>)</span>
<span id="cb12-19"><a href="inter-omics.html#cb12-19"></a>diagnostics_rn_cleaned &lt;-<span class="st"> </span><span class="kw">ordination</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;RP neg&quot;</span>, </span>
<span id="cb12-20"><a href="inter-omics.html#cb12-20"></a>    <span class="dt">filtering =</span> <span class="st">&quot;cleaned&quot;</span>)</span>
<span id="cb12-21"><a href="inter-omics.html#cb12-21"></a></span>
<span id="cb12-22"><a href="inter-omics.html#cb12-22"></a>diagnostics_hp_ion &lt;-<span class="st"> </span><span class="kw">ordination</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;HILIC pos&quot;</span>, </span>
<span id="cb12-23"><a href="inter-omics.html#cb12-23"></a>    <span class="dt">filtering =</span> <span class="st">&quot;ion&quot;</span>)</span>
<span id="cb12-24"><a href="inter-omics.html#cb12-24"></a>diagnostics_hn_ion &lt;-<span class="st"> </span><span class="kw">ordination</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;HILIC neg&quot;</span>, </span>
<span id="cb12-25"><a href="inter-omics.html#cb12-25"></a>    <span class="dt">filtering =</span> <span class="st">&quot;ion&quot;</span>)</span>
<span id="cb12-26"><a href="inter-omics.html#cb12-26"></a>diagnostics_rp_ion &lt;-<span class="st"> </span><span class="kw">ordination</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;RP pos&quot;</span>, </span>
<span id="cb12-27"><a href="inter-omics.html#cb12-27"></a>    <span class="dt">filtering =</span> <span class="st">&quot;ion&quot;</span>)</span>
<span id="cb12-28"><a href="inter-omics.html#cb12-28"></a>diagnostics_rn_ion &lt;-<span class="st"> </span><span class="kw">ordination</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;RP neg&quot;</span>, </span>
<span id="cb12-29"><a href="inter-omics.html#cb12-29"></a>    <span class="dt">filtering =</span> <span class="st">&quot;ion&quot;</span>)</span>
<span id="cb12-30"><a href="inter-omics.html#cb12-30"></a></span>
<span id="cb12-31"><a href="inter-omics.html#cb12-31"></a>diagnostics_hp_pc_group &lt;-<span class="st"> </span><span class="kw">ordination</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;HILIC pos&quot;</span>, </span>
<span id="cb12-32"><a href="inter-omics.html#cb12-32"></a>    <span class="dt">filtering =</span> <span class="st">&quot;pc_group&quot;</span>)</span>
<span id="cb12-33"><a href="inter-omics.html#cb12-33"></a>diagnostics_hn_pc_group &lt;-<span class="st"> </span><span class="kw">ordination</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;HILIC neg&quot;</span>, </span>
<span id="cb12-34"><a href="inter-omics.html#cb12-34"></a>    <span class="dt">filtering =</span> <span class="st">&quot;pc_group&quot;</span>)</span>
<span id="cb12-35"><a href="inter-omics.html#cb12-35"></a>diagnostics_rp_pc_group &lt;-<span class="st"> </span><span class="kw">ordination</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;RP pos&quot;</span>, </span>
<span id="cb12-36"><a href="inter-omics.html#cb12-36"></a>    <span class="dt">filtering =</span> <span class="st">&quot;pc_group&quot;</span>)</span>
<span id="cb12-37"><a href="inter-omics.html#cb12-37"></a>diagnostics_rn_pc_group &lt;-<span class="st"> </span><span class="kw">ordination</span>(<span class="dt">congruent_dfs =</span> congruent_dfs, <span class="dt">experiment =</span> <span class="st">&quot;RP neg&quot;</span>, </span>
<span id="cb12-38"><a href="inter-omics.html#cb12-38"></a>    <span class="dt">filtering =</span> <span class="st">&quot;pc_group&quot;</span>)</span>
<span id="cb12-39"><a href="inter-omics.html#cb12-39"></a></span>
<span id="cb12-40"><a href="inter-omics.html#cb12-40"></a></span>
<span id="cb12-41"><a href="inter-omics.html#cb12-41"></a><span class="co">### Combine all test results into one file</span></span>
<span id="cb12-42"><a href="inter-omics.html#cb12-42"></a></span>
<span id="cb12-43"><a href="inter-omics.html#cb12-43"></a>diagnostics &lt;-<span class="st"> </span>diagnostics_hp_cleaned</span>
<span id="cb12-44"><a href="inter-omics.html#cb12-44"></a>diagnostics &lt;-<span class="st"> </span><span class="kw">rbind</span>(diagnostics, diagnostics_hn_cleaned, diagnostics_rp_cleaned, </span>
<span id="cb12-45"><a href="inter-omics.html#cb12-45"></a>    diagnostics_rn_cleaned, diagnostics_hp_pc_group, diagnostics_hn_pc_group, diagnostics_rp_pc_group, </span>
<span id="cb12-46"><a href="inter-omics.html#cb12-46"></a>    diagnostics_rn_pc_group, diagnostics_hp_ion, diagnostics_hn_ion, diagnostics_rp_ion, </span>
<span id="cb12-47"><a href="inter-omics.html#cb12-47"></a>    diagnostics_rn_ion)</span>
<span id="cb12-48"><a href="inter-omics.html#cb12-48"></a>diagnostics</span>
<span id="cb12-49"><a href="inter-omics.html#cb12-49"></a></span>
<span id="cb12-50"><a href="inter-omics.html#cb12-50"></a><span class="kw">write.csv</span>(diagnostics, <span class="st">&quot;protest_stats_FUN.csv&quot;</span>)</span>
<span id="cb12-51"><a href="inter-omics.html#cb12-51"></a></span>
<span id="cb12-52"><a href="inter-omics.html#cb12-52"></a><span class="kw">rm</span>(diagnostics_hp_cleaned, diagnostics_hn_cleaned, diagnostics_rp_cleaned, diagnostics_rn_cleaned, </span>
<span id="cb12-53"><a href="inter-omics.html#cb12-53"></a>    diagnostics_hp_ion, diagnostics_hn_ion, diagnostics_rp_ion, diagnostics_rn_ion, </span>
<span id="cb12-54"><a href="inter-omics.html#cb12-54"></a>    diagnostics_hp_pc_group, diagnostics_hn_pc_group, diagnostics_rp_pc_group, diagnostics_rn_pc_group, </span>
<span id="cb12-55"><a href="inter-omics.html#cb12-55"></a>    diagnostics)</span>
<span id="cb12-56"><a href="inter-omics.html#cb12-56"></a><span class="kw">rm</span>(hilic_pos, hilic_neg, rp_pos, rp_neg)</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="inter-omics.html#cb13-1"></a>diagnostics &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/protest_stats_FUN.csv&quot;</span>)</span>
<span id="cb13-2"><a href="inter-omics.html#cb13-2"></a>diagnostics<span class="op">$</span>X &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb13-3"><a href="inter-omics.html#cb13-3"></a>diagnostics &lt;-<span class="st"> </span>diagnostics[, <span class="kw">c</span>(<span class="st">&quot;Procrustes.SS&quot;</span>, <span class="st">&quot;correlation.in.sym..rotation&quot;</span>, <span class="st">&quot;signif&quot;</span>, </span>
<span id="cb13-4"><a href="inter-omics.html#cb13-4"></a>    <span class="st">&quot;micro_samples&quot;</span>, <span class="st">&quot;micro_OTUs&quot;</span>, <span class="st">&quot;meta_samples&quot;</span>, <span class="st">&quot;meta_features&quot;</span>, <span class="st">&quot;Sponge.species&quot;</span>, </span>
<span id="cb13-5"><a href="inter-omics.html#cb13-5"></a>    <span class="st">&quot;Experiment&quot;</span>, <span class="st">&quot;data.set&quot;</span>)]</span>
<span id="cb13-6"><a href="inter-omics.html#cb13-6"></a></span>
<span id="cb13-7"><a href="inter-omics.html#cb13-7"></a><span class="kw">options</span>(<span class="dt">kableExtra.html.bsTable =</span> T)</span>
<span id="cb13-8"><a href="inter-omics.html#cb13-8"></a><span class="kw">kable</span>(diagnostics, <span class="dt">col.names =</span> <span class="kw">c</span>(<span class="st">&quot;Procrustes sum of squares&quot;</span>, <span class="st">&quot;correlation in symmetric rotation&quot;</span>, </span>
<span id="cb13-9"><a href="inter-omics.html#cb13-9"></a>    <span class="st">&quot;significance&quot;</span>, <span class="st">&quot;N microbiome samples&quot;</span>, <span class="st">&quot;N OTUs&quot;</span>, <span class="st">&quot;N metabolome samples&quot;</span>, <span class="st">&quot;N features&quot;</span>, </span>
<span id="cb13-10"><a href="inter-omics.html#cb13-10"></a>    <span class="st">&quot;Sponge species&quot;</span>, <span class="st">&quot;Experiment&quot;</span>, <span class="st">&quot;data set&quot;</span>), <span class="dt">longtable =</span> T, <span class="dt">booktabs =</span> T, <span class="dt">caption =</span> <span class="st">&quot;Protest diagnostics comparing the microbiome and metabolome of the same sponge specimens&quot;</span>, </span>
<span id="cb13-11"><a href="inter-omics.html#cb13-11"></a>    <span class="dt">row.names =</span> <span class="ot">FALSE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">add_header_above</span>(<span class="kw">c</span>(<span class="dt">Diagnostics =</span> <span class="dv">7</span>, <span class="st">`</span><span class="dt">Data set attribution</span><span class="st">`</span> =<span class="st"> </span><span class="dv">3</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb13-12"><a href="inter-omics.html#cb13-12"></a><span class="st">    </span><span class="kw">kable_styling</span>(<span class="dt">bootstrap_options =</span> <span class="kw">c</span>(<span class="st">&quot;striped&quot;</span>, <span class="st">&quot;hover&quot;</span>, <span class="st">&quot;bordered&quot;</span>, <span class="st">&quot;condensed&quot;</span>, </span>
<span id="cb13-13"><a href="inter-omics.html#cb13-13"></a>        <span class="st">&quot;responsive&quot;</span>), <span class="dt">font_size =</span> <span class="dv">12</span>, <span class="dt">full_width =</span> F, <span class="dt">latex_options =</span> <span class="kw">c</span>(<span class="st">&quot;striped&quot;</span>, </span>
<span id="cb13-14"><a href="inter-omics.html#cb13-14"></a>        <span class="st">&quot;scale_down&quot;</span>))</span></code></pre></div>
<table class="table table-striped table-hover table-bordered table-condensed table-responsive" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
(#tab:io6_prt)Protest diagnostics comparing the microbiome and metabolome of the same sponge specimens
</caption>
<thead>
<tr>
<th style="border-bottom:hidden; padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="7">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Diagnostics
</div>
</th>
<th style="border-bottom:hidden; padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="3">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Data set attribution
</div>
</th>
</tr>
<tr>
<th style="text-align:right;">
Procrustes sum of squares
</th>
<th style="text-align:right;">
correlation in symmetric rotation
</th>
<th style="text-align:right;">
significance
</th>
<th style="text-align:right;">
N microbiome samples
</th>
<th style="text-align:right;">
N OTUs
</th>
<th style="text-align:right;">
N metabolome samples
</th>
<th style="text-align:right;">
N features
</th>
<th style="text-align:left;">
Sponge species
</th>
<th style="text-align:left;">
Experiment
</th>
<th style="text-align:left;">
data set
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
0.4901473
</td>
<td style="text-align:right;">
0.7140397
</td>
<td style="text-align:right;">
0.009
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
3507
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
HILIC pos
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.3360196
</td>
<td style="text-align:right;">
0.8148499
</td>
<td style="text-align:right;">
0.001
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
3507
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
HILIC pos
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.6451196
</td>
<td style="text-align:right;">
0.5957184
</td>
<td style="text-align:right;">
0.009
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
3507
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
HILIC pos
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.5056719
</td>
<td style="text-align:right;">
0.7030847
</td>
<td style="text-align:right;">
0.015
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
2808
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
HILIC neg
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.7144829
</td>
<td style="text-align:right;">
0.5343380
</td>
<td style="text-align:right;">
0.043
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
2808
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
HILIC neg
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.6454847
</td>
<td style="text-align:right;">
0.5954119
</td>
<td style="text-align:right;">
0.003
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
2808
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
HILIC neg
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.7241690
</td>
<td style="text-align:right;">
0.5251961
</td>
<td style="text-align:right;">
0.124
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
4673
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
RP pos
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.5535784
</td>
<td style="text-align:right;">
0.6681479
</td>
<td style="text-align:right;">
0.005
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
4673
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
RP pos
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.9089831
</td>
<td style="text-align:right;">
0.3016901
</td>
<td style="text-align:right;">
0.513
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
4673
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
RP pos
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.7744081
</td>
<td style="text-align:right;">
0.4749652
</td>
<td style="text-align:right;">
0.229
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
3166
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
RP neg
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.7769037
</td>
<td style="text-align:right;">
0.4723307
</td>
<td style="text-align:right;">
0.107
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
3166
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
RP neg
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.9426954
</td>
<td style="text-align:right;">
0.2393838
</td>
<td style="text-align:right;">
0.656
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
3166
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
RP neg
</td>
<td style="text-align:left;">
cleaned
</td>
</tr>
<tr>
<td style="text-align:right;">
0.4903060
</td>
<td style="text-align:right;">
0.7139285
</td>
<td style="text-align:right;">
0.008
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
2212
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
HILIC pos
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.6122646
</td>
<td style="text-align:right;">
0.6226840
</td>
<td style="text-align:right;">
0.009
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
2212
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
HILIC pos
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.5611609
</td>
<td style="text-align:right;">
0.6624494
</td>
<td style="text-align:right;">
0.002
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
2212
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
HILIC pos
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.4442919
</td>
<td style="text-align:right;">
0.7454583
</td>
<td style="text-align:right;">
0.007
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
1351
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
HILIC neg
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.6072203
</td>
<td style="text-align:right;">
0.6267214
</td>
<td style="text-align:right;">
0.013
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
1351
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
HILIC neg
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.7647882
</td>
<td style="text-align:right;">
0.4849864
</td>
<td style="text-align:right;">
0.062
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
1351
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
HILIC neg
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.7332277
</td>
<td style="text-align:right;">
0.5165001
</td>
<td style="text-align:right;">
0.146
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
2736
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
RP pos
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.5557670
</td>
<td style="text-align:right;">
0.6665081
</td>
<td style="text-align:right;">
0.002
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
2736
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
RP pos
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.9184545
</td>
<td style="text-align:right;">
0.2855616
</td>
<td style="text-align:right;">
0.579
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
2736
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
RP pos
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.7952845
</td>
<td style="text-align:right;">
0.4524550
</td>
<td style="text-align:right;">
0.314
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
1678
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
RP neg
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.7513407
</td>
<td style="text-align:right;">
0.4986575
</td>
<td style="text-align:right;">
0.089
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
1678
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
RP neg
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.9669093
</td>
<td style="text-align:right;">
0.1819085
</td>
<td style="text-align:right;">
0.851
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
1678
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
RP neg
</td>
<td style="text-align:left;">
pc_group
</td>
</tr>
<tr>
<td style="text-align:right;">
0.7744285
</td>
<td style="text-align:right;">
0.4749437
</td>
<td style="text-align:right;">
0.254
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
3166
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
HILIC pos
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
0.7768298
</td>
<td style="text-align:right;">
0.4724090
</td>
<td style="text-align:right;">
0.119
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
3166
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
HILIC pos
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
0.9427350
</td>
<td style="text-align:right;">
0.2393010
</td>
<td style="text-align:right;">
0.671
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
3166
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
HILIC pos
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
0.3977778
</td>
<td style="text-align:right;">
0.7760297
</td>
<td style="text-align:right;">
0.005
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
123
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
HILIC neg
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
0.7620196
</td>
<td style="text-align:right;">
0.4878323
</td>
<td style="text-align:right;">
0.101
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
123
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
HILIC neg
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
0.6263664
</td>
<td style="text-align:right;">
0.6112558
</td>
<td style="text-align:right;">
0.007
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
123
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
HILIC neg
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
0.6879735
</td>
<td style="text-align:right;">
0.5585933
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
171
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
RP pos
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
0.6735153
</td>
<td style="text-align:right;">
0.5713884
</td>
<td style="text-align:right;">
0.038
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
171
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
RP pos
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
0.9830213
</td>
<td style="text-align:right;">
0.1303022
</td>
<td style="text-align:right;">
0.961
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
171
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
RP pos
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
0.7788793
</td>
<td style="text-align:right;">
0.4702347
</td>
<td style="text-align:right;">
0.271
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
420
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
105
</td>
<td style="text-align:left;">
Geodia barretti
</td>
<td style="text-align:left;">
RP neg
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
0.9754054
</td>
<td style="text-align:right;">
0.1568265
</td>
<td style="text-align:right;">
0.767
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
461
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
105
</td>
<td style="text-align:left;">
Stryphnus fortis
</td>
<td style="text-align:left;">
RP neg
</td>
<td style="text-align:left;">
ion
</td>
</tr>
<tr>
<td style="text-align:right;">
0.9595716
</td>
<td style="text-align:right;">
0.2010681
</td>
<td style="text-align:right;">
0.791
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
105
</td>
<td style="text-align:left;">
Weberella bursa
</td>
<td style="text-align:left;">
RP neg
</td>
<td style="text-align:left;">
ion
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="inter-omics.html#cb14-1"></a>dig &lt;-<span class="st"> </span>diagnostics[diagnostics<span class="op">$</span>signif <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.05</span>, ]</span>
<span id="cb14-2"><a href="inter-omics.html#cb14-2"></a>a &lt;-<span class="st"> </span><span class="kw">aggregate</span>(dig, <span class="dt">by =</span> <span class="kw">list</span>(dig<span class="op">$</span>Sponge.species, dig<span class="op">$</span>Experiment, dig<span class="op">$</span>data.set), </span>
<span id="cb14-3"><a href="inter-omics.html#cb14-3"></a>    <span class="dt">FUN =</span> <span class="st">&quot;length&quot;</span>)</span>
<span id="cb14-4"><a href="inter-omics.html#cb14-4"></a><span class="kw">summary</span>(a<span class="op">$</span>Group<span class="fl">.1</span>)</span></code></pre></div>
<pre><code>##  Geodia barretti Stryphnus fortis  Weberella bursa 
##                5                7                4</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="inter-omics.html#cb16-1"></a><span class="kw">summary</span>(a<span class="op">$</span>Group<span class="fl">.2</span>)</span></code></pre></div>
<pre><code>## HILIC neg HILIC pos    RP neg    RP pos 
##         7         6         0         3</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="inter-omics.html#cb18-1"></a><span class="kw">summary</span>(a<span class="op">$</span>Group<span class="fl">.3</span>)</span></code></pre></div>
<pre><code>##  cleaned      ion pc_group 
##        7        3        6</code></pre>
<p>Out of the 36 tests performed, 16 are significant (<em>p</em> <span class="math inline">\(\leq\)</span> 0.05). Immediately above you can see the the significant tests broken down by sponge species, HPCL-experiment and filtering approach.</p>
</div>
</div>
<div id="correlating-barettin-and-prokaryotic-relative-abundance" class="section level2">
<h2><span class="header-section-number">4.2</span> Correlating barettin and prokaryotic relative abundance</h2>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="inter-omics.html#cb20-1"></a>cmp &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/metabolite_master_20190605.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb20-2"><a href="inter-omics.html#cb20-2"></a>cmp<span class="op">$</span>X &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb20-3"><a href="inter-omics.html#cb20-3"></a>meta_data &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/Steffen_et_al_metadata_PANGAEA.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb20-4"><a href="inter-omics.html#cb20-4"></a>micro &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/OTU_all_R.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb20-5"><a href="inter-omics.html#cb20-5"></a></span>
<span id="cb20-6"><a href="inter-omics.html#cb20-6"></a><span class="co"># preparing meta data</span></span>
<span id="cb20-7"><a href="inter-omics.html#cb20-7"></a>meta_data_prep &lt;-<span class="st"> </span><span class="cf">function</span>(meta_data) {</span>
<span id="cb20-8"><a href="inter-omics.html#cb20-8"></a>    meta_data &lt;-<span class="st"> </span>meta_data[, <span class="kw">c</span>(<span class="st">&quot;unified_ID&quot;</span>, <span class="st">&quot;Depth&quot;</span>, <span class="st">&quot;Latitude&quot;</span>, <span class="st">&quot;Longitude&quot;</span>, <span class="st">&quot;MeanBottomTemp_Cdeg&quot;</span>, </span>
<span id="cb20-9"><a href="inter-omics.html#cb20-9"></a>        <span class="st">&quot;MeanBotSalinity_PSU&quot;</span>, <span class="st">&quot;YEAR&quot;</span>)]</span>
<span id="cb20-10"><a href="inter-omics.html#cb20-10"></a>    <span class="kw">colnames</span>(meta_data) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;unified_ID&quot;</span>, <span class="st">&quot;Depth&quot;</span>, <span class="st">&quot;Latitude&quot;</span>, <span class="st">&quot;Longitude&quot;</span>, <span class="st">&quot;Temperature&quot;</span>, </span>
<span id="cb20-11"><a href="inter-omics.html#cb20-11"></a>        <span class="st">&quot;Salinity&quot;</span>, <span class="st">&quot;Year&quot;</span>)</span>
<span id="cb20-12"><a href="inter-omics.html#cb20-12"></a>    meta_data &lt;-<span class="st"> </span>meta_data[<span class="op">!</span>(<span class="kw">str_sub</span>(meta_data<span class="op">$</span>unified_ID, <span class="dv">1</span>, <span class="dv">2</span>) <span class="op">==</span><span class="st"> &quot;QC&quot;</span>), ]</span>
<span id="cb20-13"><a href="inter-omics.html#cb20-13"></a>    meta_data[] &lt;-<span class="st"> </span><span class="kw">lapply</span>(meta_data, <span class="cf">function</span>(x) <span class="cf">if</span> (<span class="kw">is.factor</span>(x)) </span>
<span id="cb20-14"><a href="inter-omics.html#cb20-14"></a>        <span class="kw">factor</span>(x) <span class="cf">else</span> x)</span>
<span id="cb20-15"><a href="inter-omics.html#cb20-15"></a>    <span class="co"># Gb12, Gb20 and Gb21 are missing temperature and salinity. Imputing data from</span></span>
<span id="cb20-16"><a href="inter-omics.html#cb20-16"></a>    <span class="co"># closeby samples:</span></span>
<span id="cb20-17"><a href="inter-omics.html#cb20-17"></a>    meta_data<span class="op">$</span>Salinity[meta_data<span class="op">$</span>unified_ID <span class="op">==</span><span class="st"> &quot;Gb12&quot;</span>] &lt;-<span class="st"> </span><span class="fl">34.92</span></span>
<span id="cb20-18"><a href="inter-omics.html#cb20-18"></a>    meta_data<span class="op">$</span>Salinity[meta_data<span class="op">$</span>unified_ID <span class="op">==</span><span class="st"> &quot;Gb20&quot;</span>] &lt;-<span class="st"> </span><span class="fl">34.92</span></span>
<span id="cb20-19"><a href="inter-omics.html#cb20-19"></a>    meta_data<span class="op">$</span>Salinity[meta_data<span class="op">$</span>unified_ID <span class="op">==</span><span class="st"> &quot;Gb21&quot;</span>] &lt;-<span class="st"> </span><span class="fl">34.56</span></span>
<span id="cb20-20"><a href="inter-omics.html#cb20-20"></a>    meta_data<span class="op">$</span>Temperature[meta_data<span class="op">$</span>unified_ID <span class="op">==</span><span class="st"> &quot;Gb12&quot;</span>] &lt;-<span class="st"> </span><span class="fl">3.71</span></span>
<span id="cb20-21"><a href="inter-omics.html#cb20-21"></a>    meta_data<span class="op">$</span>Temperature[meta_data<span class="op">$</span>unified_ID <span class="op">==</span><span class="st"> &quot;Gb20&quot;</span>] &lt;-<span class="st"> </span><span class="fl">3.65</span></span>
<span id="cb20-22"><a href="inter-omics.html#cb20-22"></a>    meta_data<span class="op">$</span>Temperature[meta_data<span class="op">$</span>unified_ID <span class="op">==</span><span class="st"> &quot;Gb21&quot;</span>] &lt;-<span class="st"> </span><span class="fl">2.32</span></span>
<span id="cb20-23"><a href="inter-omics.html#cb20-23"></a>    meta_data[<span class="st">&quot;spec&quot;</span>] &lt;-<span class="st"> </span><span class="kw">str_sub</span>(meta_data<span class="op">$</span>unified_ID, <span class="dv">1</span>, <span class="dv">2</span>)</span>
<span id="cb20-24"><a href="inter-omics.html#cb20-24"></a>    meta_data &lt;-<span class="st"> </span>meta_data[<span class="kw">order</span>(meta_data<span class="op">$</span>unified_ID), ]</span>
<span id="cb20-25"><a href="inter-omics.html#cb20-25"></a>    <span class="kw">return</span>(meta_data)</span>
<span id="cb20-26"><a href="inter-omics.html#cb20-26"></a>}</span>
<span id="cb20-27"><a href="inter-omics.html#cb20-27"></a></span>
<span id="cb20-28"><a href="inter-omics.html#cb20-28"></a>meta_data &lt;-<span class="st"> </span><span class="kw">meta_data_prep</span>(meta_data)</span>
<span id="cb20-29"><a href="inter-omics.html#cb20-29"></a></span>
<span id="cb20-30"><a href="inter-omics.html#cb20-30"></a><span class="co"># separating OTU tables by sponge</span></span>
<span id="cb20-31"><a href="inter-omics.html#cb20-31"></a>OTU_prep_sqrt &lt;-<span class="st"> </span><span class="cf">function</span>(micro) {</span>
<span id="cb20-32"><a href="inter-omics.html#cb20-32"></a>    <span class="kw">rownames</span>(micro) &lt;-<span class="st"> </span>micro<span class="op">$</span>Sample_ID</span>
<span id="cb20-33"><a href="inter-omics.html#cb20-33"></a>    micro<span class="op">$</span>Sample_ID &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb20-34"><a href="inter-omics.html#cb20-34"></a>    <span class="co"># micro &lt;- sqrt(micro) #sqrt could be toggled on/off here</span></span>
<span id="cb20-35"><a href="inter-omics.html#cb20-35"></a>    </span>
<span id="cb20-36"><a href="inter-omics.html#cb20-36"></a>    micro_gb &lt;-<span class="st"> </span>micro[(<span class="kw">str_sub</span>(<span class="kw">rownames</span>(micro), <span class="dv">1</span>, <span class="dv">2</span>) <span class="op">==</span><span class="st"> &quot;Gb&quot;</span>), ]</span>
<span id="cb20-37"><a href="inter-omics.html#cb20-37"></a>    micro_sf &lt;-<span class="st"> </span>micro[(<span class="kw">str_sub</span>(<span class="kw">rownames</span>(micro), <span class="dv">1</span>, <span class="dv">2</span>) <span class="op">==</span><span class="st"> &quot;Sf&quot;</span>), ]</span>
<span id="cb20-38"><a href="inter-omics.html#cb20-38"></a>    micro_wb &lt;-<span class="st"> </span>micro[(<span class="kw">str_sub</span>(<span class="kw">rownames</span>(micro), <span class="dv">1</span>, <span class="dv">2</span>) <span class="op">==</span><span class="st"> &quot;Wb&quot;</span>), ]</span>
<span id="cb20-39"><a href="inter-omics.html#cb20-39"></a>    </span>
<span id="cb20-40"><a href="inter-omics.html#cb20-40"></a>    micro_gb &lt;-<span class="st"> </span>micro_gb[, <span class="kw">colSums</span>(micro_gb <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>]</span>
<span id="cb20-41"><a href="inter-omics.html#cb20-41"></a>    micro_sf &lt;-<span class="st"> </span>micro_sf[, <span class="kw">colSums</span>(micro_sf <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>]</span>
<span id="cb20-42"><a href="inter-omics.html#cb20-42"></a>    micro_wb &lt;-<span class="st"> </span>micro_wb[, <span class="kw">colSums</span>(micro_wb <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>]</span>
<span id="cb20-43"><a href="inter-omics.html#cb20-43"></a>    micros &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">gb =</span> micro_gb, <span class="dt">sf =</span> micro_sf, <span class="dt">wb =</span> micro_wb)</span>
<span id="cb20-44"><a href="inter-omics.html#cb20-44"></a>    <span class="kw">return</span>(micros)</span>
<span id="cb20-45"><a href="inter-omics.html#cb20-45"></a>}</span>
<span id="cb20-46"><a href="inter-omics.html#cb20-46"></a></span>
<span id="cb20-47"><a href="inter-omics.html#cb20-47"></a>micro_ds &lt;-<span class="st"> </span><span class="kw">OTU_prep_sqrt</span>(micro)</span>
<span id="cb20-48"><a href="inter-omics.html#cb20-48"></a></span>
<span id="cb20-49"><a href="inter-omics.html#cb20-49"></a><span class="co"># calculating overall relative abundance of each OTU per sample</span></span>
<span id="cb20-50"><a href="inter-omics.html#cb20-50"></a>overall_rabdc &lt;-<span class="st"> </span><span class="cf">function</span>(micros) {</span>
<span id="cb20-51"><a href="inter-omics.html#cb20-51"></a>    mic &lt;-<span class="st"> </span>micros</span>
<span id="cb20-52"><a href="inter-omics.html#cb20-52"></a>    n &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb20-53"><a href="inter-omics.html#cb20-53"></a>    k &lt;-<span class="st"> </span><span class="kw">dim</span>(mic)[<span class="dv">1</span>]</span>
<span id="cb20-54"><a href="inter-omics.html#cb20-54"></a>    mic[<span class="st">&quot;rowsum&quot;</span>] &lt;-<span class="st"> </span><span class="kw">apply</span>(mic, <span class="dv">1</span>, sum)</span>
<span id="cb20-55"><a href="inter-omics.html#cb20-55"></a>    </span>
<span id="cb20-56"><a href="inter-omics.html#cb20-56"></a>    <span class="cf">while</span> (n <span class="op">&lt;</span><span class="st"> </span>k) {</span>
<span id="cb20-57"><a href="inter-omics.html#cb20-57"></a>        n &lt;-<span class="st"> </span>n <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb20-58"><a href="inter-omics.html#cb20-58"></a>        mic[n, ] &lt;-<span class="st"> </span>mic[n, ]<span class="op">/</span>(mic<span class="op">$</span>rowsum[n])</span>
<span id="cb20-59"><a href="inter-omics.html#cb20-59"></a>    }</span>
<span id="cb20-60"><a href="inter-omics.html#cb20-60"></a>    </span>
<span id="cb20-61"><a href="inter-omics.html#cb20-61"></a>    mic<span class="op">$</span>rowsum &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb20-62"><a href="inter-omics.html#cb20-62"></a>    mic &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">t</span>(mic))</span>
<span id="cb20-63"><a href="inter-omics.html#cb20-63"></a>    <span class="co"># mic[&#39;avg_rel_abdc&#39;] &lt;- apply(mic, 1, mean) mic[&#39;occurrence&#39;] &lt;-</span></span>
<span id="cb20-64"><a href="inter-omics.html#cb20-64"></a>    <span class="co"># ifelse(mic$avg&gt;0.0025, &#39;common&#39;, &#39;rare&#39;)</span></span>
<span id="cb20-65"><a href="inter-omics.html#cb20-65"></a>    <span class="kw">return</span>(mic)</span>
<span id="cb20-66"><a href="inter-omics.html#cb20-66"></a>}</span>
<span id="cb20-67"><a href="inter-omics.html#cb20-67"></a></span>
<span id="cb20-68"><a href="inter-omics.html#cb20-68"></a>rabdc &lt;-<span class="st"> </span><span class="kw">lapply</span>(micro_ds, overall_rabdc)</span>
<span id="cb20-69"><a href="inter-omics.html#cb20-69"></a></span>
<span id="cb20-70"><a href="inter-omics.html#cb20-70"></a><span class="co"># preparing congruent data sets</span></span>
<span id="cb20-71"><a href="inter-omics.html#cb20-71"></a>common_samples &lt;-<span class="st"> </span><span class="kw">intersect</span>(<span class="kw">colnames</span>(rabdc<span class="op">$</span>gb), cmp<span class="op">$</span>unified_ID)</span>
<span id="cb20-72"><a href="inter-omics.html#cb20-72"></a>rabdc &lt;-<span class="st"> </span>rabdc<span class="op">$</span>gb[, <span class="kw">colnames</span>(rabdc<span class="op">$</span>gb) <span class="op">%in%</span><span class="st"> </span>common_samples]</span>
<span id="cb20-73"><a href="inter-omics.html#cb20-73"></a>rabdc &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">t</span>(rabdc))</span>
<span id="cb20-74"><a href="inter-omics.html#cb20-74"></a></span>
<span id="cb20-75"><a href="inter-omics.html#cb20-75"></a>cmp &lt;-<span class="st"> </span>cmp[cmp<span class="op">$</span>unified_ID <span class="op">%in%</span><span class="st"> </span>common_samples, ]</span>
<span id="cb20-76"><a href="inter-omics.html#cb20-76"></a>cmp[] &lt;-<span class="st"> </span><span class="kw">lapply</span>(cmp, <span class="cf">function</span>(x) <span class="cf">if</span> (<span class="kw">is.factor</span>(x)) <span class="kw">factor</span>(x) <span class="cf">else</span> x)</span>
<span id="cb20-77"><a href="inter-omics.html#cb20-77"></a>cmp &lt;-<span class="st"> </span>cmp[<span class="kw">order</span>((cmp<span class="op">$</span>unified_ID)), ]</span>
<span id="cb20-78"><a href="inter-omics.html#cb20-78"></a></span>
<span id="cb20-79"><a href="inter-omics.html#cb20-79"></a><span class="co"># all(rownames(rabdc)==cmp$unified_ID)</span></span>
<span id="cb20-80"><a href="inter-omics.html#cb20-80"></a></span>
<span id="cb20-81"><a href="inter-omics.html#cb20-81"></a><span class="co"># CORRELATION for Gb and Sf</span></span>
<span id="cb20-82"><a href="inter-omics.html#cb20-82"></a></span>
<span id="cb20-83"><a href="inter-omics.html#cb20-83"></a>bar_cor &lt;-<span class="st"> </span><span class="cf">function</span>(rabdc_df, cmp) {</span>
<span id="cb20-84"><a href="inter-omics.html#cb20-84"></a>    barettin &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">colnames</span>(rabdc_df))</span>
<span id="cb20-85"><a href="inter-omics.html#cb20-85"></a>    <span class="kw">colnames</span>(barettin) &lt;-<span class="st"> &quot;XOTU&quot;</span></span>
<span id="cb20-86"><a href="inter-omics.html#cb20-86"></a>    barettin[<span class="st">&quot;barettin_estimate_P&quot;</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></span>
<span id="cb20-87"><a href="inter-omics.html#cb20-87"></a>    barettin[<span class="st">&quot;barettin_p_val_P&quot;</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></span>
<span id="cb20-88"><a href="inter-omics.html#cb20-88"></a>    barettin[<span class="st">&quot;barettin_estimate_S&quot;</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></span>
<span id="cb20-89"><a href="inter-omics.html#cb20-89"></a>    barettin[<span class="st">&quot;barettin_p_val_S&quot;</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></span>
<span id="cb20-90"><a href="inter-omics.html#cb20-90"></a>    </span>
<span id="cb20-91"><a href="inter-omics.html#cb20-91"></a>    n &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb20-92"><a href="inter-omics.html#cb20-92"></a>    k &lt;-<span class="st"> </span><span class="kw">dim</span>(barettin)[<span class="dv">1</span>]</span>
<span id="cb20-93"><a href="inter-omics.html#cb20-93"></a>    <span class="cf">while</span> (n <span class="op">&lt;</span><span class="st"> </span>k) {</span>
<span id="cb20-94"><a href="inter-omics.html#cb20-94"></a>        n &lt;-<span class="st"> </span>n <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb20-95"><a href="inter-omics.html#cb20-95"></a>        barettin<span class="op">$</span>barettin_estimate_P[n] &lt;-<span class="st"> </span><span class="kw">cor.test</span>(<span class="kw">as.numeric</span>(rabdc_df[, n]), cmp<span class="op">$</span>bar, </span>
<span id="cb20-96"><a href="inter-omics.html#cb20-96"></a>            <span class="dt">method =</span> <span class="st">&quot;pearson&quot;</span>)<span class="op">$</span>estimate</span>
<span id="cb20-97"><a href="inter-omics.html#cb20-97"></a>        barettin<span class="op">$</span>barettin_p_val_P[n] &lt;-<span class="st"> </span><span class="kw">cor.test</span>(<span class="kw">as.numeric</span>(rabdc_df[, n]), cmp<span class="op">$</span>bar, </span>
<span id="cb20-98"><a href="inter-omics.html#cb20-98"></a>            <span class="dt">method =</span> <span class="st">&quot;pearson&quot;</span>)<span class="op">$</span>p.value</span>
<span id="cb20-99"><a href="inter-omics.html#cb20-99"></a>        barettin<span class="op">$</span>barettin_estimate_S[n] &lt;-<span class="st"> </span><span class="kw">cor.test</span>(<span class="kw">as.numeric</span>(rabdc_df[, n]), cmp<span class="op">$</span>bar, </span>
<span id="cb20-100"><a href="inter-omics.html#cb20-100"></a>            <span class="dt">method =</span> <span class="st">&quot;spearman&quot;</span>)<span class="op">$</span>estimate</span>
<span id="cb20-101"><a href="inter-omics.html#cb20-101"></a>        barettin<span class="op">$</span>barettin_p_val_S[n] &lt;-<span class="st"> </span><span class="kw">cor.test</span>(<span class="kw">as.numeric</span>(rabdc_df[, n]), cmp<span class="op">$</span>bar, </span>
<span id="cb20-102"><a href="inter-omics.html#cb20-102"></a>            <span class="dt">method =</span> <span class="st">&quot;spearman&quot;</span>)<span class="op">$</span>p.value</span>
<span id="cb20-103"><a href="inter-omics.html#cb20-103"></a>    }</span>
<span id="cb20-104"><a href="inter-omics.html#cb20-104"></a>    </span>
<span id="cb20-105"><a href="inter-omics.html#cb20-105"></a>    barettin[<span class="st">&quot;barettin_fdr_P&quot;</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></span>
<span id="cb20-106"><a href="inter-omics.html#cb20-106"></a>    barettin[<span class="st">&quot;barettin_fdr_S&quot;</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></span>
<span id="cb20-107"><a href="inter-omics.html#cb20-107"></a>    barettin<span class="op">$</span>barettin_fdr_P &lt;-<span class="st"> </span><span class="kw">p.adjust</span>(barettin<span class="op">$</span>barettin_p_val_P, <span class="dt">method =</span> <span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb20-108"><a href="inter-omics.html#cb20-108"></a>    barettin<span class="op">$</span>barettin_fdr_S &lt;-<span class="st"> </span><span class="kw">p.adjust</span>(barettin<span class="op">$</span>barettin_p_val_S, <span class="dt">method =</span> <span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb20-109"><a href="inter-omics.html#cb20-109"></a>    <span class="kw">return</span>(barettin)</span>
<span id="cb20-110"><a href="inter-omics.html#cb20-110"></a>}</span>
<span id="cb20-111"><a href="inter-omics.html#cb20-111"></a></span>
<span id="cb20-112"><a href="inter-omics.html#cb20-112"></a>barettin_cor &lt;-<span class="st"> </span><span class="kw">bar_cor</span>(rabdc, cmp)</span>
<span id="cb20-113"><a href="inter-omics.html#cb20-113"></a></span>
<span id="cb20-114"><a href="inter-omics.html#cb20-114"></a><span class="co"># write.csv(barettin_cor, &#39;data/GB_OTU_barettin_correlation.csv&#39;, row.names = F)</span></span>
<span id="cb20-115"><a href="inter-omics.html#cb20-115"></a></span>
<span id="cb20-116"><a href="inter-omics.html#cb20-116"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>))</span>
<span id="cb20-117"><a href="inter-omics.html#cb20-117"></a><span class="kw">hist</span>(barettin_cor<span class="op">$</span>barettin_estimate_P, <span class="dt">main =</span> <span class="st">&quot;Pearson&#39;s pmcc: </span><span class="ch">\n</span><span class="st"> OTU rel abdc ~ barettin signal&quot;</span>, </span>
<span id="cb20-118"><a href="inter-omics.html#cb20-118"></a>    <span class="dt">xlab =</span> <span class="st">&quot;Product moment </span><span class="ch">\n</span><span class="st"> correlation coefficient&quot;</span>)</span>
<span id="cb20-119"><a href="inter-omics.html#cb20-119"></a><span class="kw">hist</span>(barettin_cor<span class="op">$</span>barettin_p_val_P, <span class="dt">breaks =</span> <span class="dv">20</span>, <span class="dt">main =</span> <span class="st">&quot;Pearson&#39;s pmcc p-value&quot;</span>, </span>
<span id="cb20-120"><a href="inter-omics.html#cb20-120"></a>    <span class="dt">xlab =</span> <span class="st">&quot;p-value&quot;</span>)</span>
<span id="cb20-121"><a href="inter-omics.html#cb20-121"></a><span class="kw">hist</span>(barettin_cor<span class="op">$</span>barettin_fdr_P, <span class="dt">breaks =</span> <span class="dv">20</span>, <span class="dt">main =</span> <span class="st">&quot;Pearson&#39;s pmcc </span><span class="ch">\n</span><span class="st"> fdr corrected p-value&quot;</span>, </span>
<span id="cb20-122"><a href="inter-omics.html#cb20-122"></a>    <span class="dt">xlab =</span> <span class="st">&quot;FDR corrected p-value&quot;</span>)</span></code></pre></div>
<p><img src="Oceanographic-setting-influences-the-prokaryote-community-and-metabolome-in-deep-sea-sponges_files/figure-html/barettin-correlation-1.png" width="672" /></p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="inter-omics.html#cb21-1"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb21-2"><a href="inter-omics.html#cb21-2"></a></span>
<span id="cb21-3"><a href="inter-omics.html#cb21-3"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>))</span>
<span id="cb21-4"><a href="inter-omics.html#cb21-4"></a><span class="kw">hist</span>(barettin_cor<span class="op">$</span>barettin_estimate_S, <span class="dt">main =</span> <span class="st">&quot;Spearman&#39;s rho: </span><span class="ch">\n</span><span class="st"> OTU rel abdc ~ barettin signal&quot;</span>, </span>
<span id="cb21-5"><a href="inter-omics.html#cb21-5"></a>    <span class="dt">xlab =</span> <span class="st">&quot;Rho&quot;</span>)</span>
<span id="cb21-6"><a href="inter-omics.html#cb21-6"></a><span class="kw">hist</span>(barettin_cor<span class="op">$</span>barettin_p_val_S, <span class="dt">breaks =</span> <span class="dv">20</span>, <span class="dt">main =</span> <span class="st">&quot;Spearman&#39;s rho p-value&quot;</span>, </span>
<span id="cb21-7"><a href="inter-omics.html#cb21-7"></a>    <span class="dt">xlab =</span> <span class="st">&quot;p-value&quot;</span>)</span>
<span id="cb21-8"><a href="inter-omics.html#cb21-8"></a><span class="kw">hist</span>(barettin_cor<span class="op">$</span>barettin_fdr_S, <span class="dt">breaks =</span> <span class="dv">20</span>, <span class="dt">main =</span> <span class="st">&quot;Spearman&#39;s rho </span><span class="ch">\n</span><span class="st"> fdr corrected p-value&quot;</span>, </span>
<span id="cb21-9"><a href="inter-omics.html#cb21-9"></a>    <span class="dt">xlab =</span> <span class="st">&quot;FDR corrected p-value&quot;</span>)</span></code></pre></div>
<p><img src="Oceanographic-setting-influences-the-prokaryote-community-and-metabolome-in-deep-sea-sponges_files/figure-html/barettin-correlation-2.png" width="672" /></p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="inter-omics.html#cb22-1"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))</span></code></pre></div>
</div>
<div id="microbial-interaction-network" class="section level2">
<h2><span class="header-section-number">4.3</span> Microbial interaction network</h2>
<div id="generating-network-based-on-different-algorithms" class="section level3">
<h3><span class="header-section-number">4.3.1</span> Generating network based on different algorithms</h3>
<p>The overall goal of the subsequent seqctions of code is to produce a microbial interation network for <em>Geodia barretti</em>. Nodes will be OTUs/ASVs from <em>Geodia barretti</em> samples and edges represent an interaction between those OTUs/ASVs. In this first part, we will employ different algorithms for network building. Network building algorithms are mannifold and their results not uncontroversial, thus the recommended strategy is to use different methods and merge the resulting networks to one consensus representation (Weiss <em>et al</em>., 2016), which we will do in the second part.</p>
<p>The original data set from 14 specimens of <em>Geodia barretti</em> contained 420 OTUs/ASVs. To reduce sparsity, we removed OTUs/ASVs with two or less non-zero values resulting in a data set containing 289 OTUs/ASVs. This data set was used for network inference with the following methods:</p>
<ol style="list-style-type: decimal">
<li>MENA Pipeline</li>
<li>fastLSA</li>
<li>SparCC</li>
<li>Maximal information coefficient MIC</li>
</ol>
<div id="molecular-ecological-network-analysis-mena-pipeline" class="section level4">
<h4><span class="header-section-number">4.3.1.1</span> Molecular Ecological Network Analysis (MENA) Pipeline</h4>
<p>The implementation of MENA (Deng <em>et al</em>., 2012; Zhou, Deng <em>et al</em>., 2010; Zhou, Deng <em>et al</em>.,2011) can be accessed at <a href="http://ieg4.rccc.ou.edu/mena" class="uri">http://ieg4.rccc.ou.edu/mena</a>. The data set was saved as tab separated values and all zeros were converted to blanks.
No further filtering for non-zero values was done (more than two non-zero values). For data preparation, default settings were applied, i.e. missing data was only filled with 0.01 in blanks with paired valid values, logarithm was taken, Pearson correlation coefficient was selected. Likewise, Random matrix theory settings were kept at defaults, decreasing the cutoff from the top using Regress Poisson distribution only. The cutoff of 0.800 was chosen for the similarity matrix to construct the network, corresponding to a Chi-square test on Poisson distribution of 99.191 and a p-value of 0.001. This resulted in a network with 241 nodes and 3582 edges.</p>
<p>A second analysis was produced with the same settings except building the similarity matrix based on Spearman’s Rho. The cutoff of 0.820 was chosen for the similarity matrix to construct the network, corresponding to a Chi-square test on Poisson distribution of 98.417 and a p-value of 0.001. This resulted in a network with 252 nodes and 2216 edges.</p>
<p>Network properties and parameters are summarized in MENA_network_parameters_Feb2019.xlsx.</p>
</div>
<div id="local-similarity-analysis-fastlsa" class="section level4">
<h4><span class="header-section-number">4.3.1.2</span> Local Similarity Analysis: fastLSA</h4>
<p>The command line program for calcularing local similarity (Durno <em>et al</em>., 2013) was downloaded from <a href="http://hallam.microbiology.ubc.ca/fastLSA/install/index.html" class="uri">http://hallam.microbiology.ubc.ca/fastLSA/install/index.html</a> and run specifying the input file, no time lag (-d 0) and significance level alpha (-a 0.05). All other paramters were kept at their default values. The input data set was a tab delimited text file stripped of OTU labels or sample IDs.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="inter-omics.html#cb23-1"></a>$ <span class="ex">./fastLSA</span> -i ../gb_289_feb2019.txt -d 0 -a 0.05 -o ../gb_289_feb2019.out</span></code></pre></div>
<p>The output file, as specified on the website, containes five columns. ‘index1’ and ‘index2’ represent the significant paired indices ranging from 0 to n-1 (OTUs/ASVs). LSA denotes the LSA statistic of each pair, lag was set to 0 with the -d flag and the p-valueBound column provides the p-value’s upper boundary for the significantly paired p-value.</p>
<p>To produce comparable data sets, we replaced the indices with their OTU IDs and removed superfluous columns.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="inter-omics.html#cb24-1"></a>fastLSA &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/gb_289_feb2019.out&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb24-2"><a href="inter-omics.html#cb24-2"></a>key &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/fastLSA_index_otu_CORRECTEDfeb2019.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb24-3"><a href="inter-omics.html#cb24-3"></a></span>
<span id="cb24-4"><a href="inter-omics.html#cb24-4"></a>fastLSA<span class="op">$</span>index1_otu &lt;-<span class="st"> </span>key<span class="op">$</span>fastLSA_OTU[<span class="kw">match</span>(fastLSA<span class="op">$</span>index1, key<span class="op">$</span>fastLSA_index)]</span>
<span id="cb24-5"><a href="inter-omics.html#cb24-5"></a>fastLSA<span class="op">$</span>index2_otu &lt;-<span class="st"> </span>key<span class="op">$</span>fastLSA_OTU[<span class="kw">match</span>(fastLSA<span class="op">$</span>index2, key<span class="op">$</span>fastLSA_index)]</span>
<span id="cb24-6"><a href="inter-omics.html#cb24-6"></a></span>
<span id="cb24-7"><a href="inter-omics.html#cb24-7"></a>fastLSA<span class="op">$</span>index1 &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb24-8"><a href="inter-omics.html#cb24-8"></a>fastLSA<span class="op">$</span>index2 &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb24-9"><a href="inter-omics.html#cb24-9"></a>fastLSA<span class="op">$</span>lag &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb24-10"><a href="inter-omics.html#cb24-10"></a>fastLSA<span class="op">$</span>X &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb24-11"><a href="inter-omics.html#cb24-11"></a>fastLSA &lt;-<span class="st"> </span>fastLSA[, <span class="kw">c</span>(<span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">1</span>, <span class="dv">2</span>)]</span>
<span id="cb24-12"><a href="inter-omics.html#cb24-12"></a>fastLSA &lt;-<span class="st"> </span>fastLSA[<span class="kw">order</span>(fastLSA<span class="op">$</span>p.valueBound), ]</span>
<span id="cb24-13"><a href="inter-omics.html#cb24-13"></a></span>
<span id="cb24-14"><a href="inter-omics.html#cb24-14"></a><span class="co"># write.csv(fastLSA, &#39;fastLSA_for_networks.csv&#39;)</span></span>
<span id="cb24-15"><a href="inter-omics.html#cb24-15"></a><span class="kw">rm</span>(key)</span></code></pre></div>
<p>LSA scores range from -1 for strong negatively correlations to 1, for strong positive correlations. There were no negative correlations in this data setand we refrained from scaleding the LSA score furhter.
The resulting network contrained 129 nodes and 207 edges.</p>
</div>
<div id="sparcc" class="section level4">
<h4><span class="header-section-number">4.3.1.3</span> SparCC</h4>
<p>SparCC (Friedman and Alm, 2012) is a network building algorithm for compositional data and can be found at <a href="https://bitbucket.org/yonatanf/sparcc" class="uri">https://bitbucket.org/yonatanf/sparcc</a>.</p>
<p>Prior to running it I had to get help as there was a minor issue during compilation. SparCC needed specific versions of numpy, panda and python to run properly, which is easiest accomodated in a specific environment. The OTU table needs to be windows formatted text. The embedded code is an example, for the analysis, 500 iterations were combined.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb25-1"><a href="inter-omics.html#cb25-1"></a>$ <span class="bu">cd</span> to working directory with SparCC and the data set gb_289.csv</span>
<span id="cb25-2"><a href="inter-omics.html#cb25-2"></a>$ <span class="bu">source</span> activate sparcc</span>
<span id="cb25-3"><a href="inter-omics.html#cb25-3"></a><span class="ex">python</span> SparCC.py ../gb_sparcc.txt -c ../gb_sparcc_cor_file.txt -v </span>
<span id="cb25-4"><a href="inter-omics.html#cb25-4"></a><span class="ex">../gb_sparcc_coverage_file.txt</span> -i 5</span>
<span id="cb25-5"><a href="inter-omics.html#cb25-5"></a>$ <span class="ex">deactivate</span></span></code></pre></div>
<p>Create a results directory and redirect all the output there. Pseudo p-value Calculation, generates -n shuffled data sets:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb26-1"><a href="inter-omics.html#cb26-1"></a>$ <span class="fu">mkdir</span> results <span class="co">#creates output directory</span></span>
<span id="cb26-2"><a href="inter-omics.html#cb26-2"></a>$ <span class="ex">python</span> MakeBootstraps.py ../gb_sparcc.txt -n 5 -t permutation_#.txt -p ../results/ </span></code></pre></div>
<p>And run SparCC.py on all the re-shuffled data sets:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb27-1"><a href="inter-omics.html#cb27-1"></a>$ <span class="ex">python</span> SparCC.py ../results/permutation_0.txt -i 5 --cor_file=../results/perm_cor_0.txt</span>
<span id="cb27-2"><a href="inter-omics.html#cb27-2"></a>$ <span class="ex">python</span> SparCC.py ../results/permutation_1.txt -i 5 --cor_file=../results/perm_cor_1.txt</span>
<span id="cb27-3"><a href="inter-omics.html#cb27-3"></a>$ <span class="ex">python</span> SparCC.py ../results/permutation_2.txt -i 5 --cor_file=../results/perm_cor_2.txt</span>
<span id="cb27-4"><a href="inter-omics.html#cb27-4"></a>$ <span class="ex">python</span> SparCC.py ../results/permutation_3.txt -i 5 --cor_file=../results/perm_cor_3.txt</span>
<span id="cb27-5"><a href="inter-omics.html#cb27-5"></a>$ <span class="ex">python</span> SparCC.py ../results/permutation_4.txt -i 5 --cor_file=../results/perm_cor_4.txt</span></code></pre></div>
<p>Generate p-values:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb28-1"><a href="inter-omics.html#cb28-1"></a>$ <span class="ex">python</span> PseudoPvals.py ../results/gb_sparcc_cor_file.txt ../results/perm_cor_#.txt 5 </span>
<span id="cb28-2"><a href="inter-omics.html#cb28-2"></a><span class="ex">-o</span> ../results/pvals.two_sided.txt -t two_sided</span></code></pre></div>
<p>Formatting the resulting data set like so:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="inter-omics.html#cb29-1"></a><span class="kw">library</span>(reshape2)</span>
<span id="cb29-2"><a href="inter-omics.html#cb29-2"></a>cor_file &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/gb_sparcc_cor_file_289.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb29-3"><a href="inter-omics.html#cb29-3"></a>p_vals &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/gb_289_pvals.two_sided.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb29-4"><a href="inter-omics.html#cb29-4"></a></span>
<span id="cb29-5"><a href="inter-omics.html#cb29-5"></a><span class="co"># make OTU ID the rowname</span></span>
<span id="cb29-6"><a href="inter-omics.html#cb29-6"></a><span class="kw">rownames</span>(cor_file) &lt;-<span class="st"> </span>cor_file[, <span class="dv">1</span>]</span>
<span id="cb29-7"><a href="inter-omics.html#cb29-7"></a>cor_file[, <span class="dv">1</span>] &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb29-8"><a href="inter-omics.html#cb29-8"></a><span class="kw">rownames</span>(p_vals) &lt;-<span class="st"> </span>p_vals[, <span class="dv">1</span>]</span>
<span id="cb29-9"><a href="inter-omics.html#cb29-9"></a>p_vals[, <span class="dv">1</span>] &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb29-10"><a href="inter-omics.html#cb29-10"></a><span class="co"># check ds congruency</span></span>
<span id="cb29-11"><a href="inter-omics.html#cb29-11"></a><span class="kw">all</span>(<span class="kw">colnames</span>(cor_file) <span class="op">==</span><span class="st"> </span><span class="kw">rownames</span>(cor_file))</span>
<span id="cb29-12"><a href="inter-omics.html#cb29-12"></a><span class="kw">all</span>(<span class="kw">colnames</span>(p_vals) <span class="op">==</span><span class="st"> </span><span class="kw">rownames</span>(p_vals))</span>
<span id="cb29-13"><a href="inter-omics.html#cb29-13"></a></span>
<span id="cb29-14"><a href="inter-omics.html#cb29-14"></a><span class="co"># melt into long format, all vs all comparison: 289^2=83521 rows</span></span>
<span id="cb29-15"><a href="inter-omics.html#cb29-15"></a>cor_file_m &lt;-<span class="st"> </span><span class="kw">melt</span>(<span class="kw">as.matrix</span>(cor_file))</span>
<span id="cb29-16"><a href="inter-omics.html#cb29-16"></a>p_vals_m &lt;-<span class="st"> </span><span class="kw">melt</span>(<span class="kw">as.matrix</span>(p_vals))</span>
<span id="cb29-17"><a href="inter-omics.html#cb29-17"></a></span>
<span id="cb29-18"><a href="inter-omics.html#cb29-18"></a><span class="kw">all</span>(cor_file_m<span class="op">$</span>Var1 <span class="op">==</span><span class="st"> </span>p_vals_m<span class="op">$</span>Var1)</span>
<span id="cb29-19"><a href="inter-omics.html#cb29-19"></a><span class="kw">all</span>(cor_file_m<span class="op">$</span>Var2 <span class="op">==</span><span class="st"> </span>p_vals_m<span class="op">$</span>Var2)</span>
<span id="cb29-20"><a href="inter-omics.html#cb29-20"></a></span>
<span id="cb29-21"><a href="inter-omics.html#cb29-21"></a><span class="co"># complete data set with p_vals and &#39;correlation coeff&#39;</span></span>
<span id="cb29-22"><a href="inter-omics.html#cb29-22"></a>cor_file_m[<span class="st">&quot;p_vals&quot;</span>] &lt;-<span class="st"> </span>p_vals_m<span class="op">$</span>value</span>
<span id="cb29-23"><a href="inter-omics.html#cb29-23"></a></span>
<span id="cb29-24"><a href="inter-omics.html#cb29-24"></a><span class="co"># This removes AB - BA duplicates but still contains self comprisons, AA, BB,CC</span></span>
<span id="cb29-25"><a href="inter-omics.html#cb29-25"></a><span class="co"># etc.</span></span>
<span id="cb29-26"><a href="inter-omics.html#cb29-26"></a>cols &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Var1&quot;</span>, <span class="st">&quot;Var2&quot;</span>)</span>
<span id="cb29-27"><a href="inter-omics.html#cb29-27"></a>newdf &lt;-<span class="st"> </span>cor_file_m[, cols]  <span class="co">#generate new data set with just those two</span></span>
<span id="cb29-28"><a href="inter-omics.html#cb29-28"></a><span class="co"># a &lt;- Sys.time()</span></span>
<span id="cb29-29"><a href="inter-omics.html#cb29-29"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(cor_file_m)) {</span>
<span id="cb29-30"><a href="inter-omics.html#cb29-30"></a>    newdf[i, ] =<span class="st"> </span><span class="kw">sort</span>(cor_file_m[i, cols])</span>
<span id="cb29-31"><a href="inter-omics.html#cb29-31"></a>}</span>
<span id="cb29-32"><a href="inter-omics.html#cb29-32"></a><span class="co"># b &lt;- Sys.time() b-a</span></span>
<span id="cb29-33"><a href="inter-omics.html#cb29-33"></a></span>
<span id="cb29-34"><a href="inter-omics.html#cb29-34"></a>cor_file_shortened &lt;-<span class="st"> </span>cor_file_m[<span class="op">!</span><span class="kw">duplicated</span>(newdf), ]  <span class="co">#and can be removed with duplicate</span></span>
<span id="cb29-35"><a href="inter-omics.html#cb29-35"></a>cor_file_shortened &lt;-<span class="st"> </span>cor_file_shortened[<span class="kw">which</span>(cor_file_shortened<span class="op">$</span>Var1 <span class="op">!=</span><span class="st"> </span>cor_file_shortened<span class="op">$</span>Var2), </span>
<span id="cb29-36"><a href="inter-omics.html#cb29-36"></a>    ]  <span class="co"># removing self comparison</span></span>
<span id="cb29-37"><a href="inter-omics.html#cb29-37"></a><span class="kw">colnames</span>(cor_file_shortened) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Var1_SparCC&quot;</span>, <span class="st">&quot;Var2_SparCC&quot;</span>, <span class="st">&quot;SparCC&quot;</span>, <span class="st">&quot;pSparCC&quot;</span>)  <span class="co">#41616</span></span>
<span id="cb29-38"><a href="inter-omics.html#cb29-38"></a><span class="kw">write.csv</span>(cor_file_shortened, <span class="st">&quot;data/SparCC_for_networks.csv&quot;</span>)</span>
<span id="cb29-39"><a href="inter-omics.html#cb29-39"></a><span class="kw">rm</span>(cor_file_m, newdf, p_vals_m, i, cols)</span></code></pre></div>
</div>
<div id="maximal-information-coefficient-mic" class="section level4">
<h4><span class="header-section-number">4.3.1.4</span> Maximal information coefficient MIC</h4>
<p>MIC for pairwise interaction was calculated with the R package minearva. The MIC is part of a statistic called Maximal Information-Based Nonparametric Exploration (MINE).</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="inter-omics.html#cb30-1"></a><span class="kw">library</span>(minerva)</span>
<span id="cb30-2"><a href="inter-omics.html#cb30-2"></a></span>
<span id="cb30-3"><a href="inter-omics.html#cb30-3"></a>OTU &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/gb_289.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb30-4"><a href="inter-omics.html#cb30-4"></a><span class="kw">rownames</span>(OTU) &lt;-<span class="st"> </span>OTU[, <span class="dv">1</span>]</span>
<span id="cb30-5"><a href="inter-omics.html#cb30-5"></a>OTU[, <span class="dv">1</span>] &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb30-6"><a href="inter-omics.html#cb30-6"></a>OTU &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">t</span>(OTU))</span>
<span id="cb30-7"><a href="inter-omics.html#cb30-7"></a></span>
<span id="cb30-8"><a href="inter-omics.html#cb30-8"></a><span class="co"># Calculate MIC of original data set.</span></span>
<span id="cb30-9"><a href="inter-omics.html#cb30-9"></a>MINE &lt;-<span class="st"> </span><span class="kw">mine</span>(OTU)</span>
<span id="cb30-10"><a href="inter-omics.html#cb30-10"></a>MIC &lt;-<span class="st"> </span>MINE<span class="op">$</span>MIC  <span class="co">#dim(MIC): 289 289</span></span></code></pre></div>
<p>Obtaining p-values for this statistic can be achieved by permutation of the original OTU table as below or empirically, by selecting the thousand strongest interactions.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="inter-omics.html#cb31-1"></a><span class="co"># 10 needs to be replaced with 1000 for final version, three times!!!</span></span>
<span id="cb31-2"><a href="inter-omics.html#cb31-2"></a><span class="co"># reshuffling the OTU table, saving the MIC to a list, a total of 1000 times</span></span>
<span id="cb31-3"><a href="inter-omics.html#cb31-3"></a>n &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb31-4"><a href="inter-omics.html#cb31-4"></a>results &lt;-<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb31-5"><a href="inter-omics.html#cb31-5"></a></span>
<span id="cb31-6"><a href="inter-omics.html#cb31-6"></a><span class="co"># For reproducibility, one could e.g.: set.seed(1984)</span></span>
<span id="cb31-7"><a href="inter-omics.html#cb31-7"></a></span>
<span id="cb31-8"><a href="inter-omics.html#cb31-8"></a><span class="co"># c &lt;- Sys.time()</span></span>
<span id="cb31-9"><a href="inter-omics.html#cb31-9"></a><span class="cf">while</span> (n <span class="op">&lt;</span><span class="st"> </span><span class="dv">1000</span>) {</span>
<span id="cb31-10"><a href="inter-omics.html#cb31-10"></a>    n &lt;-<span class="st"> </span>n <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb31-11"><a href="inter-omics.html#cb31-11"></a>    mock &lt;-<span class="st"> </span><span class="kw">apply</span>(OTU, <span class="dt">MARGIN =</span> <span class="dv">2</span>, sample)</span>
<span id="cb31-12"><a href="inter-omics.html#cb31-12"></a>    mock_mine &lt;-<span class="st"> </span><span class="kw">mine</span>(mock)</span>
<span id="cb31-13"><a href="inter-omics.html#cb31-13"></a>    results[[n]] &lt;-<span class="st"> </span>mock_mine<span class="op">$</span>MIC</span>
<span id="cb31-14"><a href="inter-omics.html#cb31-14"></a>}</span>
<span id="cb31-15"><a href="inter-omics.html#cb31-15"></a><span class="co"># d &lt;- Sys.time() d-c</span></span>
<span id="cb31-16"><a href="inter-omics.html#cb31-16"></a></span>
<span id="cb31-17"><a href="inter-omics.html#cb31-17"></a><span class="co"># for every element of the true matrix, go through all the same elements in the</span></span>
<span id="cb31-18"><a href="inter-omics.html#cb31-18"></a><span class="co"># 1000 generated mock matrix MIC indices and count how many of those are greater.</span></span>
<span id="cb31-19"><a href="inter-omics.html#cb31-19"></a>MIC &lt;-<span class="st"> </span>MINE<span class="op">$</span>MIC</span>
<span id="cb31-20"><a href="inter-omics.html#cb31-20"></a></span>
<span id="cb31-21"><a href="inter-omics.html#cb31-21"></a>e_values &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dt">nrow =</span> <span class="kw">nrow</span>(MIC), <span class="dt">ncol =</span> <span class="kw">ncol</span>(MIC), <span class="dt">data =</span> <span class="dv">0</span>)</span>
<span id="cb31-22"><a href="inter-omics.html#cb31-22"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(MIC)) {</span>
<span id="cb31-23"><a href="inter-omics.html#cb31-23"></a>    <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(MIC)) {</span>
<span id="cb31-24"><a href="inter-omics.html#cb31-24"></a>        n &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb31-25"><a href="inter-omics.html#cb31-25"></a>        <span class="cf">while</span> (n <span class="op">&lt;</span><span class="st"> </span><span class="dv">1000</span>) {</span>
<span id="cb31-26"><a href="inter-omics.html#cb31-26"></a>            n &lt;-<span class="st"> </span>n <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb31-27"><a href="inter-omics.html#cb31-27"></a>            <span class="cf">if</span> (results[[n]][i, j] <span class="op">&gt;=</span><span class="st"> </span>MIC[i, j]) {</span>
<span id="cb31-28"><a href="inter-omics.html#cb31-28"></a>                e_values[i, j] &lt;-<span class="st"> </span>e_values[i, j] <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb31-29"><a href="inter-omics.html#cb31-29"></a>            }</span>
<span id="cb31-30"><a href="inter-omics.html#cb31-30"></a>        }</span>
<span id="cb31-31"><a href="inter-omics.html#cb31-31"></a>    }</span>
<span id="cb31-32"><a href="inter-omics.html#cb31-32"></a>}</span>
<span id="cb31-33"><a href="inter-omics.html#cb31-33"></a></span>
<span id="cb31-34"><a href="inter-omics.html#cb31-34"></a>e_values &lt;-<span class="st"> </span>e_values<span class="op">/</span><span class="dv">1000</span></span>
<span id="cb31-35"><a href="inter-omics.html#cb31-35"></a><span class="co"># write.csv(e_values, &#39;data/MIC_e_values.csv&#39;)</span></span></code></pre></div>
<p>For n=1000, the first part takes about 8 mins on one core, the second part about 2 min. The relevant output is saved in the initial calculations of the MIC and the corresponding e-values are in e_values. These are symmetric matrices that will be reduced to a long table with unique OTUs/ASVs pairs, their MIC and the e-value. For clarity, all other files are removed.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="inter-omics.html#cb32-1"></a><span class="kw">rm</span>(MINE, mock, mock_mine, results)</span>
<span id="cb32-2"><a href="inter-omics.html#cb32-2"></a></span>
<span id="cb32-3"><a href="inter-omics.html#cb32-3"></a><span class="co"># transforming symmetric matrix to unique-pair long format</span></span>
<span id="cb32-4"><a href="inter-omics.html#cb32-4"></a>cor_file &lt;-<span class="st"> </span><span class="kw">data.frame</span>(MIC)</span>
<span id="cb32-5"><a href="inter-omics.html#cb32-5"></a>p_vals &lt;-<span class="st"> </span><span class="kw">data.frame</span>(e_values)</span>
<span id="cb32-6"><a href="inter-omics.html#cb32-6"></a></span>
<span id="cb32-7"><a href="inter-omics.html#cb32-7"></a><span class="co"># inspect the files, adapt them and test congruency</span></span>
<span id="cb32-8"><a href="inter-omics.html#cb32-8"></a><span class="kw">rownames</span>(p_vals) &lt;-<span class="st"> </span><span class="kw">rownames</span>(cor_file)</span>
<span id="cb32-9"><a href="inter-omics.html#cb32-9"></a><span class="kw">colnames</span>(p_vals) &lt;-<span class="st"> </span><span class="kw">colnames</span>(cor_file)</span>
<span id="cb32-10"><a href="inter-omics.html#cb32-10"></a><span class="kw">all</span>(<span class="kw">colnames</span>(cor_file) <span class="op">==</span><span class="st"> </span><span class="kw">rownames</span>(cor_file))</span>
<span id="cb32-11"><a href="inter-omics.html#cb32-11"></a></span>
<span id="cb32-12"><a href="inter-omics.html#cb32-12"></a><span class="co"># melt into long format, all vs all comparison: 289*289=83521 rows</span></span>
<span id="cb32-13"><a href="inter-omics.html#cb32-13"></a>cor_file_m &lt;-<span class="st"> </span><span class="kw">melt</span>(<span class="kw">as.matrix</span>(cor_file))</span>
<span id="cb32-14"><a href="inter-omics.html#cb32-14"></a><span class="kw">all</span>(<span class="kw">colnames</span>(p_vals) <span class="op">==</span><span class="st"> </span><span class="kw">rownames</span>(p_vals))</span>
<span id="cb32-15"><a href="inter-omics.html#cb32-15"></a>p_vals_m &lt;-<span class="st"> </span><span class="kw">melt</span>(<span class="kw">as.matrix</span>(p_vals))</span>
<span id="cb32-16"><a href="inter-omics.html#cb32-16"></a></span>
<span id="cb32-17"><a href="inter-omics.html#cb32-17"></a><span class="co"># complete data set with p_vals</span></span>
<span id="cb32-18"><a href="inter-omics.html#cb32-18"></a>cor_file_m[<span class="st">&quot;p_vals&quot;</span>] &lt;-<span class="st"> </span>p_vals_m<span class="op">$</span>value</span>
<span id="cb32-19"><a href="inter-omics.html#cb32-19"></a></span>
<span id="cb32-20"><a href="inter-omics.html#cb32-20"></a><span class="co"># This removes AB - BA duplicates but still contains self comprisons, AA, BB,</span></span>
<span id="cb32-21"><a href="inter-omics.html#cb32-21"></a><span class="co"># etc.</span></span>
<span id="cb32-22"><a href="inter-omics.html#cb32-22"></a>cols &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Var1&quot;</span>, <span class="st">&quot;Var2&quot;</span>)</span>
<span id="cb32-23"><a href="inter-omics.html#cb32-23"></a>newdf &lt;-<span class="st"> </span>cor_file_m[, cols]  <span class="co">#generate new data set with just those two</span></span>
<span id="cb32-24"><a href="inter-omics.html#cb32-24"></a></span>
<span id="cb32-25"><a href="inter-omics.html#cb32-25"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(cor_file_m)) {</span>
<span id="cb32-26"><a href="inter-omics.html#cb32-26"></a>    newdf[i, ] &lt;-<span class="st"> </span><span class="kw">sort</span>(cor_file_m[i, cols])</span>
<span id="cb32-27"><a href="inter-omics.html#cb32-27"></a>}</span>
<span id="cb32-28"><a href="inter-omics.html#cb32-28"></a></span>
<span id="cb32-29"><a href="inter-omics.html#cb32-29"></a>cor_file_shortened &lt;-<span class="st"> </span>cor_file_m[<span class="op">!</span><span class="kw">duplicated</span>(newdf), ]  <span class="co">#and can be removed with duplicate</span></span>
<span id="cb32-30"><a href="inter-omics.html#cb32-30"></a>cor_file_shortened &lt;-<span class="st"> </span>cor_file_shortened[<span class="kw">which</span>(cor_file_shortened<span class="op">$</span>Var1 <span class="op">!=</span><span class="st"> </span>cor_file_shortened<span class="op">$</span>Var2), </span>
<span id="cb32-31"><a href="inter-omics.html#cb32-31"></a>    ]  <span class="co"># removing self comparison</span></span>
<span id="cb32-32"><a href="inter-omics.html#cb32-32"></a><span class="kw">rm</span>(cor_file_m, newdf, p_vals_m, i, cols)</span>
<span id="cb32-33"><a href="inter-omics.html#cb32-33"></a></span>
<span id="cb32-34"><a href="inter-omics.html#cb32-34"></a><span class="co"># write.csv(cor_file_shortened, &#39;data/MIC_for_networks.csv&#39;)</span></span></code></pre></div>
</div>
</div>
<div id="consolidation-of-the-different-networks" class="section level3">
<h3><span class="header-section-number">4.3.2</span> Consolidation of the different networks</h3>
<div id="mic" class="section level4">
<h4><span class="header-section-number">4.3.2.1</span> MIC</h4>
<p>MIC allows to detect a variety of interactions. According to the manual of the R wrapper minerva, the resulting MIC score “is related to the relationship strenght and it can be interpreted as a correlation measure. It is symmetric and it ranges in [0,1], where it tends to 0 for statistically independent data and it approaches 1 in probability for noiseless functional relationships”. Thus, it also contains strong negative relationship up to mutual exclusivity, which we want to filter out.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="inter-omics.html#cb33-1"></a>mic &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/MIC_for_networks.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb33-2"><a href="inter-omics.html#cb33-2"></a>mic<span class="op">$</span>X &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb33-3"><a href="inter-omics.html#cb33-3"></a><span class="kw">colnames</span>(mic) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;node1_mic&quot;</span>, <span class="st">&quot;node2_mic&quot;</span>, <span class="st">&quot;MIC&quot;</span>, <span class="st">&quot;pMIC&quot;</span>)</span></code></pre></div>
<p>Initially, the MIC network generated by the R wrapper minerva contained MIC values for all possible edges (i.e. 41616). Of those, 4370 edges/interactions had a p-value <span class="math inline">\(\leq\)</span> 0.05. As we will only include those edges in the final network, we select those and calculate the linear regression coefficient and p-value for the regression, to test whether we are able to distinguish negative from positive interaction.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="inter-omics.html#cb34-1"></a><span class="co"># Original OTU table for regressions</span></span>
<span id="cb34-2"><a href="inter-omics.html#cb34-2"></a>OTU &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/gb_289.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb34-3"><a href="inter-omics.html#cb34-3"></a><span class="kw">rownames</span>(OTU) &lt;-<span class="st"> </span>OTU[, <span class="dv">1</span>]</span>
<span id="cb34-4"><a href="inter-omics.html#cb34-4"></a>OTU[, <span class="dv">1</span>] &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb34-5"><a href="inter-omics.html#cb34-5"></a>OTU[<span class="st">&quot;ID&quot;</span>] &lt;-<span class="st"> </span><span class="kw">row.names</span>(OTU)</span></code></pre></div>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="inter-omics.html#cb35-1"></a><span class="co"># goal: in mic data frame, set to &#39;NA&#39; MIC and pMIC of edges with a significant</span></span>
<span id="cb35-2"><a href="inter-omics.html#cb35-2"></a><span class="co"># p-value for MIC that have a significant negative regression</span></span>
<span id="cb35-3"><a href="inter-omics.html#cb35-3"></a></span>
<span id="cb35-4"><a href="inter-omics.html#cb35-4"></a>mic[<span class="st">&quot;regression&quot;</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></span>
<span id="cb35-5"><a href="inter-omics.html#cb35-5"></a>mic[<span class="st">&quot;p_regression&quot;</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></span>
<span id="cb35-6"><a href="inter-omics.html#cb35-6"></a></span>
<span id="cb35-7"><a href="inter-omics.html#cb35-7"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(mic)) {</span>
<span id="cb35-8"><a href="inter-omics.html#cb35-8"></a>    bac1 &lt;-<span class="st"> </span><span class="kw">factor</span>(mic[i, <span class="dv">1</span>])</span>
<span id="cb35-9"><a href="inter-omics.html#cb35-9"></a>    bac2 &lt;-<span class="st"> </span><span class="kw">factor</span>(mic[i, <span class="dv">2</span>])</span>
<span id="cb35-10"><a href="inter-omics.html#cb35-10"></a>    temp_ds &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">t</span>(<span class="kw">rbind</span>(OTU[OTU<span class="op">$</span>ID <span class="op">==</span><span class="st"> </span>bac1, ], OTU[OTU<span class="op">$</span>ID <span class="op">==</span><span class="st"> </span>bac2, ])))</span>
<span id="cb35-11"><a href="inter-omics.html#cb35-11"></a>    temp_ds &lt;-<span class="st"> </span>temp_ds[<span class="op">-</span><span class="kw">c</span>(<span class="dv">15</span>), ]</span>
<span id="cb35-12"><a href="inter-omics.html#cb35-12"></a>    temp_ds[] &lt;-<span class="st"> </span><span class="kw">lapply</span>(temp_ds, <span class="cf">function</span>(x) <span class="cf">if</span> (<span class="kw">is.factor</span>(x)) </span>
<span id="cb35-13"><a href="inter-omics.html#cb35-13"></a>        <span class="kw">factor</span>(x) <span class="cf">else</span> x)  <span class="co"># removes factors, not sure if necessary</span></span>
<span id="cb35-14"><a href="inter-omics.html#cb35-14"></a>    mic<span class="op">$</span>regression[i] &lt;-<span class="st"> </span><span class="kw">summary</span>(<span class="kw">lm</span>(<span class="kw">c</span>(temp_ds[, <span class="dv">1</span>]) <span class="op">~</span><span class="st"> </span><span class="kw">c</span>(temp_ds[, <span class="dv">2</span>])))<span class="op">$</span>coefficients[<span class="dv">2</span>, </span>
<span id="cb35-15"><a href="inter-omics.html#cb35-15"></a>        <span class="dv">1</span>]  <span class="co">#slope</span></span>
<span id="cb35-16"><a href="inter-omics.html#cb35-16"></a>    mic<span class="op">$</span>p_regression[i] &lt;-<span class="st"> </span><span class="kw">summary</span>(<span class="kw">lm</span>(<span class="kw">c</span>(temp_ds[, <span class="dv">1</span>]) <span class="op">~</span><span class="st"> </span><span class="kw">c</span>(temp_ds[, <span class="dv">2</span>])))<span class="op">$</span>coefficients[<span class="dv">2</span>, </span>
<span id="cb35-17"><a href="inter-omics.html#cb35-17"></a>        <span class="dv">4</span>]  <span class="co">#p-val</span></span>
<span id="cb35-18"><a href="inter-omics.html#cb35-18"></a>}</span>
<span id="cb35-19"><a href="inter-omics.html#cb35-19"></a></span>
<span id="cb35-20"><a href="inter-omics.html#cb35-20"></a>before &lt;-<span class="st"> </span><span class="kw">sum</span>(mic<span class="op">$</span>pMIC <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.05</span>)  <span class="co">#4370</span></span>
<span id="cb35-21"><a href="inter-omics.html#cb35-21"></a>mic<span class="op">$</span>MIC &lt;-<span class="st"> </span><span class="kw">ifelse</span>((mic<span class="op">$</span>pMIC <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>mic<span class="op">$</span>regression <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span> <span class="op">&amp;</span><span class="st"> </span>mic<span class="op">$</span>p_regression <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.05</span>), </span>
<span id="cb35-22"><a href="inter-omics.html#cb35-22"></a>    <span class="ot">NA</span>, mic<span class="op">$</span>MIC)</span>
<span id="cb35-23"><a href="inter-omics.html#cb35-23"></a>mic<span class="op">$</span>pMIC &lt;-<span class="st"> </span><span class="kw">ifelse</span>((mic<span class="op">$</span>pMIC <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>mic<span class="op">$</span>regression <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span> <span class="op">&amp;</span><span class="st"> </span>mic<span class="op">$</span>p_regression <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.05</span>), </span>
<span id="cb35-24"><a href="inter-omics.html#cb35-24"></a>    <span class="ot">NA</span>, mic<span class="op">$</span>pMIC)</span>
<span id="cb35-25"><a href="inter-omics.html#cb35-25"></a>after &lt;-<span class="st"> </span><span class="kw">sum</span>(mic<span class="op">$</span>pMIC <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.05</span>, <span class="dt">na.rm =</span> T)  <span class="co">#3522</span></span>
<span id="cb35-26"><a href="inter-omics.html#cb35-26"></a></span>
<span id="cb35-27"><a href="inter-omics.html#cb35-27"></a><span class="kw">write.csv</span>(mic, <span class="st">&quot;data/MIC.csv&quot;</span>)</span>
<span id="cb35-28"><a href="inter-omics.html#cb35-28"></a><span class="kw">rm</span>(temp_ds, bac1, bac2, i, mic)</span></code></pre></div>
<p>The MIC data set initially contained 41616 edges, 4370 of which were significant prior to the removal of negative correlations and leaving 3522 edges with a p-value <span class="math inline">\(\leq\)</span> 0.05.</p>
</div>
<div id="sparcc-1" class="section level4">
<h4><span class="header-section-number">4.3.2.2</span> SparCC</h4>
<p>The next network data set is based on the SparCC algorithm for computing correlations in compositional data.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="inter-omics.html#cb36-1"></a>sparcc &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/SparCC_for_networks.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb36-2"><a href="inter-omics.html#cb36-2"></a>sparcc<span class="op">$</span>X &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb36-3"><a href="inter-omics.html#cb36-3"></a><span class="kw">colnames</span>(sparcc) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;node1_sparcc&quot;</span>, <span class="st">&quot;node2_sparcc&quot;</span>, <span class="st">&quot;SparCC&quot;</span>, <span class="st">&quot;pSparCC&quot;</span>)</span>
<span id="cb36-4"><a href="inter-omics.html#cb36-4"></a><span class="co"># hist(sparcc$SparCC) hist(sparcc$pSparCC)</span></span>
<span id="cb36-5"><a href="inter-omics.html#cb36-5"></a>sparcc<span class="op">$</span>pSparCC &lt;-<span class="st"> </span><span class="kw">ifelse</span>((sparcc<span class="op">$</span>SparCC <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>), <span class="ot">NA</span>, sparcc<span class="op">$</span>pSparCC)  <span class="co">#setting the p-values of negative interactions to NA</span></span>
<span id="cb36-6"><a href="inter-omics.html#cb36-6"></a>sparcc<span class="op">$</span>SparCC &lt;-<span class="st"> </span><span class="kw">ifelse</span>((sparcc<span class="op">$</span>SparCC <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>), <span class="ot">NA</span>, sparcc<span class="op">$</span>SparCC)  <span class="co">#setting negative interactions to NA</span></span></code></pre></div>
<p>The network based on the SparCC algorithm contained 41616 edges of which 20420 negative interactions that were removed. 6622 significant positive edges remain.</p>
</div>
<div id="mena" class="section level4">
<h4><span class="header-section-number">4.3.2.3</span> MENA</h4>
<p>The next two network data sets are generated by MENA based on random matrix theory.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="inter-omics.html#cb37-1"></a>mena_pcc &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/MENA_0.800_PCC_edge_attribute.txt&quot;</span>, <span class="dt">header =</span> F, <span class="dt">sep =</span> <span class="st">&quot; &quot;</span>)</span>
<span id="cb37-2"><a href="inter-omics.html#cb37-2"></a><span class="co"># &#39;np&#39; in V2 and -1 in V5 mean negative interaction, these should be removed.</span></span>
<span id="cb37-3"><a href="inter-omics.html#cb37-3"></a><span class="kw">dim</span>(mena_pcc)[<span class="dv">1</span>] <span class="op">-</span><span class="st"> </span><span class="kw">dim</span>(mena_pcc[mena_pcc<span class="op">$</span>V5 <span class="op">==</span><span class="st"> </span><span class="dv">-1</span>, ])[<span class="dv">1</span>]  <span class="co"># Number of pos interactions</span></span></code></pre></div>
<pre><code>## [1] 171</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="inter-omics.html#cb39-1"></a>mena_pcc[<span class="st">&quot;pMENA_PCC&quot;</span>] &lt;-<span class="st"> </span><span class="kw">ifelse</span>((mena_pcc<span class="op">$</span>V5 <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>), <span class="ot">NA</span>, <span class="fl">0.001</span>)</span>
<span id="cb39-2"><a href="inter-omics.html#cb39-2"></a>mena_pcc<span class="op">$</span>V2 &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb39-3"><a href="inter-omics.html#cb39-3"></a>mena_pcc<span class="op">$</span>V4 &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb39-4"><a href="inter-omics.html#cb39-4"></a>mena_pcc<span class="op">$</span>V5 &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb39-5"><a href="inter-omics.html#cb39-5"></a><span class="kw">colnames</span>(mena_pcc) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;node1_mena&quot;</span>, <span class="st">&quot;node2_mena&quot;</span>, <span class="st">&quot;pMENA_PCC&quot;</span>)</span>
<span id="cb39-6"><a href="inter-omics.html#cb39-6"></a></span>
<span id="cb39-7"><a href="inter-omics.html#cb39-7"></a>mena_scc &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/MENA_0.820_SCC_edge_attribute.txt&quot;</span>, <span class="dt">header =</span> F, <span class="dt">sep =</span> <span class="st">&quot; &quot;</span>)</span>
<span id="cb39-8"><a href="inter-omics.html#cb39-8"></a><span class="kw">dim</span>(mena_scc)[<span class="dv">1</span>] <span class="op">-</span><span class="st"> </span><span class="kw">dim</span>(mena_scc[mena_scc<span class="op">$</span>V5 <span class="op">==</span><span class="st"> </span><span class="dv">-1</span>, ])[<span class="dv">1</span>]  <span class="co"># Number of pos interactions</span></span></code></pre></div>
<pre><code>## [1] 317</code></pre>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="inter-omics.html#cb41-1"></a>mena_scc[<span class="st">&quot;pMENA_SCC&quot;</span>] &lt;-<span class="st"> </span><span class="kw">ifelse</span>((mena_scc<span class="op">$</span>V5 <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>), <span class="ot">NA</span>, <span class="fl">0.001</span>)</span>
<span id="cb41-2"><a href="inter-omics.html#cb41-2"></a>mena_scc<span class="op">$</span>V2 &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb41-3"><a href="inter-omics.html#cb41-3"></a>mena_scc<span class="op">$</span>V4 &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb41-4"><a href="inter-omics.html#cb41-4"></a>mena_scc<span class="op">$</span>V5 &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb41-5"><a href="inter-omics.html#cb41-5"></a><span class="kw">colnames</span>(mena_scc) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;node1_mena&quot;</span>, <span class="st">&quot;node2_mena&quot;</span>, <span class="st">&quot;pMENA_SCC&quot;</span>)</span></code></pre></div>
<p>MENA network with Pearson correlation contained 3582 edges of which 3411 negative interactions were removed. For Spearman correlations, the network contained 2216 edges of which 1899 negative interactions were removed.</p>
</div>
<div id="lsa" class="section level4">
<h4><span class="header-section-number">4.3.2.4</span> LSA</h4>
<p>The next network data set is based on local similarity. It does not contain any negative values for LSA, so we do not exclude any edges.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="inter-omics.html#cb42-1"></a>lsa &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/fastLSA_for_networks.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb42-2"><a href="inter-omics.html#cb42-2"></a>lsa<span class="op">$</span>X &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb42-3"><a href="inter-omics.html#cb42-3"></a><span class="kw">colnames</span>(lsa) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;node1_lsa&quot;</span>, <span class="st">&quot;node2_lsa&quot;</span>, <span class="st">&quot;LSA&quot;</span>, <span class="st">&quot;pLSA&quot;</span>)</span>
<span id="cb42-4"><a href="inter-omics.html#cb42-4"></a><span class="kw">dim</span>(lsa)[<span class="dv">1</span>]  <span class="co"># Number of edges</span></span></code></pre></div>
<pre><code>## [1] 208</code></pre>
</div>
<div id="integration-of-the-networks" class="section level4">
<h4><span class="header-section-number">4.3.2.5</span> Integration of the networks</h4>
<p>Now we combine all five networks into one data set.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="inter-omics.html#cb44-1"></a>mic &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/MIC.csv&quot;</span>, <span class="dt">header =</span> T)</span>
<span id="cb44-2"><a href="inter-omics.html#cb44-2"></a>mic<span class="op">$</span>X &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb44-3"><a href="inter-omics.html#cb44-3"></a>mic<span class="op">$</span>regression &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb44-4"><a href="inter-omics.html#cb44-4"></a>mic<span class="op">$</span>p_regression &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb44-5"><a href="inter-omics.html#cb44-5"></a>master_summary &lt;-<span class="st"> </span>mic</span>
<span id="cb44-6"><a href="inter-omics.html#cb44-6"></a></span>
<span id="cb44-7"><a href="inter-omics.html#cb44-7"></a><span class="kw">library</span>(dplyr)</span>
<span id="cb44-8"><a href="inter-omics.html#cb44-8"></a>master_summary &lt;-<span class="st"> </span><span class="kw">full_join</span>(master_summary, lsa, <span class="dt">by =</span> <span class="kw">c</span>(<span class="dt">node1_mic =</span> <span class="st">&quot;node2_lsa&quot;</span>, </span>
<span id="cb44-9"><a href="inter-omics.html#cb44-9"></a>    <span class="dt">node2_mic =</span> <span class="st">&quot;node1_lsa&quot;</span>))</span>
<span id="cb44-10"><a href="inter-omics.html#cb44-10"></a><span class="co"># sum(!is.na(master_summary$pLSA))==nrow(lsa) #TRUE</span></span>
<span id="cb44-11"><a href="inter-omics.html#cb44-11"></a>master_summary &lt;-<span class="st"> </span><span class="kw">full_join</span>(master_summary, sparcc, <span class="dt">by =</span> <span class="kw">c</span>(<span class="dt">node1_mic =</span> <span class="st">&quot;node1_sparcc&quot;</span>, </span>
<span id="cb44-12"><a href="inter-omics.html#cb44-12"></a>    <span class="dt">node2_mic =</span> <span class="st">&quot;node2_sparcc&quot;</span>))</span>
<span id="cb44-13"><a href="inter-omics.html#cb44-13"></a><span class="co"># sum(!is.na(master_summary$pSparCC))==sum(!is.na(sparcc$pSparCC)) #TRUE</span></span>
<span id="cb44-14"><a href="inter-omics.html#cb44-14"></a>master_summary &lt;-<span class="st"> </span><span class="kw">full_join</span>(master_summary, mena_pcc, <span class="dt">by =</span> <span class="kw">c</span>(<span class="dt">node1_mic =</span> <span class="st">&quot;node2_mena&quot;</span>, </span>
<span id="cb44-15"><a href="inter-omics.html#cb44-15"></a>    <span class="dt">node2_mic =</span> <span class="st">&quot;node1_mena&quot;</span>))</span>
<span id="cb44-16"><a href="inter-omics.html#cb44-16"></a><span class="co"># sum(!is.na(master_summary$pMENA_PCC))==sum(!is.na(mena_pcc$pMENA_PCC)) #TRUE</span></span>
<span id="cb44-17"><a href="inter-omics.html#cb44-17"></a>master_summary &lt;-<span class="st"> </span><span class="kw">full_join</span>(master_summary, mena_scc, <span class="dt">by =</span> <span class="kw">c</span>(<span class="dt">node1_mic =</span> <span class="st">&quot;node2_mena&quot;</span>, </span>
<span id="cb44-18"><a href="inter-omics.html#cb44-18"></a>    <span class="dt">node2_mic =</span> <span class="st">&quot;node1_mena&quot;</span>))</span>
<span id="cb44-19"><a href="inter-omics.html#cb44-19"></a><span class="co"># sum(!is.na(master_summary$pMENA_SCC))==sum(!is.na(mena_scc$pMENA_SCC)) #TRUE</span></span>
<span id="cb44-20"><a href="inter-omics.html#cb44-20"></a>master_summary &lt;-<span class="st"> </span>master_summary[, <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">6</span>, <span class="dv">8</span>, <span class="dv">9</span>, <span class="dv">10</span>, <span class="dv">3</span>, <span class="dv">5</span>, <span class="dv">7</span>)]  <span class="co">#reorder columns</span></span>
<span id="cb44-21"><a href="inter-omics.html#cb44-21"></a><span class="kw">head</span>(master_summary)</span>
<span id="cb44-22"><a href="inter-omics.html#cb44-22"></a><span class="co"># write.csv(master_summary, &#39;data/master_summary_networks_1.csv&#39;, row.names =</span></span>
<span id="cb44-23"><a href="inter-omics.html#cb44-23"></a><span class="co"># FALSE)</span></span></code></pre></div>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="inter-omics.html#cb45-1"></a><span class="co"># For p-value merging: metap::sumlog, or EmpiricalBrownsMethod::EBM</span></span>
<span id="cb45-2"><a href="inter-omics.html#cb45-2"></a><span class="kw">library</span>(metap)</span>
<span id="cb45-3"><a href="inter-omics.html#cb45-3"></a></span>
<span id="cb45-4"><a href="inter-omics.html#cb45-4"></a>ms &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/master_summary_networks_1.csv&quot;</span>, <span class="dt">header =</span> T)</span>
<span id="cb45-5"><a href="inter-omics.html#cb45-5"></a></span>
<span id="cb45-6"><a href="inter-omics.html#cb45-6"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">3</span>))</span>
<span id="cb45-7"><a href="inter-omics.html#cb45-7"></a><span class="kw">hist</span>(ms<span class="op">$</span>pMIC)</span>
<span id="cb45-8"><a href="inter-omics.html#cb45-8"></a><span class="kw">hist</span>(ms<span class="op">$</span>pLSA)  <span class="co">#xlim = range(0,1)</span></span>
<span id="cb45-9"><a href="inter-omics.html#cb45-9"></a><span class="kw">hist</span>(ms<span class="op">$</span>pSparCC)</span>
<span id="cb45-10"><a href="inter-omics.html#cb45-10"></a><span class="kw">plot</span>(ms<span class="op">$</span>pMENA_PCC)</span>
<span id="cb45-11"><a href="inter-omics.html#cb45-11"></a><span class="kw">plot</span>(ms<span class="op">$</span>pMENA_SCC)</span>
<span id="cb45-12"><a href="inter-omics.html#cb45-12"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))</span></code></pre></div>
<p><img src="Oceanographic-setting-influences-the-prokaryote-community-and-metabolome-in-deep-sea-sponges_files/figure-html/integration2-1.png" width="80%" /></p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="inter-omics.html#cb46-1"></a><span class="co"># metap::sumlog doesn&#39;t think 0 is a valid p-value, replace all zeros with small</span></span>
<span id="cb46-2"><a href="inter-omics.html#cb46-2"></a><span class="co"># non-zero values, e.g. half-minimum</span></span>
<span id="cb46-3"><a href="inter-omics.html#cb46-3"></a>ms<span class="op">$</span>pSparCC[ms<span class="op">$</span>pSparCC <span class="op">==</span><span class="st"> </span><span class="dv">0</span>] &lt;-<span class="st"> </span><span class="fl">0.005</span></span>
<span id="cb46-4"><a href="inter-omics.html#cb46-4"></a>ms<span class="op">$</span>pMIC[ms<span class="op">$</span>pMIC <span class="op">==</span><span class="st"> </span><span class="dv">0</span>] &lt;-<span class="st"> </span><span class="fl">5e-04</span></span>
<span id="cb46-5"><a href="inter-omics.html#cb46-5"></a></span>
<span id="cb46-6"><a href="inter-omics.html#cb46-6"></a>ms[<span class="st">&quot;NA_count&quot;</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></span>
<span id="cb46-7"><a href="inter-omics.html#cb46-7"></a>ms[<span class="st">&quot;signif_0.05&quot;</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></span>
<span id="cb46-8"><a href="inter-omics.html#cb46-8"></a>ms[<span class="st">&quot;signif_0.001&quot;</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></span>
<span id="cb46-9"><a href="inter-omics.html#cb46-9"></a>ms[<span class="st">&quot;sumlog&quot;</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></span>
<span id="cb46-10"><a href="inter-omics.html#cb46-10"></a></span>
<span id="cb46-11"><a href="inter-omics.html#cb46-11"></a>n &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb46-12"><a href="inter-omics.html#cb46-12"></a>k &lt;-<span class="st"> </span><span class="kw">dim</span>(ms)[<span class="dv">1</span>]</span>
<span id="cb46-13"><a href="inter-omics.html#cb46-13"></a><span class="cf">while</span> (n <span class="op">&lt;</span><span class="st"> </span>k) {</span>
<span id="cb46-14"><a href="inter-omics.html#cb46-14"></a>    n &lt;-<span class="st"> </span>n <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb46-15"><a href="inter-omics.html#cb46-15"></a>    ms<span class="op">$</span>NA_count[n] &lt;-<span class="st"> </span><span class="kw">sum</span>(<span class="kw">is.na</span>(ms[n, <span class="dv">3</span><span class="op">:</span><span class="dv">7</span>]))</span>
<span id="cb46-16"><a href="inter-omics.html#cb46-16"></a>    ms<span class="op">$</span>signif_<span class="fl">0.05</span>[n] &lt;-<span class="st"> </span><span class="kw">sum</span>(ms[n, <span class="dv">3</span><span class="op">:</span><span class="dv">7</span>] <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.05</span>, <span class="dt">na.rm =</span> T)</span>
<span id="cb46-17"><a href="inter-omics.html#cb46-17"></a>    ms<span class="op">$</span>signif_<span class="fl">0.001</span>[n] &lt;-<span class="st"> </span><span class="kw">sum</span>(ms[n, <span class="dv">3</span><span class="op">:</span><span class="dv">7</span>] <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.001</span>, <span class="dt">na.rm =</span> T)</span>
<span id="cb46-18"><a href="inter-omics.html#cb46-18"></a>    <span class="kw">ifelse</span>((ms<span class="op">$</span>NA_count[n] <span class="op">&lt;=</span><span class="st"> </span><span class="dv">3</span>), (ms<span class="op">$</span>sumlog[n] &lt;-<span class="st"> </span><span class="kw">sumlog</span>(ms[n, <span class="dv">3</span><span class="op">:</span><span class="dv">7</span>][<span class="op">!</span><span class="kw">is.na</span>(ms[n, </span>
<span id="cb46-19"><a href="inter-omics.html#cb46-19"></a>        <span class="dv">3</span><span class="op">:</span><span class="dv">7</span>])])<span class="op">$</span>p), <span class="ot">NA</span>)</span>
<span id="cb46-20"><a href="inter-omics.html#cb46-20"></a>}</span>
<span id="cb46-21"><a href="inter-omics.html#cb46-21"></a><span class="kw">rm</span>(k, n)</span>
<span id="cb46-22"><a href="inter-omics.html#cb46-22"></a></span>
<span id="cb46-23"><a href="inter-omics.html#cb46-23"></a>ms[<span class="st">&quot;p.adjust_Bonferroni&quot;</span>] &lt;-<span class="st"> </span><span class="kw">p.adjust</span>(ms<span class="op">$</span>sumlog, <span class="dt">method =</span> <span class="st">&quot;bonferroni&quot;</span>)</span>
<span id="cb46-24"><a href="inter-omics.html#cb46-24"></a>ms[<span class="st">&quot;p.adjust_FDR&quot;</span>] &lt;-<span class="st"> </span><span class="kw">p.adjust</span>(ms<span class="op">$</span>sumlog, <span class="dt">method =</span> <span class="st">&quot;fdr&quot;</span>)  <span class="co">#aka Benjamini &amp; Hochberg</span></span>
<span id="cb46-25"><a href="inter-omics.html#cb46-25"></a></span>
<span id="cb46-26"><a href="inter-omics.html#cb46-26"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>))</span>
<span id="cb46-27"><a href="inter-omics.html#cb46-27"></a><span class="kw">hist</span>(ms<span class="op">$</span>signif_<span class="fl">0.05</span>, <span class="dt">main =</span> <span class="st">&quot;p-values &lt;= 0.05&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;Counts per edge &quot;</span>, <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, </span>
<span id="cb46-28"><a href="inter-omics.html#cb46-28"></a>    <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>), <span class="dt">labels =</span> <span class="ot">TRUE</span>)</span>
<span id="cb46-29"><a href="inter-omics.html#cb46-29"></a><span class="kw">hist</span>(ms<span class="op">$</span>signif_<span class="fl">0.001</span>, <span class="dt">main =</span> <span class="st">&quot;p-values &lt;= 0.001&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;Counts per edge &quot;</span>, <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, </span>
<span id="cb46-30"><a href="inter-omics.html#cb46-30"></a>    <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>), <span class="dt">labels =</span> <span class="ot">TRUE</span>)</span>
<span id="cb46-31"><a href="inter-omics.html#cb46-31"></a><span class="kw">hist</span>(ms<span class="op">$</span>NA_count, <span class="dt">main =</span> <span class="st">&quot;NAs&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;Counts per edge &quot;</span>, <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, </span>
<span id="cb46-32"><a href="inter-omics.html#cb46-32"></a>    <span class="dv">4</span>, <span class="dv">5</span>), <span class="dt">labels =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="Oceanographic-setting-influences-the-prokaryote-community-and-metabolome-in-deep-sea-sponges_files/figure-html/integration2-2.png" width="80%" /></p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="inter-omics.html#cb47-1"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb47-2"><a href="inter-omics.html#cb47-2"></a></span>
<span id="cb47-3"><a href="inter-omics.html#cb47-3"></a>ds &lt;-<span class="st"> </span><span class="kw">split</span>(ms, ms<span class="op">$</span>signif_<span class="fl">0.05</span>)</span>
<span id="cb47-4"><a href="inter-omics.html#cb47-4"></a>raw_nodes &lt;-<span class="st"> </span><span class="kw">rbind</span>(ds<span class="op">$</span><span class="st">`</span><span class="dt">4</span><span class="st">`</span>, ds<span class="op">$</span><span class="st">`</span><span class="dt">5</span><span class="st">`</span>)  <span class="co"># Selection/inclusion criterion</span></span>
<span id="cb47-5"><a href="inter-omics.html#cb47-5"></a><span class="kw">rm</span>(ds)</span>
<span id="cb47-6"><a href="inter-omics.html#cb47-6"></a></span>
<span id="cb47-7"><a href="inter-omics.html#cb47-7"></a>edges &lt;-<span class="st"> </span>raw_nodes[, <span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]</span>
<span id="cb47-8"><a href="inter-omics.html#cb47-8"></a>nodes &lt;-<span class="st"> </span><span class="kw">union</span>(raw_nodes<span class="op">$</span>node1_mic, raw_nodes<span class="op">$</span>node2_mic)</span>
<span id="cb47-9"><a href="inter-omics.html#cb47-9"></a></span>
<span id="cb47-10"><a href="inter-omics.html#cb47-10"></a><span class="kw">write.csv</span>(edges, <span class="st">&quot;data/master_summary_networks_2.csv&quot;</span>, <span class="dt">row.names =</span> F)</span></code></pre></div>
<p>We selected all edges with at least 4 p-values <span class="math inline">\(\leq\)</span> 0.05 to visualise in the network. This c network has 136 edges and 113 nodes.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="inter-omics.html#cb48-1"></a><span class="co"># THANK YOU: https://kateto.net/networks-r-igraph</span></span>
<span id="cb48-2"><a href="inter-omics.html#cb48-2"></a></span>
<span id="cb48-3"><a href="inter-omics.html#cb48-3"></a>edges &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/master_summary_networks_2.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb48-4"><a href="inter-omics.html#cb48-4"></a>nodes &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">union</span>(raw_nodes<span class="op">$</span>node1_mic, raw_nodes<span class="op">$</span>node2_mic))</span>
<span id="cb48-5"><a href="inter-omics.html#cb48-5"></a></span>
<span id="cb48-6"><a href="inter-omics.html#cb48-6"></a><span class="co"># annotation data required: inc-dec/depth response, taxonomy, barettin corelation</span></span>
<span id="cb48-7"><a href="inter-omics.html#cb48-7"></a><span class="co"># hex color code</span></span>
<span id="cb48-8"><a href="inter-omics.html#cb48-8"></a>depth &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/gb_OTUs_overall_rabdc_annotated.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb48-9"><a href="inter-omics.html#cb48-9"></a>taxonomy &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/microbiome_taxonomy.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb48-10"><a href="inter-omics.html#cb48-10"></a>barettin &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/GB_OTU_barettin_correlation.csv&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb48-11"><a href="inter-omics.html#cb48-11"></a></span>
<span id="cb48-12"><a href="inter-omics.html#cb48-12"></a><span class="co"># matching IDs</span></span>
<span id="cb48-13"><a href="inter-omics.html#cb48-13"></a><span class="kw">colnames</span>(nodes) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;OTU_long&quot;</span>)</span>
<span id="cb48-14"><a href="inter-omics.html#cb48-14"></a>nodes[<span class="st">&quot;OTU&quot;</span>] &lt;-<span class="st"> </span><span class="kw">str_replace</span>(nodes<span class="op">$</span>OTU_long, <span class="st">&quot;OTU196900&quot;</span>, <span class="st">&quot;&quot;</span>)</span>
<span id="cb48-15"><a href="inter-omics.html#cb48-15"></a>depth[<span class="st">&quot;OTU&quot;</span>] &lt;-<span class="st"> </span><span class="kw">str_replace</span>(depth<span class="op">$</span>XOTU, <span class="st">&quot;X196900&quot;</span>, <span class="st">&quot;&quot;</span>)</span>
<span id="cb48-16"><a href="inter-omics.html#cb48-16"></a>taxonomy[<span class="st">&quot;OTU&quot;</span>] &lt;-<span class="st"> </span><span class="kw">str_replace</span>(taxonomy<span class="op">$</span>OTU_ID, <span class="st">&quot;196900&quot;</span>, <span class="st">&quot;&quot;</span>)</span>
<span id="cb48-17"><a href="inter-omics.html#cb48-17"></a>barettin[<span class="st">&quot;OTU&quot;</span>] &lt;-<span class="st"> </span><span class="kw">str_replace</span>(barettin<span class="op">$</span>XOTU, <span class="st">&quot;X196900&quot;</span>, <span class="st">&quot;&quot;</span>)</span>
<span id="cb48-18"><a href="inter-omics.html#cb48-18"></a></span>
<span id="cb48-19"><a href="inter-omics.html#cb48-19"></a><span class="kw">colnames</span>(taxonomy)</span></code></pre></div>
<pre><code>## [1] &quot;OTU_ID&quot;  &quot;Kingdom&quot; &quot;Phylum&quot;  &quot;Class&quot;   &quot;Order&quot;   &quot;Family&quot;  &quot;Genus&quot;  
## [8] &quot;OTU&quot;</code></pre>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="inter-omics.html#cb50-1"></a><span class="co"># downsizing to relevant columns nodes$OTU_long &lt;- NULL</span></span>
<span id="cb50-2"><a href="inter-omics.html#cb50-2"></a>depth &lt;-<span class="st"> </span>depth[, <span class="kw">c</span>(<span class="st">&quot;ttest_pval&quot;</span>, <span class="st">&quot;ttest_fdr&quot;</span>, <span class="st">&quot;inc_dec_estimate&quot;</span>, <span class="st">&quot;inc_dec_p_val&quot;</span>, </span>
<span id="cb50-3"><a href="inter-omics.html#cb50-3"></a>    <span class="st">&quot;fdr&quot;</span>, <span class="st">&quot;classification&quot;</span>, <span class="st">&quot;OTU&quot;</span>)]</span>
<span id="cb50-4"><a href="inter-omics.html#cb50-4"></a><span class="kw">colnames</span>(depth) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;ttest_pval&quot;</span>, <span class="st">&quot;ttest_fdr&quot;</span>, <span class="st">&quot;inc_dec_estimate&quot;</span>, <span class="st">&quot;inc_dec_p_val&quot;</span>, </span>
<span id="cb50-5"><a href="inter-omics.html#cb50-5"></a>    <span class="st">&quot;inc_dec_fdr&quot;</span>, <span class="st">&quot;inc_dec_classification&quot;</span>, <span class="st">&quot;OTU&quot;</span>)</span>
<span id="cb50-6"><a href="inter-omics.html#cb50-6"></a>barettin<span class="op">$</span>XOTU &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb50-7"><a href="inter-omics.html#cb50-7"></a>taxonomy &lt;-<span class="st"> </span>taxonomy[, <span class="kw">c</span>(<span class="st">&quot;Kingdom&quot;</span>, <span class="st">&quot;Phylum&quot;</span>, <span class="st">&quot;Class&quot;</span>, <span class="st">&quot;OTU&quot;</span>)]</span>
<span id="cb50-8"><a href="inter-omics.html#cb50-8"></a>taxonomy[] &lt;-<span class="st"> </span><span class="kw">lapply</span>(taxonomy, str_trim)</span>
<span id="cb50-9"><a href="inter-omics.html#cb50-9"></a></span>
<span id="cb50-10"><a href="inter-omics.html#cb50-10"></a>nodes &lt;-<span class="st"> </span><span class="kw">left_join</span>(nodes, depth)</span>
<span id="cb50-11"><a href="inter-omics.html#cb50-11"></a>nodes &lt;-<span class="st"> </span><span class="kw">left_join</span>(nodes, barettin)</span>
<span id="cb50-12"><a href="inter-omics.html#cb50-12"></a>nodes &lt;-<span class="st"> </span><span class="kw">left_join</span>(nodes, taxonomy)</span>
<span id="cb50-13"><a href="inter-omics.html#cb50-13"></a></span>
<span id="cb50-14"><a href="inter-omics.html#cb50-14"></a><span class="co"># Adding categorical information</span></span>
<span id="cb50-15"><a href="inter-omics.html#cb50-15"></a>nodes[<span class="st">&quot;barettin_c&quot;</span>] &lt;-<span class="st"> </span><span class="kw">ifelse</span>(nodes<span class="op">$</span>barettin_estimate_P <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span> <span class="op">&amp;</span><span class="st"> </span>nodes<span class="op">$</span>barettin_p_val_P <span class="op">&lt;</span><span class="st"> </span></span>
<span id="cb50-16"><a href="inter-omics.html#cb50-16"></a><span class="st">    </span><span class="fl">0.05</span>, <span class="dv">1</span>, <span class="dv">0</span>)  <span class="co"># for scaling node size</span></span>
<span id="cb50-17"><a href="inter-omics.html#cb50-17"></a>nodes[<span class="st">&quot;inc_dec_c&quot;</span>] &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb50-18"><a href="inter-omics.html#cb50-18"></a>nodes<span class="op">$</span>inc_dec_c[nodes<span class="op">$</span>inc_dec_p_val <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>] &lt;-<span class="st"> </span><span class="dv">1</span>  <span class="co"># for scaling node size</span></span>
<span id="cb50-19"><a href="inter-omics.html#cb50-19"></a>nodes[<span class="st">&quot;inc_dec_c_group&quot;</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></span>
<span id="cb50-20"><a href="inter-omics.html#cb50-20"></a>nodes<span class="op">$</span>inc_dec_c_group[nodes<span class="op">$</span>inc_dec_c <span class="op">==</span><span class="st"> </span><span class="dv">1</span> <span class="op">&amp;</span><span class="st"> </span>nodes<span class="op">$</span>inc_dec_estimate <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;deep&quot;</span>)  <span class="co"># for colouring shallow vs. deep</span></span>
<span id="cb50-21"><a href="inter-omics.html#cb50-21"></a>nodes<span class="op">$</span>inc_dec_c_group[nodes<span class="op">$</span>inc_dec_c <span class="op">==</span><span class="st"> </span><span class="dv">1</span> <span class="op">&amp;</span><span class="st"> </span>nodes<span class="op">$</span>inc_dec_estimate <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;shallow&quot;</span>)  <span class="co"># for colouring shallow vs. deep</span></span>
<span id="cb50-22"><a href="inter-omics.html#cb50-22"></a><span class="co"># Pearson pmcc &gt; 0 &amp; p &lt; 0.05 56 fdr &lt; 0.05 22 Spearman rho &gt; 0 &amp; p &lt; 0.05 42 fdr</span></span>
<span id="cb50-23"><a href="inter-omics.html#cb50-23"></a><span class="co"># &lt; 0.05 1</span></span>
<span id="cb50-24"><a href="inter-omics.html#cb50-24"></a></span>
<span id="cb50-25"><a href="inter-omics.html#cb50-25"></a>nodes[] &lt;-<span class="st"> </span><span class="kw">lapply</span>(nodes, <span class="cf">function</span>(x) <span class="cf">if</span> (<span class="kw">is.factor</span>(x)) <span class="kw">factor</span>(x) <span class="cf">else</span> x)</span>
<span id="cb50-26"><a href="inter-omics.html#cb50-26"></a></span>
<span id="cb50-27"><a href="inter-omics.html#cb50-27"></a><span class="kw">library</span>(igraph)</span>
<span id="cb50-28"><a href="inter-omics.html#cb50-28"></a>net &lt;-<span class="st"> </span><span class="kw">graph_from_data_frame</span>(<span class="dt">d =</span> edges, <span class="dt">vertices =</span> nodes, <span class="dt">directed =</span> F)</span>
<span id="cb50-29"><a href="inter-omics.html#cb50-29"></a>l &lt;-<span class="st"> </span><span class="kw">layout_with_kk</span>(net)</span>
<span id="cb50-30"><a href="inter-omics.html#cb50-30"></a></span>
<span id="cb50-31"><a href="inter-omics.html#cb50-31"></a><span class="co"># 11 colourblind-friendly colours from https://medialab.github.io/iwanthue/</span></span>
<span id="cb50-32"><a href="inter-omics.html#cb50-32"></a></span>
<span id="cb50-33"><a href="inter-omics.html#cb50-33"></a><span class="co"># Taxonomy coloring &amp; barettin: Class</span></span>
<span id="cb50-34"><a href="inter-omics.html#cb50-34"></a>ecol &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="st">&quot;gray80&quot;</span>, <span class="kw">ecount</span>(net))</span>
<span id="cb50-35"><a href="inter-omics.html#cb50-35"></a>vcol &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="st">&quot;grey40&quot;</span>, <span class="kw">vcount</span>(net))</span>
<span id="cb50-36"><a href="inter-omics.html#cb50-36"></a>vcol[<span class="kw">V</span>(net)<span class="op">$</span>Class <span class="op">==</span><span class="st"> &quot;Subgroup_26&quot;</span>] &lt;-<span class="st"> &quot;#628ed6&quot;</span></span>
<span id="cb50-37"><a href="inter-omics.html#cb50-37"></a>vcol[<span class="kw">V</span>(net)<span class="op">$</span>Class <span class="op">==</span><span class="st"> &quot;Subgroup_15&quot;</span>] &lt;-<span class="st"> &quot;#957d34&quot;</span></span>
<span id="cb50-38"><a href="inter-omics.html#cb50-38"></a>vcol[<span class="kw">V</span>(net)<span class="op">$</span>Class <span class="op">==</span><span class="st"> &quot;Subgroup_6&quot;</span>] &lt;-<span class="st"> &quot;#45c097&quot;</span></span>
<span id="cb50-39"><a href="inter-omics.html#cb50-39"></a>vcol[<span class="kw">V</span>(net)<span class="op">$</span>Class <span class="op">==</span><span class="st"> &quot;Anaerolineae&quot;</span>] &lt;-<span class="st"> &quot;#ba4758&quot;</span></span>
<span id="cb50-40"><a href="inter-omics.html#cb50-40"></a>vcol[<span class="kw">V</span>(net)<span class="op">$</span>Class <span class="op">==</span><span class="st"> &quot;JG30-KF-CM66&quot;</span>] &lt;-<span class="st"> &quot;#b2467e&quot;</span></span>
<span id="cb50-41"><a href="inter-omics.html#cb50-41"></a>vcol[<span class="kw">V</span>(net)<span class="op">$</span>Class <span class="op">==</span><span class="st"> &quot;SAR202_clade&quot;</span>] &lt;-<span class="st"> &quot;#5b3687&quot;</span></span>
<span id="cb50-42"><a href="inter-omics.html#cb50-42"></a>vcol[<span class="kw">V</span>(net)<span class="op">$</span>Class <span class="op">==</span><span class="st"> &quot;TK10&quot;</span>] &lt;-<span class="st"> &quot;#ba5437&quot;</span></span>
<span id="cb50-43"><a href="inter-omics.html#cb50-43"></a>vcol[<span class="kw">V</span>(net)<span class="op">$</span>Class <span class="op">==</span><span class="st"> &quot;BD2-11_terrestrial_group&quot;</span>] &lt;-<span class="st"> &quot;#69ab54&quot;</span></span>
<span id="cb50-44"><a href="inter-omics.html#cb50-44"></a>vcol[<span class="kw">V</span>(net)<span class="op">$</span>Class <span class="op">==</span><span class="st"> &quot;Alphaproteobacteria&quot;</span>] &lt;-<span class="st"> &quot;#c777cb&quot;</span></span>
<span id="cb50-45"><a href="inter-omics.html#cb50-45"></a>vcol[<span class="kw">V</span>(net)<span class="op">$</span>Class <span class="op">==</span><span class="st"> &quot;JTB23&quot;</span>] &lt;-<span class="st"> &quot;#c3a63e&quot;</span></span>
<span id="cb50-46"><a href="inter-omics.html#cb50-46"></a>vcol[<span class="kw">V</span>(net)<span class="op">$</span>Class <span class="op">==</span><span class="st"> &quot;Acidimicrobiia&quot;</span>] &lt;-<span class="st"> &quot;#6a70d7&quot;</span></span>
<span id="cb50-47"><a href="inter-omics.html#cb50-47"></a></span>
<span id="cb50-48"><a href="inter-omics.html#cb50-48"></a>colrs &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;#628ed6&quot;</span>, <span class="st">&quot;#957d34&quot;</span>, <span class="st">&quot;#45c097&quot;</span>, <span class="st">&quot;#ba4758&quot;</span>, <span class="st">&quot;#b2467e&quot;</span>, <span class="st">&quot;#5b3687&quot;</span>, <span class="st">&quot;#ba5437&quot;</span>, </span>
<span id="cb50-49"><a href="inter-omics.html#cb50-49"></a>    <span class="st">&quot;#69ab54&quot;</span>, <span class="st">&quot;#c777cb&quot;</span>, <span class="st">&quot;#c3a63e&quot;</span>, <span class="st">&quot;#6a70d7&quot;</span>)</span>
<span id="cb50-50"><a href="inter-omics.html#cb50-50"></a><span class="kw">V</span>(net)<span class="op">$</span>color &lt;-<span class="st"> </span>colrs[<span class="kw">V</span>(net)<span class="op">$</span>class]</span>
<span id="cb50-51"><a href="inter-omics.html#cb50-51"></a></span>
<span id="cb50-52"><a href="inter-omics.html#cb50-52"></a><span class="co"># set vertex.label=V(net)$OTU for OTU numbers</span></span>
<span id="cb50-53"><a href="inter-omics.html#cb50-53"></a><span class="kw">plot</span>(net, <span class="dt">vertex.color =</span> vcol, <span class="dt">edge.color =</span> ecol, <span class="dt">vertex.size =</span> <span class="kw">V</span>(net)<span class="op">$</span>barettin_c <span class="op">*</span><span class="st"> </span></span>
<span id="cb50-54"><a href="inter-omics.html#cb50-54"></a><span class="st">    </span><span class="dv">10</span>, <span class="dt">vertex.label =</span> <span class="ot">NA</span>, <span class="dt">layout =</span> l)</span>
<span id="cb50-55"><a href="inter-omics.html#cb50-55"></a><span class="kw">legend</span>(<span class="dt">x =</span> <span class="dv">5</span>, <span class="dt">y =</span> <span class="fl">0.5</span>, <span class="kw">c</span>(<span class="st">&quot;Subgroup_26 (Acidobacteria)&quot;</span>, <span class="st">&quot;Subgroup_15 (Acidobacteria)&quot;</span>, </span>
<span id="cb50-56"><a href="inter-omics.html#cb50-56"></a>    <span class="st">&quot;Subgroup_6 (Acidobacteria)&quot;</span>, <span class="st">&quot;Anaerolineae (Chloroflexi)&quot;</span>, <span class="st">&quot;JG30-KF-CM66 (Chloroflexi)&quot;</span>, </span>
<span id="cb50-57"><a href="inter-omics.html#cb50-57"></a>    <span class="st">&quot;SAR202_clade (Chloroflexi)&quot;</span>, <span class="st">&quot;TK10 (Chloroflexi)&quot;</span>, <span class="st">&quot;BD2-11_terrestrial_group (Gemmatimonadetes)&quot;</span>, </span>
<span id="cb50-58"><a href="inter-omics.html#cb50-58"></a>    <span class="st">&quot;Alphaproteobacteria (Proteobacteria)&quot;</span>, <span class="st">&quot;JTB23 (Proteobacteria)&quot;</span>, <span class="st">&quot;Acidimicrobiia (Actinobacteria)&quot;</span>), </span>
<span id="cb50-59"><a href="inter-omics.html#cb50-59"></a>    <span class="dt">pt.bg =</span> colrs, <span class="dt">pch =</span> <span class="dv">21</span>, <span class="dt">col =</span> <span class="st">&quot;#777777&quot;</span>, <span class="dt">pt.cex =</span> <span class="fl">1.5</span>, <span class="dt">cex =</span> <span class="fl">0.7</span>, <span class="dt">bty =</span> <span class="st">&quot;n&quot;</span>, </span>
<span id="cb50-60"><a href="inter-omics.html#cb50-60"></a>    <span class="dt">ncol =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="Oceanographic-setting-influences-the-prokaryote-community-and-metabolome-in-deep-sea-sponges_files/figure-html/igraph1-1.png" width="672" /></p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="inter-omics.html#cb51-1"></a><span class="co"># Depth by correlation</span></span>
<span id="cb51-2"><a href="inter-omics.html#cb51-2"></a>ecol &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="st">&quot;gray80&quot;</span>, <span class="kw">ecount</span>(net))</span>
<span id="cb51-3"><a href="inter-omics.html#cb51-3"></a>vcol &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="st">&quot;grey40&quot;</span>, <span class="kw">vcount</span>(net))</span>
<span id="cb51-4"><a href="inter-omics.html#cb51-4"></a>vcol[<span class="kw">V</span>(net)<span class="op">$</span>inc_dec_c_group <span class="op">==</span><span class="st"> &quot;shallow&quot;</span>] &lt;-<span class="st"> &quot;gold&quot;</span></span>
<span id="cb51-5"><a href="inter-omics.html#cb51-5"></a>vcol[<span class="kw">V</span>(net)<span class="op">$</span>inc_dec_c_group <span class="op">==</span><span class="st"> &quot;deep&quot;</span>] &lt;-<span class="st"> &quot;blue&quot;</span></span>
<span id="cb51-6"><a href="inter-omics.html#cb51-6"></a></span>
<span id="cb51-7"><a href="inter-omics.html#cb51-7"></a><span class="kw">V</span>(net)<span class="op">$</span>color &lt;-<span class="st"> </span>colrs[<span class="kw">V</span>(net)<span class="op">$</span>inc_dec_c_group]</span>
<span id="cb51-8"><a href="inter-omics.html#cb51-8"></a>colrs &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;gold&quot;</span>, <span class="st">&quot;blue&quot;</span>)</span>
<span id="cb51-9"><a href="inter-omics.html#cb51-9"></a></span>
<span id="cb51-10"><a href="inter-omics.html#cb51-10"></a><span class="kw">plot</span>(net, <span class="dt">vertex.color =</span> vcol, <span class="dt">edge.color =</span> ecol, <span class="dt">vertex.size =</span> <span class="kw">V</span>(net)<span class="op">$</span>inc_dec_c <span class="op">*</span><span class="st"> </span></span>
<span id="cb51-11"><a href="inter-omics.html#cb51-11"></a><span class="st">    </span><span class="dv">10</span>, <span class="dt">vertex.label =</span> <span class="ot">NA</span>, <span class="dt">layout =</span> l)</span></code></pre></div>
<p><img src="Oceanographic-setting-influences-the-prokaryote-community-and-metabolome-in-deep-sea-sponges_files/figure-html/igraph1-2.png" width="672" /></p>
</div>
</div>
</div>
<div id="shortlist" class="section level2">
<h2><span class="header-section-number">4.4</span> Shortlist</h2>
<p>We believe the producer of barettin (and related compounds) to have the following properties:</p>
<ul>
<li>common (average relative abundance &gt; 0.25%)</li>
<li>specific to <em>G. barretti</em></li>
<li>positively correlated with barettin</li>
</ul>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="inter-omics.html#cb52-1"></a><span class="kw">head</span>(rabdc_gb)</span>
<span id="cb52-2"><a href="inter-omics.html#cb52-2"></a><span class="kw">head</span>(rabdc_sf)</span>
<span id="cb52-3"><a href="inter-omics.html#cb52-3"></a><span class="kw">setdiff</span>(<span class="kw">rownames</span>(rabdc_gb), <span class="kw">rownames</span>(rabdc_sf))  <span class="co">#Setdiff finds rows that appear in first table but not in second</span></span>
<span id="cb52-4"><a href="inter-omics.html#cb52-4"></a>gb_unique &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">setdiff</span>(<span class="kw">rownames</span>(rabdc_gb), <span class="kw">rownames</span>(rabdc_sf)))</span>
<span id="cb52-5"><a href="inter-omics.html#cb52-5"></a><span class="kw">colnames</span>(gb_unique) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;XOTU&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="inter-omics.html#cb53-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 3.5.1 (2018-07-02)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS  10.15.5
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] igraph_1.2.5          metap_1.3             minerva_1.5.8        
##  [4] kableExtra_1.1.0.9000 vegan_2.5-6           lattice_0.20-41      
##  [7] permute_0.9-5         forcats_0.5.0         stringr_1.4.0        
## [10] dplyr_0.8.5           purrr_0.3.3           readr_1.3.1          
## [13] tidyr_1.0.2           tibble_3.0.0          ggplot2_3.3.0        
## [16] tidyverse_1.3.0       knitr_1.28           
## 
## loaded via a namespace (and not attached):
##  [1] nlme_3.1-145        fs_1.4.0            lubridate_1.7.4    
##  [4] webshot_0.5.2       httr_1.4.1          numDeriv_2016.8-1.1
##  [7] tools_3.5.1         backports_1.1.5     R6_2.4.1           
## [10] BiocGenerics_0.28.0 DBI_1.1.0           mgcv_1.8-31        
## [13] colorspace_1.4-1    sn_1.6-1            withr_2.1.2        
## [16] tidyselect_1.0.0    mnormt_1.5-6        compiler_3.5.1     
## [19] Biobase_2.42.0      cli_2.0.2           rvest_0.3.5        
## [22] formatR_1.7         xml2_1.3.0          TFisher_0.2.0      
## [25] sandwich_2.5-1      bookdown_0.18       scales_1.1.0       
## [28] mvtnorm_1.1-0       digest_0.6.25       rmarkdown_2.1      
## [31] pkgconfig_2.0.3     htmltools_0.4.0     bibtex_0.4.2.2     
## [34] plotrix_3.7-7       dbplyr_1.4.2        highr_0.8          
## [37] rlang_0.4.5         readxl_1.3.1        rstudioapi_0.11    
## [40] generics_0.0.2      zoo_1.8-7           jsonlite_1.6.1     
## [43] magrittr_1.5        Matrix_1.2-18       Rcpp_1.0.4         
## [46] munsell_0.5.0       fansi_0.4.1         lifecycle_0.2.0    
## [49] stringi_1.4.6       multcomp_1.4-12     yaml_2.2.1         
## [52] gbRd_0.4-11         MASS_7.3-51.5       grid_3.5.1         
## [55] parallel_3.5.1      crayon_1.3.4        haven_2.2.0        
## [58] splines_3.5.1       multtest_2.38.0     hms_0.5.3          
## [61] pillar_1.4.3        codetools_0.2-16    stats4_3.5.1       
## [64] reprex_0.3.0        mutoss_0.1-12       glue_1.3.2         
## [67] evaluate_0.14       modelr_0.1.6        vctrs_0.2.4        
## [70] Rdpack_0.11-1       cellranger_1.1.0    gtable_0.3.0       
## [73] assertthat_0.2.1    xfun_0.13.1         broom_0.5.5        
## [76] survival_3.1-11     viridisLite_0.3.0   cluster_2.1.0      
## [79] TH.data_1.0-10      ellipsis_0.3.0</code></pre>


</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="microbiota.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="funding-disclaimer.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Oceanographic-setting-influences-the-prokaryote-community-and-metabolome-in-deep-sea-sponges.pdf", "Oceanographic-setting-influences-the-prokaryote-community-and-metabolome-in-deep-sea-sponges.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
